<!DOCTYPE HTML>
<html lang="en">


<head>
    <meta charset="utf-8">
    <meta name="keywords" content="Aligning sequences, Chaoqun Yin">
    <meta name="description" content="Veni, Vidi, Vici">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="renderer" content="webkit|ie-stand|ie-comp">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="format-detection" content="telephone=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <!-- Global site tag (gtag.js) - Google Analytics -->


    <title>Aligning sequences | Chaoqun Yin</title>
    <link rel="icon" type="image/png" href="/favicon.png">

    <link rel="stylesheet" type="text/css" href="/libs/awesome/css/all.css">
    <link rel="stylesheet" type="text/css" href="/libs/materialize/materialize.min.css">
    <link rel="stylesheet" type="text/css" href="/libs/aos/aos.css">
    <link rel="stylesheet" type="text/css" href="/libs/animate/animate.min.css">
    <link rel="stylesheet" type="text/css" href="/libs/lightGallery/css/lightgallery.min.css">
    <link rel="stylesheet" type="text/css" href="/css/matery.css">
    <link rel="stylesheet" type="text/css" href="/css/my.css">

    <script src="/libs/jquery/jquery.min.js"></script>

<meta name="generator" content="Hexo 5.3.0"></head>




<body>
    <header class="navbar-fixed">
    <nav id="headNav" class="bg-color nav-transparent">
        <div id="navContainer" class="nav-wrapper container">
            <div class="brand-logo">
                <a href="/" class="waves-effect waves-light">
                    
                    <img src="/medias/logo.png" class="logo-img" alt="LOGO">
                    
                    <span class="logo-span">Chaoqun Yin</span>
                </a>
            </div>
            

<a href="#" data-target="mobile-nav" class="sidenav-trigger button-collapse"><i class="fas fa-bars"></i></a>
<ul class="right nav-menu">
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/" class="waves-effect waves-light">
      
      <i class="fas fa-home" style="zoom: 0.6;"></i>
      
      <span>Index</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/tags" class="waves-effect waves-light">
      
      <i class="fas fa-tags" style="zoom: 0.6;"></i>
      
      <span>Tags</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/categories" class="waves-effect waves-light">
      
      <i class="fas fa-bookmark" style="zoom: 0.6;"></i>
      
      <span>Categories</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/archives" class="waves-effect waves-light">
      
      <i class="fas fa-archive" style="zoom: 0.6;"></i>
      
      <span>Archives</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/about" class="waves-effect waves-light">
      
      <i class="fas fa-user-circle" style="zoom: 0.6;"></i>
      
      <span>About</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/contact" class="waves-effect waves-light">
      
      <i class="fas fa-comments" style="zoom: 0.6;"></i>
      
      <span>Contact</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/friends" class="waves-effect waves-light">
      
      <i class="fas fa-address-book" style="zoom: 0.6;"></i>
      
      <span>Friends</span>
    </a>
    
  </li>
  
  <li>
    <a href="#searchModal" class="modal-trigger waves-effect waves-light">
      <i id="searchIcon" class="fas fa-search" title="Search" style="zoom: 0.85;"></i>
    </a>
  </li>
</ul>


<div id="mobile-nav" class="side-nav sidenav">

    <div class="mobile-head bg-color">
        
        <img src="/medias/logo.png" class="logo-img circle responsive-img">
        
        <div class="logo-name">Chaoqun Yin</div>
        <div class="logo-desc">
            
            Veni, Vidi, Vici
            
        </div>
    </div>

    

    <ul class="menu-list mobile-menu-list">
        
        <li class="m-nav-item">
	  
		<a href="/" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-home"></i>
			
			Index
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/tags" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-tags"></i>
			
			Tags
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/categories" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-bookmark"></i>
			
			Categories
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/archives" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-archive"></i>
			
			Archives
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/about" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-user-circle"></i>
			
			About
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/contact" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-comments"></i>
			
			Contact
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/friends" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-address-book"></i>
			
			Friends
		</a>
          
        </li>
        
        
        <li><div class="divider"></div></li>
        <li>
            <a href="https://github.com/blinkfox/hexo-theme-matery" class="waves-effect waves-light" target="_blank">
                <i class="fab fa-github-square fa-fw"></i>Fork Me
            </a>
        </li>
        
    </ul>
</div>


        </div>

        
            <style>
    .nav-transparent .github-corner {
        display: none !important;
    }

    .github-corner {
        position: absolute;
        z-index: 10;
        top: 0;
        right: 0;
        border: 0;
        transform: scale(1.1);
    }

    .github-corner svg {
        color: #0f9d58;
        fill: #fff;
        height: 64px;
        width: 64px;
    }

    .github-corner:hover .octo-arm {
        animation: a 0.56s ease-in-out;
    }

    .github-corner .octo-arm {
        animation: none;
    }

    @keyframes a {
        0%,
        to {
            transform: rotate(0);
        }
        20%,
        60% {
            transform: rotate(-25deg);
        }
        40%,
        80% {
            transform: rotate(10deg);
        }
    }
</style>

<a href="https://github.com/blinkfox/hexo-theme-matery" class="github-corner tooltipped hide-on-med-and-down" target="_blank"
   data-tooltip="Fork Me" data-position="left" data-delay="50">
    <svg viewBox="0 0 250 250" aria-hidden="true">
        <path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path>
        <path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
              fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path>
        <path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
              fill="currentColor" class="octo-body"></path>
    </svg>
</a>
        
    </nav>

</header>

    



<div class="bg-cover pd-header post-cover" style="background-image: url('/medias/featureimages/9.jpg')">
    <div class="container" style="right: 0px;left: 0px;">
        <div class="row">
            <div class="col s12 m12 l12">
                <div class="brand">
                    <h1 class="description center-align post-title">Aligning sequences</h1>
                </div>
            </div>
        </div>
    </div>
</div>




<main class="post-container content">

    
    <link rel="stylesheet" href="/libs/tocbot/tocbot.css">
<style>
    #articleContent h1::before,
    #articleContent h2::before,
    #articleContent h3::before,
    #articleContent h4::before,
    #articleContent h5::before,
    #articleContent h6::before {
        display: block;
        content: " ";
        height: 100px;
        margin-top: -100px;
        visibility: hidden;
    }

    #articleContent :focus {
        outline: none;
    }

    .toc-fixed {
        position: fixed;
        top: 64px;
    }

    .toc-widget {
        width: 345px;
        padding-left: 20px;
    }

    .toc-widget .toc-title {
        padding: 35px 0 15px 17px;
        font-size: 1.5rem;
        font-weight: bold;
        line-height: 1.5rem;
    }

    .toc-widget ol {
        padding: 0;
        list-style: none;
    }

    #toc-content {
        padding-bottom: 30px;
        overflow: auto;
    }

    #toc-content ol {
        padding-left: 10px;
    }

    #toc-content ol li {
        padding-left: 10px;
    }

    #toc-content .toc-link:hover {
        color: #42b983;
        font-weight: 700;
        text-decoration: underline;
    }

    #toc-content .toc-link::before {
        background-color: transparent;
        max-height: 25px;

        position: absolute;
        right: 23.5vw;
        display: block;
    }

    #toc-content .is-active-link {
        color: #42b983;
    }

    #floating-toc-btn {
        position: fixed;
        right: 15px;
        bottom: 76px;
        padding-top: 15px;
        margin-bottom: 0;
        z-index: 998;
    }

    #floating-toc-btn .btn-floating {
        width: 48px;
        height: 48px;
    }

    #floating-toc-btn .btn-floating i {
        line-height: 48px;
        font-size: 1.4rem;
    }
</style>
<div class="row">
    <div id="main-content" class="col s12 m12 l9">
        <!-- 文章内容详情 -->
<div id="artDetail">
    <div class="card">
        <div class="card-content article-info">
            <div class="row tag-cate">
                <div class="col s7">
                    
                    <div class="article-tag">
                        
                            <a href="/tags/Algorithms-in-Computational-biology-and-bioinformatics/">
                                <span class="chip bg-color">Algorithms in Computational biology and bioinformatics</span>
                            </a>
                        
                    </div>
                    
                </div>
                <div class="col s5 right-align">
                    
                    <div class="post-cate">
                        <i class="fas fa-bookmark fa-fw icon-category"></i>
                        
                            <a href="/categories/Bioinformatics/" class="post-category">
                                Bioinformatics
                            </a>
                        
                    </div>
                    
                </div>
            </div>

            <div class="post-info">
                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-minus fa-fw"></i>Publish Date:&nbsp;&nbsp;
                    2021-02-20
                </div>
                

                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-check fa-fw"></i>Update Date:&nbsp;&nbsp;
                    2021-08-08
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-file-word fa-fw"></i>Word Count:&nbsp;&nbsp;
                    3.7k
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-clock fa-fw"></i>Read Times:&nbsp;&nbsp;
                    23 Min
                </div>
                

                
                    <div id="busuanzi_container_page_pv" class="info-break-policy">
                        <i class="far fa-eye fa-fw"></i>Read Count:&nbsp;&nbsp;
                        <span id="busuanzi_value_page_pv"></span>
                    </div>
				
            </div>
        </div>
        <hr class="clearfix">

        
        <!-- 是否加载使用自带的 prismjs. -->
        <link rel="stylesheet" href="/libs/prism/prism.css">
        

        

        <div class="card-content article-card-content">
            <div id="articleContent">
                <h1 id="aligning-sequences">Aligning sequences</h1>
<p><a href="https://yintelligence.tech/2021/02/11/2021-02-18-Course-Introduction/">VUB 4018221FNR Algorithms in Computational biology and bioinformatics</a></p>
<p><a target="_blank" rel="noopener" href="https://vub-my.sharepoint.com/personal/wim_vranken_vub_be/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fwim%5Fvranken%5Fvub%5Fbe%2FDocuments%2FOpnamen%2FAlgorithms%20in%20bioinfo%5Fcomputational%20biology%2D20210218%5F100244%2DMeeting%20Recording%2Emp4&amp;parent=%2Fpersonal%2Fwim%5Fvranken%5Fvub%5Fbe%2FDocuments%2FOpnamen&amp;originalPath=aHR0cHM6Ly92dWItbXkuc2hhcmVwb2ludC5jb20vOnY6L2cvcGVyc29uYWwvd2ltX3ZyYW5rZW5fdnViX2JlL0Vkazl3OWRfbXR4UGljMlk1MkJMbnRNQjNZZzZabEdhWlNldnRaNjlTNjE2MEE_cnRpbWU9WDV1eVYwTFkyRWc">Video 7:07 ~ end</a> + <a target="_blank" rel="noopener" href="https://vub-my.sharepoint.com/personal/wim_vranken_vub_be/_layouts/15/onedrive.aspx?originalPath=aHR0cHM6Ly92dWItbXkuc2hhcmVwb2ludC5jb20vOmY6L2cvcGVyc29uYWwvd2ltX3ZyYW5rZW5fdnViX2JlL0VnNzMxZWdaRFhwQ3QwYmZYZS1ESnEwQjhNWVZRbTg4VlQtX2RFeWRyVllVSkE%5FcnRpbWU9MHlNTE41NWEyVWc&amp;id=%2Fpersonal%2Fwim%5Fvranken%5Fvub%5Fbe%2FDocuments%2Fteaching%2FalgorithmsBioinfoVUB%2FINFO%2DF%2D439%2F2021%2FpublicFolder%2F210218%5Faligning%20sequences%2Epdf&amp;parent=%2Fpersonal%2Fwim%5Fvranken%5Fvub%5Fbe%2FDocuments%2Fteaching%2FalgorithmsBioinfoVUB%2FINFO%2DF%2D439%2F2021%2FpublicFolder">Slides</a></p>
<p><a target="_blank" rel="noopener" href="https://www.csie.ntu.edu.tw/~kmchao/seq07fall/alignment.pdf">NTU: A Class Note on Sequence Alignment</a></p>
<p><a target="_blank" rel="noopener" href="https://ocw.mit.edu/courses/electrical-engineering-and-computer-science/6-096-algorithms-for-computational-biology-spring-2005/lecture-notes/lecture5_newest.pdf">MIT: Sequence Alignment and Dynamic Programming</a></p>
<p><a target="_blank" rel="noopener" href="http://sites.fas.harvard.edu/~mcb211/lectures/2004-12-08_Lecture_Notes.pdf">Harvard MCB 211: Sequence Alignment</a></p>
<p>In nature, even a single amino acid sequence contains all the information necessary to determine the fold of the protein. However, the folding process is still mysterious to us, and some valuable information can be revealed by sequence comparison. Take a look at the following sequence: <span class="math display">\[
THETR\ UTHIS\ MOREI\ MPORT\ ANTTH\ ANTHE\ FACTS
\]</span> What did you see in the above sequence? By comparing it with the words in the dictionary, we find the tokens “FACTS,” “IMPORTANT,” “IS,” “MORE,” “THAN,” ”THE,” and “TRUTH.” Then we figure out the above is the sentence “The truth is more important than the facts.”</p>
<h2 id="definitions-3140---3443">Definitions 31:40 - 34:43</h2>
<ul>
<li><p><strong>Identity</strong>: The level of identity is the fraction of residues which are identical between two sequences</p>
<blockquote>
<p>The identity between two randomly generated sequences is never 0%</p>
</blockquote></li>
<li><p><strong>Similarity</strong>: Two residues are similar when their substitution has no effect on the functionality of the protein. The level of similarity is the fraction of residues that are similar.</p>
<p><img src="/images/ADVMB/1914351.png" /></p></li>
<li><p><strong>Homology</strong>: Two sequences are homologous when the similarity between them is derived from a common ancestor</p>
<blockquote>
<p>Homology is a binary relationship</p>
</blockquote></li>
</ul>
<h2 id="why-alignment-2830">Why alignment? 28:30</h2>
<p>To know whether these sequences are <strong>homologs</strong>, a sequence comparison has to be performed.</p>
<p><img src="/images/ADVMB/191435.png" /></p>
<blockquote>
<p>Even though we have not decoded the DNA and protein languages, the emerging flood of sequence data has provided us with a golden opportunity of investigating the evolution and function of biomolecular sequences.</p>
<p>Sequence comparison plays a major role in this line of research, and thus becomes the most basic tool of bioinformatics.</p>
</blockquote>
<h2 id="the-gaps-11804---11924">The gaps 1:18:04 - 1:19:24</h2>
<p>Homologous sequences often have a different length</p>
<p><img src="/images/ADVMB/1984350.png" /></p>
<p>If one wants to align these to sequences in an optimal manner, gaps need to be introduced.</p>
<p>One often makes the difference between the start of a gap （penalty of -10 and -15） or the extension of existing gaps （penalty of -0.5 or -2 ）.</p>
<h3 id="challenge-3002">Challenge 30:02</h3>
<p><strong>Comparing two DNA sequences</strong>: given two possibly related strings S1 and S2, What is the longest common subsequence?</p>
<p><img src="/images/ADVMB/19143500.png" /></p>
<p>Comparison may be difficult due to</p>
<ul>
<li>substitutions</li>
<li>insertions or deletions <code>indels</code></li>
</ul>
<p><img src="/images/ADVMB/1914350.png" /></p>
<p>Goal: Look for the alignment that produces the <strong><u>maximum similarity</u></strong>.</p>
<h3 id="why-the-maximum-similarity-3444">Why the maximum similarity? 34:44</h3>
<p>If we have two genes or proteins that come from the same common ancestor, we want the <strong>amino acids</strong> to align properly.</p>
<figure>
<img src="/images/ADVMB/1914352.png" alt="" /><figcaption>The best way to find this mapping is to look for the maximum similarity between sequences</figcaption>
</figure>
<h2 id="different-kinds-of-alignment-11925">Different kinds of alignment 1:19:25</h2>
<p>There are two big classes of alignment methods</p>
<ol type="1">
<li><strong>Global alignment</strong> : The alignment between the complete sequences
<ul>
<li>Interesting to compare two related sequences</li>
</ul></li>
<li><strong>Local alignment</strong> : aligning of conserved regions in two different proteins
<ul>
<li>To detect for instance identical domains or motives</li>
</ul></li>
</ol>
<h3 id="global-alignment-12021"><a target="_blank" rel="noopener" href="https://www.cs.princeton.edu/~mona/Lecture/align1.pdf">Global alignment</a> 1:20:21</h3>
<p>Dynamic programming to align two sequences by Saul Needleman and Christian Wunsch in 1970.</p>
<p>Dynamic programming has become the method of choice for “rigorous” alignment of DNA and protein sequences. For a number of useful alignment-scoring schemes, this method is guaranteed to produce an alignment of two given sequences having the highest possible score.</p>
<p>Given two sequences A = （a<sub>1</sub>, a<sub>2</sub>, . . . , a<sub>M</sub>）, and B = （b<sub>1</sub>, b<sub>2</sub>, . . . , b<sub>N</sub>）, an <em>alignment</em> of A and B is obtained by introducing dashes into the two sequences such that the lengths of the two resulting sequences are identical and no column contains two dashes. Let Σ denote the input symbol alphabet.</p>
<p>To simplify the presentation, we employ a very simple scoring scheme as follows. A score <code>w(a, b)</code> is defined for each （a, b） ∈ Σ×Σ. Each gap symbol is penalized by a constant g. The score of an alignment is the sum of w scores of all columns with no dashes minus the penalties of the gaps.</p>
<p>Here is an example of an alignment’s score. An optimal alignment is an alignment that maximizes the score. By global alignment we mean that both sequences are aligned globally, i.e., from their first symbols to their last.</p>
<figure>
<img src="/images/ADVMB/19843511.png" alt="" /><figcaption>We assume the following simple scoring scheme: match = 8, mismatch = −5, and g = 3. That is, w（a, b） = 8 if a and b are the same; w（a, b） = −5 if a and b are different; and the gap penalty for each gap symbol is 3.</figcaption>
</figure>
<p>It is quite helpful to recast the problem of aligning two sequences as an equivalent problem of finding a maximum-score path in a certain graph.</p>
<p><img src="/images/ADVMB/19843512.png" /></p>
<p>Let S（i, j）denote the score of an optimal alignment between （a<sub>1</sub>, a<sub>2</sub>, . . . , a<sub>M</sub>）, and （b<sub>1</sub>, b<sub>2</sub>, . . . , b<sub>N</sub>）. With proper initializations, S（i, j）can be computed by the following recurrence: <span class="math display">\[
S(i,j) = max \left\{
\begin{aligned}
S(i-1,j)-g \\
S(i,j-1) -g\\
S(i-1,j-1) + w(a_i,b_j)
\end{aligned}
\right.
\]</span> The figure below explains the recurrence by showing that there are three possible ways entering into the grid point（i, j）, and we take the maximum of their path weights. The value S（M,N） is the score of an optimal alignment between sequences A and B.</p>
<figure>
<img src="/images/ADVMB/19843513.png" alt="" /><figcaption>There are three ways entering the point （i, j）. In this simple scoring scheme, w(ai , −) = w(−, bj ) = −g.</figcaption>
</figure>
<p>Illustrates the tabular computation for S（i, j）:</p>
<figure>
<img src="/images/ADVMB/19843514.png" alt="" /><figcaption>The tabular computation. The maximum alignment score is 14. The shaded area is the trace of backtracking.</figcaption>
</figure>
<p>Gives an optimal alignment that is corresponding to the backtracking:</p>
<figure>
<img src="/images/ADVMB/19843515.png" alt="" /><figcaption>An optimal （global） alignment</figcaption>
</figure>
<p><img src="/images/ADVMB/1984351.png" /></p>
<p><img src="/images/ADVMB/1984352.png" /></p>
<h4 id="the-semiglobal-alignment">The semiglobal alignment</h4>
<p>Make the penalties on S（i, 0） and S（0, j） equal to 0. in this way we remove the obligation that the sequences have to align at the start and the end of the sequences：</p>
<figure>
<img src="/images/ADVMB/1984353.png" alt="" /><figcaption>The backtracking starts at the position in the last row that has the highest score</figcaption>
</figure>
<p>All the previous methods used a linear penalty for the gaps in the alignment <span class="math display">\[
g(n_{gap}) = -n_{gap}E\ \ linear\ gap\ penalty
\]</span> As mentioned earlier, we an differentiate between starting a gap and extending one <span class="math display">\[
g(n_{gap}) = -I-(n_{gap}-1) E\  \ Affine\ gap\ penalty
\]</span> <strong>How does the DP algorithm change?</strong></p>
<p>Do we open or extend a gap?</p>
<p><img src="/images/ADVMB/1984354.png" /></p>
<p>The algorithm can be adapted so that we obtain a solution in <code>O(mn)</code></p>
<p><img src="/images/ADVMB/1984355.png" /></p>
<h4 id="an-example-14225">An example 1:42:25</h4>
<p><img src="/images/ADVMB/1984356.png" /></p>
<p><img src="/images/ADVMB/1984358.png" /></p>
<p><img src="/images/ADVMB/1984359.png" /></p>
<h3 id="local-alignment-14611"><a target="_blank" rel="noopener" href="https://www.cs.princeton.edu/~mona/Lecture/align2.pdf">Local alignment</a> 1:46:11</h3>
<p>Sometimes we only want to find the subsequences that are similar between two sequences.</p>
<p>Smith and Waterman have proposed a method which resembles strongly the previous system.</p>
<p><img src="/images/ADVMB/1984550.png" /></p>
<p>The recurrence relation for the Smith-Waterman algorithm: <span class="math display">\[
S(i,j) = max \left\{
\begin{aligned}
S(i-1,j)+t(i,j) \\
[jS(i-n_{gap1},j)+g(n_{gap1})]\ 1≤n_{gap1}≤i\\
[jS(i,j-n_{gap2})+g(n_{gap2})]\ 1≤n_{gap2}≤j
\end{aligned}
\right.
\]</span> When the score of an alignment becomes negative it is replaced by zero and the values of S（i, 0） and S（0, j） are also 0 as in the semiglobal alignment.</p>
<blockquote>
<p>The backtracking starts in this case a the highest value in the matrix</p>
</blockquote>
<p><img src="/images/ADVMB/1984552.png" /></p>
<p>Sub-optimal alignments provide information concerning other repetitions common between the two sequences （for instance in case of multiple similar domains）.</p>
<p><img src="/images/ADVMB/1984553.png" /></p>
<blockquote>
<p>When the highest scoring subsequence is found, all values in the DP matrix for this sequence are set to zero</p>
<p>The elements in the neighbourhood are recalculated using this new information</p>
<p>Only the elements to the right and below the sequence are affected</p>
<p>When an element does not change, the recalculation is stopped</p>
</blockquote>
<figure>
<img src="/images/ADVMB/1984554.png" alt="" /><figcaption>Example using LALIGN</figcaption>
</figure>
<h2 id="interference-of-mutation-3819">Interference of mutation 38:19</h2>
<p>Homology implies a common ancestor and at the same time it may mean that both sequences have a similar structure or function.</p>
<p><strong>But</strong> this is not always the case: <strong>Mutation</strong> may result in sequences with different functions or some protein sequences may be completely different, but still perform the same function or two sequences <u>may be similar but have <strong>no</strong> common ancestor</u>.</p>
<blockquote>
<p>One always has to be careful and look for additional information to make sure the claim sticks.</p>
</blockquote>
<figure>
<img src="/images/ADVMB/1914353.png" alt="" /><figcaption>Convergent evolution</figcaption>
</figure>
<p>Since the common ancestor is often not known, an alignment is just an <strong>hypothesis</strong> about a relationship between two sequences.</p>
<h3 id="method-a-quantitative-score-4612">Method: A quantitative score 46:12</h3>
<p>A <strong>quantitative method</strong> is required that takes into account the changes accepted by evolution.</p>
<p>The idea is that a quantitative method provides a higher score to the alignment of sequences that are indeed derived from a common ancestor than between sequences that are related by chance.</p>
<p>However, there is currently no quantitative method that uses a complete and exact model of evolution. Therefore, the best score is not necessarily the correct alignment （in terms of a common ancestor）.</p>
<blockquote>
<p>The best score is for the exact alignment = <strong>Optimal alignment</strong></p>
<p>These methods are used to look for similar （MAX） or different （MIN）sequences.</p>
</blockquote>
<h3 id="example-4729">Example 47:29</h3>
<p><img src="/images/ADVMB/1914354.png" /></p>
<h2 id="score-models-4923">Score models 49:23</h2>
<p>There are two models that can explain the similarities and differences between two sequences</p>
<ol type="1">
<li>a <strong>random model</strong></li>
<li>a <strong>non-random model</strong> （<strong>evolutionary</strong>）</li>
</ol>
<p>When the <strong>probability of a match</strong> between the sequences is determined, one can decide which of the two models is more likely.</p>
<h3 id="random-model-5009">Random model 50:09</h3>
<p>In this model there are <strong>no</strong> constraints on the sequence composition. Meaning that the choice of each residue is <strong>independent</strong> from the other residues in the sequence</p>
<p>The probability depends on the <strong>frequency</strong> of every amino acid in the population （= all characters in the sequences）.</p>
<ul>
<li>When the frequency of the amino acids a is p<sub>a</sub>, the probability of having a at that position is also p<sub>a</sub></li>
<li>When two residues a and b are aligned, the <strong>probability</strong> of the match is <strong>p<sub>a</sub> p<sub>b</sub></strong></li>
</ul>
<h3 id="non-random-or-evolutionary-model-5150">Non-random （or evolutionary） model 51:50</h3>
<p>This model proposes that the sequences are related because of <strong>common descent</strong>, this implies that there are <strong>constraints</strong> on the composition of the sequence</p>
<p>The probability of a match depends on the <strong>residue at the same position</strong> in the common ancestor.</p>
<ul>
<li>The probability is q<sub>a,b</sub>, the actual value of q<sub>a,b</sub> depends on the evolutionary mechanism</li>
<li>When two residues a and b are aligned, the <strong>probability of a match</strong> is <strong>q<sub>a,b</sub></strong> since the residues are correlated</li>
</ul>
<h3 id="comparation-5312">Comparation 53:12</h3>
<p>The <strong>odds ratio</strong> q<sub>a,b</sub>/p<sub>a</sub>p<sub>b</sub> shows which model is more likely</p>
<ul>
<li>When q<sub>a,b</sub>/p<sub>a</sub>p<sub>b</sub> &gt;1 the evolutionary model explains better the alignment</li>
</ul>
<p>The model has to explain the complete alignment</p>
<ul>
<li><p>All odds ratio’s need to be combined <span class="math display">\[
\prod_{u}(\frac{q_{a,b}}{p_aP_b})_u
\]</span></p></li>
<li><p>It is often easier to use sums instead of products: <span class="math display">\[
\sum_u\log(\frac{q_{a,b}}{p_aP_b})_u
\]</span></p></li>
</ul>
<blockquote>
<p>u is the position in the alignment</p>
</blockquote>
<p>The <strong>log-odds ratio</strong> is the value that is written in a substitution matrix: <span class="math display">\[
S=\sum_u(S_{a,b})_u
\]</span></p>
<h2 id="multiple-substitutions">Multiple substitutions</h2>
<p>Amino acids, nucleotides or any other evolutionary character are replaced by others at some rate.</p>
<p>For example, imagine an evolutionary sequence with three possible states, A, B and C. If substitution model is time-reversible, there will be three transition rates, A&lt;&gt;B, B&lt;&gt;C and A&lt;&gt;C.</p>
<p>Suppose the rates are 1, 1 and 0 respectively in units of substitution per 100 characters per unit of time. After one unit of time, in a 300 character long sequence originally composed equally of As, Bs and Cs, we expect there to have been one A to B substitution and one B to C substitution. If we are comparing two homologous sequences in living organisms, because one unit of time has passed for both sequences, we would expect two A to B and two B to C substitutions between the present day sequences.</p>
<p>No matter how long we run this process for, there will never be a <em>direct</em> replacement of A by C. There will also never be an A to C substitution under a so-called infinite sites model, where no more than one substitution can occur at a single site.</p>
<p>However since A to B and B to C substitutions are common, under a finite sites model eventually B will be replaced by C at a site where A was previously replaced by B. This indirect replacement of A by C （or equivalently in a time-reversible model, C by A） becomes more likely the longer the time period separating the homologous sequences.</p>
<p>We simulated sequence evolution based on the the above scenario, running the simulation for 10 units of time. From this substitution I observed the following counts for each site pattern:</p>
<table>
<thead>
<tr class="header">
<th style="text-align: right;"></th>
<th style="text-align: right;">A</th>
<th style="text-align: right;">B</th>
<th style="text-align: right;">C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">A</td>
<td style="text-align: right;">91</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: right;">B</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">86</td>
<td style="text-align: right;">9</td>
</tr>
<tr class="odd">
<td style="text-align: right;">C</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">9</td>
<td style="text-align: right;">91</td>
</tr>
</tbody>
</table>
<p>Within this relatively short duration, it doesn’t appear as if any A&lt;&gt;C substitutions have occured. However when we reran the simulation for 100 units of time:</p>
<table>
<thead>
<tr class="header">
<th style="text-align: right;"></th>
<th style="text-align: right;">A</th>
<th style="text-align: right;">B</th>
<th style="text-align: right;">C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">A</td>
<td style="text-align: right;">55</td>
<td style="text-align: right;">35</td>
<td style="text-align: right;">10</td>
</tr>
<tr class="even">
<td style="text-align: right;">B</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">35</td>
</tr>
<tr class="odd">
<td style="text-align: right;">C</td>
<td style="text-align: right;">20</td>
<td style="text-align: right;">36</td>
<td style="text-align: right;">44</td>
</tr>
</tbody>
</table>
<p>As you can see, many “A” characters have been replaced with “C” and <em>vice versa</em>. More generally, under a finite sites model multiple substitutions cause the distribution of site pattern counts to become much flatter beyond simply increasing the proportion of off-diagonal relative to diagonal counts.</p>
<p>The PAM and BLOSUM score matrices account for <strong>multiple substitutions</strong> in radically different ways.</p>
<h3 id="substitution-matrices-10313">Substitution matrices 1:03:13</h3>
<p>Two methods have been used to determine the values of all S<sub>a,b</sub>.</p>
<ol type="1">
<li><p>PAM <code>Modeled Change in Protein Sequences</code>:</p>
<ul>
<li>All PAM matrices were derived from <u>one manually constructed matrix</u> （PAM1）.</li>
<li>71 groups of manually aligned protein sequences （no gaps)）were used and in each group there was an identity of at least 85% between each sequence</li>
</ul>
<p><img src="/images/ADVMB/1914358.png" /></p></li>
<li><p>BLOSUM <code>Experimentally Derived Matrix</code>:</p></li>
</ol>
<ul>
<li>Every BLOSUM matrix was constructed from a <u>clustered set of highly conserved blocks</u> of amino acids</li>
<li>The clustering was also based on identity, meaning that each sequence in each cluster has at least x% identity</li>
<li>In case of BLOSUM62, this is 62% identity</li>
</ul>
<p><img src="/images/ADVMB/1914359.png" /></p>
<h4 id="pam-matrices-10523">PAM matrices 1:05:23</h4>
<p>The name PAM comes from “<u>point accepted mutation</u>”, and refers to the replacement of a single amino acid in a protein with a different amino acid.</p>
<p>These mutations were identified by comparing highly similar sequences with at least 85% identity, and it is assumed that any observed substitutions were the result of a single mutation between the ancestral sequence and one of the present day sequences.</p>
<figure>
<img src="/images/ADVMB/1924350.png" alt="" /><figcaption>The idea was to create a similarity distance measure</figcaption>
</figure>
<p>The basic 1-PAM reflects an amount of evolution producing on an average one mutation per hundred amino acids. Two sequences S1 and S2 are at evolutionary distance of one PAM, if S1 has converted to S2 with an average of one accepted point-mutation event per 100 amino acids.</p>
<ul>
<li><p>PAM1 = unit of evolutionary divergence in which 1% of the amino acids have been changed.</p></li>
<li><p>PAM120 = 120 mutations accepted for 100 amino acids</p>
<blockquote>
<p>The PAM1 matrix multiplied against itself 120 times.</p>
<p>multiple mutations at same position possible!</p>
<p>PAM250 probability matrix, describing the replacement probabilities given 250 PAM units of time, was derived by raising the PAM1 probability matrix to the power 250 （all elements were scaled by 100 for legibility）.</p>
<p>The replacement probabilities derived using this exponentiation correctly account for multiple substitutions. Not only are the off-diagonal probabilities proportionally larger as you would expect for a longer time duration, but they are flatter. For example, the probability of a valine （V） to isoleucine（I） replacement is 33 × larger than a V to histadine （H)）replacement in the PAM1 matrix, but only 4.5× larger in the PAM250 matrix.</p>
<p>Score matrices can then be calculated from the probability matrices and observed base frequencies.</p>
</blockquote></li>
</ul>
<p><img src="/images/ADVMB/1924351.png" /></p>
<p><strong>Question</strong>: when two sequence have an identity of 40%, which matrix should we use ?</p>
<p><img src="/images/ADVMB/1924352.png" /></p>
<h4 id="blosum-matrices">BLOSUM matrices</h4>
<p>The BLOSUM matrices, developed by Steven and Jorja Henikoff and published in 1992, takes a very different approach. Whereas PAM is implicitly applying a stationary finite sites model of evolution using matrix exponentiation, the effect of multiple substitutions is dealt with implicitly in BLOSUM by constructing <strong>different score matrices for different time scales</strong>.</p>
<p>BLOSUM matrices were created by using blocks of conserved amino acids in clusters of sequences nd then applying statistical methods to the data to obtain the similarity score.</p>
<blockquote>
<p>Initially a simple unit matrix is used, where the score is 1 for a match and 0 for a mismatch. Then, from the sets of proteins from public databases that have been grouped into related families, some blocks of sequences （ungapped sequences） that appear to have been conserved are found .</p>
</blockquote>
<p>Thus this is not dependent on any specific scoring scheme as the starting point is the unit matrix.</p>
<p>Let us consider a very simple example: Consider only the alphabet A, B, C and only one block consisting of 6 sequences.</p>
<table>
<thead>
<tr class="header">
<th>Seq1</th>
<th>B</th>
<th>A</th>
<th>B</th>
<th>A</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Seq2</td>
<td>A</td>
<td>A</td>
<td>A</td>
<td>C</td>
</tr>
<tr class="even">
<td>Seq3</td>
<td>A</td>
<td>A</td>
<td>C</td>
<td>C</td>
</tr>
<tr class="odd">
<td>Seq4</td>
<td>A</td>
<td>A</td>
<td>B</td>
<td>A</td>
</tr>
<tr class="even">
<td>Seq5</td>
<td>A</td>
<td>A</td>
<td>C</td>
<td>C</td>
</tr>
<tr class="odd">
<td>Seq6</td>
<td>A</td>
<td>A</td>
<td>B</td>
<td>C</td>
</tr>
</tbody>
</table>
<p>Using this data block, the number of occurrences of each amino acid and also the number of pairs of amino acids aligned in the same column are counted. Here we see that A occurs 14 times, B occurs 4 times and C occurs 6 times out of a total of 24. Counting the total number of aligned pairs (vertically - column wise), there are 60 aligned pairs. The frequency of the pair AA is the greatest and that of BB the leasst.</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Pair</th>
<th style="text-align: center;">Frequency of occurrenc</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">AA</td>
<td style="text-align: center;">26</td>
</tr>
<tr class="even">
<td style="text-align: center;">AB</td>
<td style="text-align: center;">8</td>
</tr>
<tr class="odd">
<td style="text-align: center;">AC</td>
<td style="text-align: center;">10</td>
</tr>
<tr class="even">
<td style="text-align: center;">BB</td>
<td style="text-align: center;">3</td>
</tr>
<tr class="odd">
<td style="text-align: center;">BC</td>
<td style="text-align: center;">6</td>
</tr>
<tr class="even">
<td style="text-align: center;">CC</td>
<td style="text-align: center;">7</td>
</tr>
</tbody>
</table>
<p>Now these observed frequencies are compared to the expected number of times an amino acid pair is aligned if a random collection of amino acids is observed. If two sequences are chosen at random with the probability of occurrence of A as 14/24, B as 4/24 and C as 6/24 and then aligned, then the probability of an A aligning with another A is 14/24 × 14/24. Similarly the probability of a B aligning with an A is 2 × 14/24 × 4/24. （There is a factor of 2 as the alignment could be A from the first sequence and B from the second or it could be B from the first sequence and A from the second.）</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;">Pair</th>
<th style="text-align: center;">Observed （O）</th>
<th style="text-align: center;">Expected 9E）</th>
<th style="text-align: center;">2log<sub>2</sub>（O/E）</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">AA</td>
<td style="text-align: center;">26/60</td>
<td style="text-align: center;">196/576</td>
<td style="text-align: center;">0.7</td>
</tr>
<tr class="even">
<td style="text-align: center;">AB</td>
<td style="text-align: center;">8/60</td>
<td style="text-align: center;">112/576</td>
<td style="text-align: center;">-1.09</td>
</tr>
<tr class="odd">
<td style="text-align: center;">AC</td>
<td style="text-align: center;">10/60</td>
<td style="text-align: center;">168/576</td>
<td style="text-align: center;">-1.61</td>
</tr>
<tr class="even">
<td style="text-align: center;">BB</td>
<td style="text-align: center;">3/60</td>
<td style="text-align: center;">16/576</td>
<td style="text-align: center;">1.70</td>
</tr>
<tr class="odd">
<td style="text-align: center;">BC</td>
<td style="text-align: center;">6/60</td>
<td style="text-align: center;">48/576</td>
<td style="text-align: center;">0.53</td>
</tr>
<tr class="even">
<td style="text-align: center;">CC</td>
<td style="text-align: center;">7/60</td>
<td style="text-align: center;">36/576</td>
<td style="text-align: center;">1.80</td>
</tr>
</tbody>
</table>
<p>The last column gives the ratio of the occurrence each amino acid combination in the observed data to the expected value of occurrence of the pair. This is called the <em>log odd ratio</em>. It is rounded off and used in the substitution matrix. The substitution matrix for this data looks as follows:</p>
<table>
<thead>
<tr class="header">
<th style="text-align: center;"></th>
<th style="text-align: center;">A</th>
<th style="text-align: center;">B</th>
<th style="text-align: left;">C</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">A</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">-1</td>
<td style="text-align: left;">2</td>
</tr>
<tr class="even">
<td style="text-align: center;">B</td>
<td style="text-align: center;">-1</td>
<td style="text-align: center;">2</td>
<td style="text-align: left;">1</td>
</tr>
<tr class="odd">
<td style="text-align: center;">C</td>
<td style="text-align: center;">-2</td>
<td style="text-align: center;">1</td>
<td style="text-align: left;">2</td>
</tr>
</tbody>
</table>
<p>This method gives a positive score to those pairs that are more likely than chance to occur and negative scores to those pairs that are less likely.</p>
<figure>
<img src="/images/ADVMB/1924353.png" alt="" /><figcaption>The 62 is the percentage of identity between the sequences used that were used to determine the substitutions valu es</figcaption>
</figure>
<p>One of the shortcomings of this method is that it overlooks an important factor that could bias the results. If there are many closely related proteins in one block and a few others that are less closely related, then the substitution matrix will be biased toward the closely related proteins.</p>
<p>To reduce this bias, grouping of closely related proteins in one block is done and their contributions are normalized.</p>
<p>In the first stage of building the BLOSUM matrix, while choosing the blocks, amino acid sequences which are more than x% identical are grouped or clustered together. Thus the contributions of multiple entries of closely related sequences is reduced, as their contribution is weighted to sum to one. The matrix built from blocks with no more than x% of similarity is called BLOSUM X, for example if the 0.85 similarity criterion is adopted, the final matrix is called BLOSUM 85.</p>
<p>A further refinement to this is that after obtaining the substitution matrix, it is used instead of the conservative “unit” matrix to construct the set of blocks. This process can be repeated again and it is from this third set of blocks that the final BLOSUM matrices are defined.</p>
<h5 id="analysis-of-blosum62-11222">Analysis of BLOSUM62 1:12:22</h5>
<figure>
<img src="/images/ADVMB/1924354.png" alt="" /><figcaption>The substitutions within families of amino acids are often favourable</figcaption>
</figure>
<p><img src="/images/ADVMB/1924355.png" /></p>
<p><img src="/images/ADVMB/1924356.png" /></p>
<p><img src="/images/ADVMB/1924357.png" /></p>
<h4 id="comparation">Comparation</h4>
<p>Although they take different routes, the final BLOSUM and PAM score matrices are actually pretty similar. According to Henikoff and Henikoff, the following PAM and BLOSUM matrices are comparible:</p>
<table>
<thead>
<tr class="header">
<th style="text-align: right;">PAM</th>
<th style="text-align: right;">BLOSUM</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: right;">PAM250</td>
<td style="text-align: right;">BLOSUM45</td>
</tr>
<tr class="even">
<td style="text-align: right;">PAM160</td>
<td style="text-align: right;">BLOSUM62</td>
</tr>
<tr class="odd">
<td style="text-align: right;">PAM120</td>
<td style="text-align: right;">BLOSUM80</td>
</tr>
</tbody>
</table>
<p>The main difference between the PAM and BLOSUM matrices is that PAM is derived from global alignments of proteins, while BLOSUM comes from alignments of shorter sequences - blocks of sequences that match each other at some defined level of similarity. The BLOSUM method incorporates much more data into its matrices and is hence supposed to be more reliable.</p>
<p>Which matrix should be used ????</p>
<ul>
<li>The PAM number underlines the evolutionary divergence between sequences</li>
<li>The BLOSUM number represents the percentage if identity between the sequences in the same group</li>
</ul>
<p>NCBI recommendations for choosing BLOSUM or PAM are:</p>
<ul>
<li>Default matrix to use is BLOSUM62 or PAM120</li>
<li>For long and distantly related proteins: BLOSUM45</li>
<li>For closely related sequences: Don’t care.</li>
<li>For short sequences: PAM30 （length &lt; 35）, PAM70 （length &gt; 50）</li>
</ul>
<p><img src="/images/ADVMB/1924358.png" /></p>
<p>To evaluate similar sequences one needs matrices that give high values on identical matches : e.g. BLOSUM 80 or PAM50</p>
<blockquote>
<p>The matrix choice is biased by the percentage of conservation the biologist expects to find between the sequences .</p>
</blockquote>
<h4 id="which-pam-or-blosum-11728">Which PAM or BLOSUM? 1:17:28</h4>
<p>Using relative entropy <code>H</code> one can verify whether the matrix is still capable of making the distinction between the evolutionary and the random model</p>
<p><img src="/images/ADVMB/1924359.png" /></p>
<h2 id="summary">Summary</h2>
<h3 id="substitution-matrices">Substitution matrices</h3>
<ul>
<li>PAM matrices were created from fully aligned protein sequence</li>
<li>BLOSUM matrices from blocks of conserved amino acids in clusters of sequences.</li>
</ul>
<h3 id="global-alignment">Global alignment</h3>
<p><img src="/images/CS61B/19243580.png" /></p>
<h4 id="semi-global-alignment">Semi-global alignment</h4>
<p>Make the penalties on S（i,0） and S（0,j） equal to 0. in this way we remove the obligation that the sequences have to align at the start and the end of the sequences.</p>
<h3 id="local-alignment">Local alignment</h3>
<p><img src="/images/CS61B/19243581.png" /></p>

                
            </div>
            <hr/>

            

    <div class="reprint" id="reprint-statement">
        
            <div class="reprint__author">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-user">
                        Author:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="/about" rel="external nofollow noreferrer">Chaoqun Yin</a>
                </span>
            </div>
            <div class="reprint__type">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-link">
                        Link:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="https://yintelligence.tech/2021/02/20/2021-02-20-Aligning-sequences/">https://yintelligence.tech/2021/02/20/2021-02-20-Aligning-sequences/</a>
                </span>
            </div>
            <div class="reprint__notice">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-copyright">
                        Reprint policy:
                    </i>
                </span>
                <span class="reprint-info">
                    All articles in this blog are used except for special statements
                    <a href="https://creativecommons.org/licenses/by/4.0/deed.zh" rel="external nofollow noreferrer" target="_blank">CC BY 4.0</a>
                    reprint polocy. If reproduced, please indicate source
                    <a href="/about" target="_blank">Chaoqun Yin</a>
                    !
                </span>
            </div>
        
    </div>

    <script async defer>
      document.addEventListener("copy", function (e) {
        let toastHTML = '<span>Copied successfully, please follow the reprint policy of this article</span><button class="btn-flat toast-action" onclick="navToReprintStatement()" style="font-size: smaller">more</a>';
        M.toast({html: toastHTML})
      });

      function navToReprintStatement() {
        $("html, body").animate({scrollTop: $("#reprint-statement").offset().top - 80}, 800);
      }
    </script>



            <div class="tag_share" style="display: block;">
                <div class="post-meta__tag-list" style="display: inline-block;">
                    
                        <div class="article-tag">
                            
                                <a href="/tags/Algorithms-in-Computational-biology-and-bioinformatics/">
                                    <span class="chip bg-color">Algorithms in Computational biology and bioinformatics</span>
                                </a>
                            
                        </div>
                    
                </div>
                <div class="post_share" style="zoom: 80%; width: fit-content; display: inline-block; float: right; margin: -0.15rem 0;">
                    <link rel="stylesheet" type="text/css" href="/libs/share/css/share.min.css">
<div id="article-share">

    
    <div class="social-share" data-sites="twitter,facebook,google,qq,qzone,wechat,weibo,douban,linkedin" data-wechat-qrcode-helper="<p>微信扫一扫即可分享！</p>"></div>
    <script src="/libs/share/js/social-share.min.js"></script>
    

    

</div>

                </div>
            </div>
            
        </div>
    </div>

    

    

    

    

    

    

    

    

<article id="prenext-posts" class="prev-next articles">
    <div class="row article-row">
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge left-badge text-color">
                <i class="fas fa-chevron-left"></i>&nbsp;Previous</div>
            <div class="card">
                <a href="/2021/02/20/2021-02-20-Intelligent-Agents/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/1.jpg" class="responsive-img" alt="Intelligent Agents">
                        
                        <span class="card-title">Intelligent Agents</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            Introducing AI Agents and their properties
                        
                    </div>
                    <div class="publish-info">
                        <span class="publish-date">
                            <i class="far fa-clock fa-fw icon-date"></i>2021-02-20
                        </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/Artificial-Intelligence/" class="post-category">
                                    Artificial Intelligence
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Techniques-of-Artificial-Intelligence/">
                        <span class="chip bg-color">Techniques of Artificial Intelligence</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge right-badge text-color">
                Next&nbsp;<i class="fas fa-chevron-right"></i>
            </div>
            <div class="card">
                <a href="/2021/02/20/2021-02-20-Proof-by-Resolution/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/17.jpg" class="responsive-img" alt="Proof by Resolution">
                        
                        <span class="card-title">Proof by Resolution</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            Learn about general-purpose theorem proving in predicate calculus
                        
                    </div>
                    <div class="publish-info">
                            <span class="publish-date">
                                <i class="far fa-clock fa-fw icon-date"></i>2021-02-20
                            </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/Artificial-Intelligence/" class="post-category">
                                    Artificial Intelligence
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Techniques-of-Artificial-Intelligence/">
                        <span class="chip bg-color">Techniques of Artificial Intelligence</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
    </div>
</article>

</div>


<script>
    $('#articleContent').on('copy', function (e) {
        // IE8 or earlier browser is 'undefined'
        if (typeof window.getSelection === 'undefined') return;

        var selection = window.getSelection();
        // if the selection is short let's not annoy our users.
        if (('' + selection).length < Number.parseInt('120')) {
            return;
        }

        // create a div outside of the visible area and fill it with the selected text.
        var bodyElement = document.getElementsByTagName('body')[0];
        var newdiv = document.createElement('div');
        newdiv.style.position = 'absolute';
        newdiv.style.left = '-99999px';
        bodyElement.appendChild(newdiv);
        newdiv.appendChild(selection.getRangeAt(0).cloneContents());

        // we need a <pre> tag workaround.
        // otherwise the text inside "pre" loses all the line breaks!
        if (selection.getRangeAt(0).commonAncestorContainer.nodeName === 'PRE') {
            newdiv.innerHTML = "<pre>" + newdiv.innerHTML + "</pre>";
        }

        var url = document.location.href;
        newdiv.innerHTML += '<br />'
            + 'From: Chaoqun Yin<br />'
            + 'Author: Chaoqun Yin<br />'
            + 'Link: <a href="' + url + '">' + url + '</a><br />'
            + '本文章著作权归作者所有，任何形式的转载都请注明出处。';

        selection.selectAllChildren(newdiv);
        window.setTimeout(function () {bodyElement.removeChild(newdiv);}, 200);
    });
</script>


<!-- 代码块功能依赖 -->
<script type="text/javascript" src="/libs/codeBlock/codeBlockFuction.js"></script>

<!-- 代码语言 -->

<script type="text/javascript" src="/libs/codeBlock/codeLang.js"></script>


<!-- 代码块复制 -->

<script type="text/javascript" src="/libs/codeBlock/codeCopy.js"></script>


<!-- 代码块收缩 -->

<script type="text/javascript" src="/libs/codeBlock/codeShrink.js"></script>


    </div>
    <div id="toc-aside" class="expanded col l3 hide-on-med-and-down">
        <div class="toc-widget card" style="background-color: white;">
            <div class="toc-title"><i class="far fa-list-alt"></i>&nbsp;&nbsp;TOC</div>
            <div id="toc-content"></div>
        </div>
    </div>
</div>

<!-- TOC 悬浮按钮. -->

<div id="floating-toc-btn" class="hide-on-med-and-down">
    <a class="btn-floating btn-large bg-color">
        <i class="fas fa-list-ul"></i>
    </a>
</div>


<script src="/libs/tocbot/tocbot.min.js"></script>
<script>
    $(function () {
        tocbot.init({
            tocSelector: '#toc-content',
            contentSelector: '#articleContent',
            headingsOffset: -($(window).height() * 0.4 - 45),
            collapseDepth: Number('0'),
            headingSelector: 'h1, h2, h3, h4'
        });

        // modify the toc link href to support Chinese.
        let i = 0;
        let tocHeading = 'toc-heading-';
        $('#toc-content a').each(function () {
            $(this).attr('href', '#' + tocHeading + (++i));
        });

        // modify the heading title id to support Chinese.
        i = 0;
        $('#articleContent').children('h1, h2, h3, h4').each(function () {
            $(this).attr('id', tocHeading + (++i));
        });

        // Set scroll toc fixed.
        let tocHeight = parseInt($(window).height() * 0.4 - 64);
        let $tocWidget = $('.toc-widget');
        $(window).scroll(function () {
            let scroll = $(window).scrollTop();
            /* add post toc fixed. */
            if (scroll > tocHeight) {
                $tocWidget.addClass('toc-fixed');
            } else {
                $tocWidget.removeClass('toc-fixed');
            }
        });

        
        /* 修复文章卡片 div 的宽度. */
        let fixPostCardWidth = function (srcId, targetId) {
            let srcDiv = $('#' + srcId);
            if (srcDiv.length === 0) {
                return;
            }

            let w = srcDiv.width();
            if (w >= 450) {
                w = w + 21;
            } else if (w >= 350 && w < 450) {
                w = w + 18;
            } else if (w >= 300 && w < 350) {
                w = w + 16;
            } else {
                w = w + 14;
            }
            $('#' + targetId).width(w);
        };

        // 切换TOC目录展开收缩的相关操作.
        const expandedClass = 'expanded';
        let $tocAside = $('#toc-aside');
        let $mainContent = $('#main-content');
        $('#floating-toc-btn .btn-floating').click(function () {
            if ($tocAside.hasClass(expandedClass)) {
                $tocAside.removeClass(expandedClass).hide();
                $mainContent.removeClass('l9');
            } else {
                $tocAside.addClass(expandedClass).show();
                $mainContent.addClass('l9');
            }
            fixPostCardWidth('artDetail', 'prenext-posts');
        });
        
    });
</script>

    

</main>


<script src="https://cdn.bootcss.com/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<script>
    MathJax.Hub.Config({
        tex2jax: {inlineMath: [['$', '$'], ['\(', '\)']]}
    });
</script>



    <footer class="page-footer bg-color">
    
    <div class="container row center-align" style="margin-bottom: 15px !important;">
        <div class="col s12 m8 l8 copy-right">
            Copyright&nbsp;&copy;
            
                <span id="year">2021</span>
            
            <span id="year">2021</span>
            <a href="/about" target="_blank">Chaoqun Yin</a>
            |&nbsp;Powered by&nbsp;<a href="https://hexo.io/" target="_blank">Hexo</a>
            |&nbsp;Theme&nbsp;<a href="https://github.com/blinkfox/hexo-theme-matery" target="_blank">Matery</a>
            <br>
            
            
            
            
            
            
            <span id="busuanzi_container_site_pv">
                |&nbsp;<i class="far fa-eye"></i>&nbsp;总访问量:&nbsp;<span id="busuanzi_value_site_pv"
                    class="white-color"></span>&nbsp;次
            </span>
            
            
            <span id="busuanzi_container_site_uv">
                |&nbsp;<i class="fas fa-users"></i>&nbsp;总访问人数:&nbsp;<span id="busuanzi_value_site_uv"
                    class="white-color"></span>&nbsp;人
            </span>
            
            <br>
            
            <span id="sitetime">载入运行时间...</span>
            <script>
                function siteTime() {
                    var seconds = 1000;
                    var minutes = seconds * 60;
                    var hours = minutes * 60;
                    var days = hours * 24;
                    var years = days * 365;
                    var today = new Date();
                    var startYear = "2021";
                    var startMonth = "1";
                    var startDate = "28";
                    var startHour = "2";
                    var startMinute = "1";
                    var startSecond = "0";
                    var todayYear = today.getFullYear();
                    var todayMonth = today.getMonth() + 1;
                    var todayDate = today.getDate();
                    var todayHour = today.getHours();
                    var todayMinute = today.getMinutes();
                    var todaySecond = today.getSeconds();
                    var t1 = Date.UTC(startYear, startMonth, startDate, startHour, startMinute, startSecond);
                    var t2 = Date.UTC(todayYear, todayMonth, todayDate, todayHour, todayMinute, todaySecond);
                    var diff = t2 - t1;
                    var diffYears = Math.floor(diff / years);
                    var diffDays = Math.floor((diff / days) - diffYears * 365);
                    var diffHours = Math.floor((diff - (diffYears * 365 + diffDays) * days) / hours);
                    var diffMinutes = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours) /
                        minutes);
                    var diffSeconds = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours -
                        diffMinutes * minutes) / seconds);
                    if (startYear == todayYear) {
                        document.getElementById("year").innerHTML = todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已安全运行 " + diffDays + " 天 " + diffHours +
                            " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    } else {
                        document.getElementById("year").innerHTML = startYear + " - " + todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已安全运行 " + diffYears + " 年 " + diffDays +
                            " 天 " + diffHours + " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    }
                }
                setInterval(siteTime, 1000);
            </script>
            
            <br>
            
        </div>
        <div class="col s12 m4 l4 social-link social-statis">
    <a href="https://github.com/yintelligence" class="tooltipped" target="_blank" data-tooltip="访问我的GitHub" data-position="top" data-delay="50">
        <i class="fab fa-github"></i>
    </a>



    <a href="mailto:Yintelligence@outlook.com" class="tooltipped" target="_blank" data-tooltip="邮件联系我" data-position="top" data-delay="50">
        <i class="fas fa-envelope-open"></i>
    </a>













</div>
    </div>
</footer>

<div class="progress-bar"></div>


    <!-- 搜索遮罩框 -->
<div id="searchModal" class="modal">
    <div class="modal-content">
        <div class="search-header">
            <span class="title"><i class="fas fa-search"></i>&nbsp;&nbsp;Search</span>
            <input type="search" id="searchInput" name="s" placeholder="Please enter a search keyword"
                   class="search-input">
        </div>
        <div id="searchResult"></div>
    </div>
</div>

<script type="text/javascript">
$(function () {
    var searchFunc = function (path, search_id, content_id) {
        'use strict';
        $.ajax({
            url: path,
            dataType: "xml",
            success: function (xmlResponse) {
                // get the contents from search data
                var datas = $("entry", xmlResponse).map(function () {
                    return {
                        title: $("title", this).text(),
                        content: $("content", this).text(),
                        url: $("url", this).text()
                    };
                }).get();
                var $input = document.getElementById(search_id);
                var $resultContent = document.getElementById(content_id);
                $input.addEventListener('input', function () {
                    var str = '<ul class=\"search-result-list\">';
                    var keywords = this.value.trim().toLowerCase().split(/[\s\-]+/);
                    $resultContent.innerHTML = "";
                    if (this.value.trim().length <= 0) {
                        return;
                    }
                    // perform local searching
                    datas.forEach(function (data) {
                        var isMatch = true;
                        var data_title = data.title.trim().toLowerCase();
                        var data_content = data.content.trim().replace(/<[^>]+>/g, "").toLowerCase();
                        var data_url = data.url;
                        data_url = data_url.indexOf('/') === 0 ? data.url : '/' + data_url;
                        var index_title = -1;
                        var index_content = -1;
                        var first_occur = -1;
                        // only match artiles with not empty titles and contents
                        if (data_title !== '' && data_content !== '') {
                            keywords.forEach(function (keyword, i) {
                                index_title = data_title.indexOf(keyword);
                                index_content = data_content.indexOf(keyword);
                                if (index_title < 0 && index_content < 0) {
                                    isMatch = false;
                                } else {
                                    if (index_content < 0) {
                                        index_content = 0;
                                    }
                                    if (i === 0) {
                                        first_occur = index_content;
                                    }
                                }
                            });
                        }
                        // show search results
                        if (isMatch) {
                            str += "<li><a href='" + data_url + "' class='search-result-title'>" + data_title + "</a>";
                            var content = data.content.trim().replace(/<[^>]+>/g, "");
                            if (first_occur >= 0) {
                                // cut out 100 characters
                                var start = first_occur - 20;
                                var end = first_occur + 80;
                                if (start < 0) {
                                    start = 0;
                                }
                                if (start === 0) {
                                    end = 100;
                                }
                                if (end > content.length) {
                                    end = content.length;
                                }
                                var match_content = content.substr(start, end);
                                // highlight all keywords
                                keywords.forEach(function (keyword) {
                                    var regS = new RegExp(keyword, "gi");
                                    match_content = match_content.replace(regS, "<em class=\"search-keyword\">" + keyword + "</em>");
                                });

                                str += "<p class=\"search-result\">" + match_content + "...</p>"
                            }
                            str += "</li>";
                        }
                    });
                    str += "</ul>";
                    $resultContent.innerHTML = str;
                });
            }
        });
    };

    searchFunc('/search.xml', 'searchInput', 'searchResult');
});
</script>

    <!-- 回到顶部按钮 -->
<div id="backTop" class="top-scroll">
    <a class="btn-floating btn-large waves-effect waves-light" href="#!">
        <i class="fas fa-arrow-up"></i>
    </a>
</div>


    <script src="/libs/materialize/materialize.min.js"></script>
    <script src="/libs/masonry/masonry.pkgd.min.js"></script>
    <script src="/libs/aos/aos.js"></script>
    <script src="/libs/scrollprogress/scrollProgress.min.js"></script>
    <script src="/libs/lightGallery/js/lightgallery-all.min.js"></script>
    <script src="/js/matery.js"></script>

    <!-- Baidu Analytics -->

    <!-- Baidu Push -->

<script>
    (function () {
        var bp = document.createElement('script');
        var curProtocol = window.location.protocol.split(':')[0];
        if (curProtocol === 'https') {
            bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
        } else {
            bp.src = 'http://push.zhanzhang.baidu.com/push.js';
        }
        var s = document.getElementsByTagName("script")[0];
        s.parentNode.insertBefore(bp, s);
    })();
</script>

    
    <script src="/libs/others/clicklove.js" async="async"></script>
    
    
    <script async src="/libs/others/busuanzi.pure.mini.js"></script>
    

    

    

	
    

    

    

    
    <script src="/libs/instantpage/instantpage.js" type="module"></script>
    

</body>

</html>
