<!DOCTYPE HTML>
<html lang="en">


<head>
    <meta charset="utf-8">
    <meta name="keywords" content="MSA, profiles and motif detection, Chaoqun Yin">
    <meta name="description" content="Veni, Vidi, Vici">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="renderer" content="webkit|ie-stand|ie-comp">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="format-detection" content="telephone=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <!-- Global site tag (gtag.js) - Google Analytics -->


    <title>MSA, profiles and motif detection | Chaoqun Yin</title>
    <link rel="icon" type="image/png" href="/favicon.png">

    <link rel="stylesheet" type="text/css" href="/libs/awesome/css/all.css">
    <link rel="stylesheet" type="text/css" href="/libs/materialize/materialize.min.css">
    <link rel="stylesheet" type="text/css" href="/libs/aos/aos.css">
    <link rel="stylesheet" type="text/css" href="/libs/animate/animate.min.css">
    <link rel="stylesheet" type="text/css" href="/libs/lightGallery/css/lightgallery.min.css">
    <link rel="stylesheet" type="text/css" href="/css/matery.css">
    <link rel="stylesheet" type="text/css" href="/css/my.css">

    <script src="/libs/jquery/jquery.min.js"></script>

<meta name="generator" content="Hexo 5.3.0"></head>




<body>
    <header class="navbar-fixed">
    <nav id="headNav" class="bg-color nav-transparent">
        <div id="navContainer" class="nav-wrapper container">
            <div class="brand-logo">
                <a href="/" class="waves-effect waves-light">
                    
                    <img src="/medias/logo.png" class="logo-img" alt="LOGO">
                    
                    <span class="logo-span">Chaoqun Yin</span>
                </a>
            </div>
            

<a href="#" data-target="mobile-nav" class="sidenav-trigger button-collapse"><i class="fas fa-bars"></i></a>
<ul class="right nav-menu">
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/" class="waves-effect waves-light">
      
      <i class="fas fa-home" style="zoom: 0.6;"></i>
      
      <span>Index</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/tags" class="waves-effect waves-light">
      
      <i class="fas fa-tags" style="zoom: 0.6;"></i>
      
      <span>Tags</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/categories" class="waves-effect waves-light">
      
      <i class="fas fa-bookmark" style="zoom: 0.6;"></i>
      
      <span>Categories</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/archives" class="waves-effect waves-light">
      
      <i class="fas fa-archive" style="zoom: 0.6;"></i>
      
      <span>Archives</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/about" class="waves-effect waves-light">
      
      <i class="fas fa-user-circle" style="zoom: 0.6;"></i>
      
      <span>About</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/contact" class="waves-effect waves-light">
      
      <i class="fas fa-comments" style="zoom: 0.6;"></i>
      
      <span>Contact</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/friends" class="waves-effect waves-light">
      
      <i class="fas fa-address-book" style="zoom: 0.6;"></i>
      
      <span>Friends</span>
    </a>
    
  </li>
  
  <li>
    <a href="#searchModal" class="modal-trigger waves-effect waves-light">
      <i id="searchIcon" class="fas fa-search" title="Search" style="zoom: 0.85;"></i>
    </a>
  </li>
</ul>


<div id="mobile-nav" class="side-nav sidenav">

    <div class="mobile-head bg-color">
        
        <img src="/medias/logo.png" class="logo-img circle responsive-img">
        
        <div class="logo-name">Chaoqun Yin</div>
        <div class="logo-desc">
            
            Veni, Vidi, Vici
            
        </div>
    </div>

    

    <ul class="menu-list mobile-menu-list">
        
        <li class="m-nav-item">
	  
		<a href="/" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-home"></i>
			
			Index
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/tags" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-tags"></i>
			
			Tags
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/categories" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-bookmark"></i>
			
			Categories
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/archives" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-archive"></i>
			
			Archives
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/about" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-user-circle"></i>
			
			About
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/contact" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-comments"></i>
			
			Contact
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/friends" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-address-book"></i>
			
			Friends
		</a>
          
        </li>
        
        
        <li><div class="divider"></div></li>
        <li>
            <a href="https://github.com/blinkfox/hexo-theme-matery" class="waves-effect waves-light" target="_blank">
                <i class="fab fa-github-square fa-fw"></i>Fork Me
            </a>
        </li>
        
    </ul>
</div>


        </div>

        
            <style>
    .nav-transparent .github-corner {
        display: none !important;
    }

    .github-corner {
        position: absolute;
        z-index: 10;
        top: 0;
        right: 0;
        border: 0;
        transform: scale(1.1);
    }

    .github-corner svg {
        color: #0f9d58;
        fill: #fff;
        height: 64px;
        width: 64px;
    }

    .github-corner:hover .octo-arm {
        animation: a 0.56s ease-in-out;
    }

    .github-corner .octo-arm {
        animation: none;
    }

    @keyframes a {
        0%,
        to {
            transform: rotate(0);
        }
        20%,
        60% {
            transform: rotate(-25deg);
        }
        40%,
        80% {
            transform: rotate(10deg);
        }
    }
</style>

<a href="https://github.com/blinkfox/hexo-theme-matery" class="github-corner tooltipped hide-on-med-and-down" target="_blank"
   data-tooltip="Fork Me" data-position="left" data-delay="50">
    <svg viewBox="0 0 250 250" aria-hidden="true">
        <path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path>
        <path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
              fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path>
        <path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
              fill="currentColor" class="octo-body"></path>
    </svg>
</a>
        
    </nav>

</header>

    



<div class="bg-cover pd-header post-cover" style="background-image: url('/medias/featureimages/20.jpg')">
    <div class="container" style="right: 0px;left: 0px;">
        <div class="row">
            <div class="col s12 m12 l12">
                <div class="brand">
                    <h1 class="description center-align post-title">MSA, profiles and motif detection</h1>
                </div>
            </div>
        </div>
    </div>
</div>




<main class="post-container content">

    
    <link rel="stylesheet" href="/libs/tocbot/tocbot.css">
<style>
    #articleContent h1::before,
    #articleContent h2::before,
    #articleContent h3::before,
    #articleContent h4::before,
    #articleContent h5::before,
    #articleContent h6::before {
        display: block;
        content: " ";
        height: 100px;
        margin-top: -100px;
        visibility: hidden;
    }

    #articleContent :focus {
        outline: none;
    }

    .toc-fixed {
        position: fixed;
        top: 64px;
    }

    .toc-widget {
        width: 345px;
        padding-left: 20px;
    }

    .toc-widget .toc-title {
        padding: 35px 0 15px 17px;
        font-size: 1.5rem;
        font-weight: bold;
        line-height: 1.5rem;
    }

    .toc-widget ol {
        padding: 0;
        list-style: none;
    }

    #toc-content {
        padding-bottom: 30px;
        overflow: auto;
    }

    #toc-content ol {
        padding-left: 10px;
    }

    #toc-content ol li {
        padding-left: 10px;
    }

    #toc-content .toc-link:hover {
        color: #42b983;
        font-weight: 700;
        text-decoration: underline;
    }

    #toc-content .toc-link::before {
        background-color: transparent;
        max-height: 25px;

        position: absolute;
        right: 23.5vw;
        display: block;
    }

    #toc-content .is-active-link {
        color: #42b983;
    }

    #floating-toc-btn {
        position: fixed;
        right: 15px;
        bottom: 76px;
        padding-top: 15px;
        margin-bottom: 0;
        z-index: 998;
    }

    #floating-toc-btn .btn-floating {
        width: 48px;
        height: 48px;
    }

    #floating-toc-btn .btn-floating i {
        line-height: 48px;
        font-size: 1.4rem;
    }
</style>
<div class="row">
    <div id="main-content" class="col s12 m12 l9">
        <!-- 文章内容详情 -->
<div id="artDetail">
    <div class="card">
        <div class="card-content article-info">
            <div class="row tag-cate">
                <div class="col s7">
                    
                    <div class="article-tag">
                        
                            <a href="/tags/Algorithms-in-Computational-biology-and-bioinformatics/">
                                <span class="chip bg-color">Algorithms in Computational biology and bioinformatics</span>
                            </a>
                        
                    </div>
                    
                </div>
                <div class="col s5 right-align">
                    
                    <div class="post-cate">
                        <i class="fas fa-bookmark fa-fw icon-category"></i>
                        
                            <a href="/categories/Bioinformatics/" class="post-category">
                                Bioinformatics
                            </a>
                        
                    </div>
                    
                </div>
            </div>

            <div class="post-info">
                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-minus fa-fw"></i>Publish Date:&nbsp;&nbsp;
                    2021-02-25
                </div>
                

                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-check fa-fw"></i>Update Date:&nbsp;&nbsp;
                    2021-04-07
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-file-word fa-fw"></i>Word Count:&nbsp;&nbsp;
                    1.8k
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-clock fa-fw"></i>Read Times:&nbsp;&nbsp;
                    11 Min
                </div>
                

                
                    <div id="busuanzi_container_page_pv" class="info-break-policy">
                        <i class="far fa-eye fa-fw"></i>Read Count:&nbsp;&nbsp;
                        <span id="busuanzi_value_page_pv"></span>
                    </div>
				
            </div>
        </div>
        <hr class="clearfix">

        
        <!-- 是否加载使用自带的 prismjs. -->
        <link rel="stylesheet" href="/libs/prism/prism.css">
        

        

        <div class="card-content article-card-content">
            <div id="articleContent">
                <h1 id="msa-profiles-and-motif-detection">MSA, profiles and motif detection</h1>
<p><a href="https://yintelligence.tech/2021/02/11/2021-02-18-Course-Introduction/">VUB 4018221FNR Algorithms in Computational biology and bioinformatics</a></p>
<p><a target="_blank" rel="noopener" href="https://vub-my.sharepoint.com/personal/wim_vranken_vub_be/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fwim%5Fvranken%5Fvub%5Fbe%2FDocuments%2FOpnamen%2FAlgorithms%20in%20bioinfo%5Fcomputational%20biology%2D20210225%5F100412%2DMeeting%20Recording%2Emp4&amp;parent=%2Fpersonal%2Fwim%5Fvranken%5Fvub%5Fbe%2FDocuments%2FOpnamen&amp;originalPath=aHR0cHM6Ly92dWItbXkuc2hhcmVwb2ludC5jb20vOnY6L2cvcGVyc29uYWwvd2ltX3ZyYW5rZW5fdnViX2JlL0VmNHlJT19TaE1GTnJGdDB1VlMxSnNvQlJlOV9heXRYanVKeGZqSGtJQ2UxSHc_cnRpbWU9TnpDSFk1cloyRWc">Video</a> + <a target="_blank" rel="noopener" href="https://owncloud.ulb.ac.be/index.php/s/75k0qKi3NFqam0E#pdfviewer">Slides</a></p>
<p><a href="https://yintelligence.tech/2021/03/13/2021-03-13-Aligning-sequences-and-detecting-motifs/">Related assignment</a></p>
<h2 id="multiple-sequence-alignment-2542">Multiple sequence alignment 25:42</h2>
<p>An alignment of 2 sequences is an hypothesis that is supported or rejected by the score one obtains for the alignment, but…</p>
<ul>
<li>the score does not guarantee that there really is a common ancestor</li>
<li>no guarantee that the elements are correctly aligned, only that the alignment is the one with maximum similarity</li>
</ul>
<p>We can <u>reduce this uncertainty</u> by aligning additional sequences, a <strong>multiple sequence alignment （MSA）</strong> algorithm performs this task and can more clearly show:</p>
<ul>
<li>The similarity between positions</li>
<li>The <strong>conservation</strong> of certain amino acids within a family of homologous proteins</li>
</ul>
<p>The alignment of multiple sequences is a method of choice to detect <strong>conserved regions</strong> in protein or DNA sequences. These particular regions are usually associated with:</p>
<ul>
<li>Signals （promoters, signatures for phosphorylation, cellular location, ...）;</li>
<li>Structure （correct folding, protein-protein interactions...）;</li>
<li>Chemical reactivity （catalytic sites,... ）.</li>
</ul>
<p>Different methods exist to build models of these conserved regions:</p>
<ul>
<li>Consensus sequences;</li>
<li>Patterns;</li>
<li>Position Specific Score Matrices （PSSMs）;</li>
<li>Profiles;</li>
<li>Hidden Markov Models （HMMs）,</li>
<li>... and a few others</li>
</ul>
<h3 id="why-msa-3014">Why MSA? 30:14</h3>
<ul>
<li><p>Help prediction of the secondary and tertiary structures of proteins of new sequences</p></li>
<li><p>Help to find motifs or signatures characteristic of protein family</p></li>
</ul>
<p><strong>Multiple sequence alignment （MSA）:</strong></p>
<ul>
<li><p><strong>Improve</strong> the alignment between two sequences</p>
<blockquote>
<p>The conserved regions: In green the identical residues and in blue the elements with the same properties</p>
</blockquote>
<p><img src="/images/ADVMB/1011170.png" /></p></li>
<li><p>Conserved regions provide information on the similarities in structure and function which can be used to align sequences, search similar sequences in the databases or annotate new sequences.</p>
<p><img src="/images/ADVMB/1011171.png" /></p></li>
</ul>
<p>Yet since the methods can only approximate the optimal solution, there can be differences between methods</p>
<p><img src="/images/ADVMB/1011172.png" /></p>
<h3 id="two-important-features-of-multiple-alignment">Two important features of multiple alignment</h3>
<ul>
<li>Position specific scoring</li>
<li>The evolutionary relationship between sequences</li>
</ul>
<h3 id="example">Example</h3>
<p><img src="/images/ADVMB/1011173.png" /></p>
<p>As shown in the picture, there are 8 fragments from immunoglobulin sequences.</p>
<p>Alignment <strong>highlights</strong>:</p>
<ul>
<li>conserved residues</li>
<li>conserved regions</li>
<li>more sophisticated patterns, like the dominance of hydrophobic residues （V,L,I） at fragment positions 1 and 3.</li>
</ul>
<p>The alignment can also enable us to infer the evolutionary history of the sequences. It looks like the first 4 sequences and the last 4 sequences are derived from 2 different common ancestors, that in turn derived from a "root" ancestor.</p>
<blockquote>
<p>But true phylogentic analysis is more complex</p>
</blockquote>
<h2 id="the-msa-problem-3552">The MSA problem 35:52</h2>
<p>Determining the alignment between N sequences is a hard problem = <strong>Combinatorial optimisation problem (COP)</strong></p>
<p>To solve this COP we need provide two systems:</p>
<ol type="1">
<li>Calculating the score of the alignment
<ul>
<li>SP scores</li>
</ul></li>
<li>Algorithm that finds the MSA with the optimal （or close to optimal） score
<ul>
<li>An optimised version of the standard pairwise <strong>DP algorithms</strong>; limited applicability but exact</li>
<li><strong>Progressive alignment</strong> （the CLUSTAL system）</li>
<li><strong>Iterative improvement</strong> methods for multiple sequence alignment （the SAGA system）</li>
</ul></li>
</ol>
<h3 id="the-scoring-system-3628">The scoring system 36:28</h3>
<p>How can we calculate the score of an MSA?</p>
<p>For pairwise alignments, we scored each column by looking at matches, mismatches, and gaps in the two sequences.</p>
<blockquote>
<p>in practice, protein sequences are scored using substitution matrices.</p>
</blockquote>
<p>Here, each column in an alignment is scored by summing the scores of all pairs of symbols in that column.</p>
<p>The score of column i （Sum-of-pairs or <strong>SP</strong>）: <span class="math display">\[
S(m_{i})=\sum_{k,l}s(m^k_i,m^l_i)
\]</span></p>
<ul>
<li><strong>m<sup>k</sup><sub>i</sub></strong> is a residue in the sequence k in column i</li>
<li><strong>s（m<sup>k</sup><sub>i</sub>, m<sup>l</sup><sub>i</sub>）</strong> The score from the substitution matrix</li>
</ul>
<blockquote>
<p>where scores s（a,b） come from a substitution scoring matrix such as a PAM or BLOSUM matrix. For simple linear gap costs, gaps are handled by defining s（a,-）and s（-,a）to be the gap cost. Otherwise gaps are scored separately.</p>
</blockquote>
<p>The score of the entire alignment is then summed over all column scores: <span class="math display">\[
S(m)=\sum_{i}S(m_{i})
\]</span></p>
<ul>
<li>So the assumption is made that the score of one column i is independent of the other columns</li>
</ul>
<h4 id="sum-of-pairs-sp-scores">Sum of pairs: SP scores</h4>
<p>SP (Sum of Pair) score is the most widely used because its simplicity.</p>
<p>The following is a MSA of 4 sequences MQPILLLV, MLRLL, MKILLL, and MPPVLILV.</p>
<p><img src="/images/ADVMB/10111730.png" /> <span class="math display">\[
\begin{align}
SP(I,-,I,V)
&amp;= score(I,-) + score(I, I) +score(I,V) + score(-,I) + score (-,V) + score(I,V)\\
&amp;= -2 + 1 + -1 + -2 + -2 + -1 \\
&amp;= -7
\end{align}
\]</span> Although there is never an entire column of gaps, if we look at any 2 sequences in the alignment, there may be columns where both have gaps. The value of score（-,-） is set equal to 0; this eliminates double counting （or extra penalization） for the use of gaps.</p>
<figure>
<img src="/images/ADVMB/1011174.png" alt="" /><figcaption>SP and its alternatives (column scores)</figcaption>
</figure>
<blockquote>
<p>Details: <a target="_blank" rel="noopener" href="https://www.cs.princeton.edu/~mona/Lecture/msa1.pdf">Princeton note</a></p>
</blockquote>
<p>Minimum entropy: <span class="math display">\[
S(m_i)=\sum_if^k_i\ln(f^k_i)
\]</span></p>
<h3 id="progressive-alignment-clustal-system-4020">Progressive alignment: CLUSTAL system 40:20</h3>
<p>CLUSTALW creates a multiple alignment by:</p>
<ol type="1">
<li><p>a pairwise alignment of the two most similar sequences is performed, producing a cluster of two aligned sequences.</p></li>
<li><p>This cluster can then be aligned to the next most related sequence or cluster of aligned sequences. Two clusters of sequences can be aligned by a simple extension of the pairwise alignment of two individual sequences.</p></li>
<li><p>The final alignment is achieved by a series of progressive, pairwise alignments that include increasingly dissimilar sequences and clusters, until all sequences have been included in the final pairwise alignment.</p>
<blockquote>
<p>Before alignment, the sequences are first clustered by similarity to produce a dendrogram （tree representation） of clustering relationships. It is this dendrogram that directs the order of the subsequent pairwise alignments.</p>
</blockquote></li>
</ol>
<p><img src="/images/ADVMB/1011175.png" /></p>
<h4 id="progressive-alignment-problems-4525">Progressive alignment problems 45:25</h4>
<p>This method suffers from at least 2 problems:</p>
<ul>
<li><strong>The local minimum problem</strong>
<ul>
<li>New sequences are added on top of existing alignments.</li>
<li><em>Hence any early errors （misaligned regions） propagate to the complete alignment （due to divergence between sequences）</em></li>
</ul></li>
<li><strong>The choice of alignment parameters</strong>
<ul>
<li>One scoring matrix, a gap-open and a gap-extension value</li>
<li><em>This works often for highly related sequences, yet starts to fail seriously as soon as sequences diverge</em></li>
</ul></li>
</ul>
<h3 id="iterative-improvement-saga-system-4622">Iterative improvement: SAGA system 46:22</h3>
<p><img src="/images/ADVMB/1011176.png" /></p>
<h2 id="profiles-5704">Profiles 57:04</h2>
<p>Using a <strong>profile</strong> we can better detect similarities between <u><strong>distantly</strong> related proteins</u>.</p>
<p>Profiles store general information properties of a MSA :</p>
<ol type="1">
<li>Frequency information concerning the residues in each column of the MSA</li>
<li>The evolutionary importance of every residue in that position</li>
</ol>
<h3 id="position-specific-scoring-matrix-pssm"><strong>Position specific scoring matrix (PSSM)</strong></h3>
<h4 id="how-to-construct-a-pssm-5756">How to construct a PSSM ? 57:56</h4>
<p>A PSSM is based on the <strong>frequencies</strong> of each residue in a specific position of a multiple alignment.</p>
<p>Determine the frequencies for each amino acid b in every column u for N<sub>seq</sub> aligned sequences, take for instance:</p>
<p><img src="/images/ADVMB/1011177.png" /> <span class="math display">\[
f_{2,R}=\frac{2}{3}=0.67\ \ f_{(i)2,R}=0.5\\
f_{6,E}=\frac{3}{3}=1\ \ f_{8,L}= \frac{1}{3}
\]</span></p>
<p><span class="math display">\[
f_{u,b}=\frac{n_{u,b}}{N_{seq}}\\
f(i)_{u,b}=\frac{\ln(1-(\frac{n_{u,b}}{N_{seq}+1}))}{\ln(\frac{1}{N_{seq}+1})}
\]</span></p>
<blockquote>
<p>Different frequency-dependent calculations possible!</p>
</blockquote>
<p><img src="/images/ADVMB/1011178.png" /></p>
<p>A profile stores for <strong>each column</strong> in the MSA, the frequency multiplied by the alignment score with every other AA. <span class="math display">\[
m_{u,a}=\sum_{b∈\{AA\}}f_{u,b}S_{a,b}
\]</span></p>
<blockquote>
<p>m<sub>u,a</sub> is actually an alignment score between a residue a and the column u. s<sub>a,b</sub> is the matrix score.</p>
</blockquote>
<p><img src="/images/ADVMB/1011179.png" /></p>
<p><img src="/images/ADVMB/1011180.png" /></p>
<blockquote>
<p>Without the penalty line, the matrix is called a PSSM （Position-specific scoring matrix）</p>
</blockquote>
<p>Probability of a match of a particular amino acid a with column u</p>
<blockquote>
<p>can be calculated in different ways!</p>
</blockquote>
<p><span class="math display">\[
m_{u,a}-\log\frac{q_{u,a}}{p_a} \ (same\ as \ pairwise)
\]</span></p>
<blockquote>
<p>This method is only useful when there are sufficient sequences and all amino acids appear at least once</p>
</blockquote>
<p>The probability p<sub>a</sub> is again the overall probability that one can find a particular amino acid in any of the positions.</p>
<p><img src="/images/ADVMB/1011181.png" /></p>
<p>The score m<sub>u,a</sub> is the score for aligning a residue a with the column u</p>
<blockquote>
<p>We can therefore use the same DP algorithm as before to align a <strong>new</strong> sequence to an <strong>existing</strong> MSA （profile）</p>
</blockquote>
<p>The biggest problem in creating profiles is that the number of sequences can be <strong>too small</strong>, resulting in q<sub>u,a</sub> of zero, and as a consequence not every amino acids is present in each column <span class="math display">\[
log 0 =-∞
\]</span> As a result the alignment can not be made using the log-formula （see PAM and BLOSUM discussion）</p>
<ul>
<li><strong>→PSEUDOCOUNTS</strong></li>
</ul>
<h4 id="pseudo-counts">Pseudo-counts</h4>
<p>Pseudo-counts are constants that one adds to the values in the profile to avoid zero’s. <span class="math display">\[
q_{u,a}=\frac{n_{u,a}+1}{N_{seq}+20}
\]</span></p>
<blockquote>
<p>Pseudo-counts provide the prior information that we have on the amino acids</p>
</blockquote>
<p>Thus, q<sub>u,a</sub> will never be 0 ! <span class="math display">\[
β=\sqrt{N_{seq}}\\
q_{u,a}=\frac{n_{u,a}+βp_a}{N_{seq}+β}
\]</span></p>
<blockquote>
<p>β is weighting factor that tunes the importance of the pseudo-counts in the value of q<sub>u,a</sub></p>
</blockquote>
<p><strong>Calculating p<sub>a</sub></strong></p>
<figure>
<img src="/images/ADVMB/1011183.png" alt="" /><figcaption><a href="%5BUniProtKB/Swiss-Prot%20Release%202021_01%20statistics%20(expasy.org)%5D(https://web.expasy.org/docs/relnotes/relstat.html)"></a></figcaption>
</figure>
<p>We can further generalize this function and introduce f<sub>u,a</sub> directly <span class="math display">\[
q_{u,a}=\frac{\alpha f_{u,a}+βp_a}{\alpha +β}
\]</span></p>
<blockquote>
<p>α is also a weighting factor that tunes the importance of the actual data.</p>
<p>Sometimes α = N<sub>seq-1</sub> is used</p>
</blockquote>
<p>When there is no data （no sequence） then the pseudo-counts will determine the values in the profiles</p>
<p>Therefore, <strong><em>pseudo-counts</em></strong> correspond to the <strong>prior distribution</strong>, which corresponds to what we know about the system before introducing the data.</p>
<p><img src="/images/ADVMB/1011185.png" /></p>
<p>Once the profile is determined one can also derive a <strong>consensus sequence</strong> which contains for every column the amino acid with the highest score m<sub>u,a</sub></p>
<p><img src="/images/ADVMB/1011186.png" /></p>
<p>Also possible to calculate the information content for every column u in the sequence （log base 2, bitscore） <span class="math display">\[
I_u=\log_220-H_u
\]</span></p>
<blockquote>
<p>information content</p>
</blockquote>
<p><span class="math display">\[
H_u=-\sum f_{u,a}\log_2f_{u,a}
\]</span></p>
<blockquote>
<p>uncertainty</p>
</blockquote>
<p>A position with one highly conserved amino acid has the lowest uncertainty, and the maximum information content.</p>
<p>The contribution of each amino acid to this score is given by f<sub>u,a</sub><em>I</em><sub>u</sub>.</p>
<figure>
<img src="/images/ADVMB/1011188.png" alt="" /><figcaption>Profiles, sequence logo 11</figcaption>
</figure>
<h4 id="aligning-to-a-profile-13317">Aligning to a profile 1:33:17</h4>
<p>Simply use either Needleman-Wunsch （L3） or Smith-Waterman （L3） to make the alignment</p>
<p><img src="/images/ADVMB/1011189.png" /></p>
<blockquote>
<p>The profile contains now the alignment scores</p>
<p>The major difficulty lies in the way in which gap penalties are assigned.</p>
</blockquote>
<p>But now use the information in the PSSM to calculate the score (<em>linear gap penalty</em>): <span class="math display">\[
S(i,j)=
\begin{cases}
S(i-1,j-1)+PSSM(seq(i)),j)\\
S(i-1,j)+PSSM(&quot;-&quot;,j)\\
S(i,j-1)+PSSM(&quot;-&quot;,j-1)\\
0
\end{cases}
\]</span></p>
<ul>
<li>j is column in PSSM</li>
<li>PSSM（“-“,j） is penalty for gap in position j</li>
<li>seq（i） is the amino acid in position i of the sequence</li>
</ul>
<p>Use in your project the following formula for calculating the values in your PSSM</p>
<p><span class="math display">\[
m_[u,a]=\log\frac{q_{u,a}}{p_a}
\]</span> With: <span class="math display">\[
q_{u,a}=\frac{\alpha f_{u,a}+βp_a}{\alpha +β}\\
\alpha=N_{seq}-1\\
β=\sqrt{N_{seq}}
\]</span></p>
<h4 id="pssm-conclusion">PSSM: Conclusion</h4>
<ul>
<li>Advantages:
<ul>
<li>Good for short, conserved regions.</li>
<li>Relatively fast and simple to implement.</li>
<li>Produce match scores that can be interpreted based on statistical theory.</li>
</ul></li>
<li>Limitations:
<ul>
<li>Insertions and deletions are strictly forbidden.</li>
<li>Relatively long sequence regions can therefore not be described with this method.</li>
</ul></li>
<li>When I use it?
<ul>
<li>To model small regions with high variability but constant length.</li>
</ul></li>
</ul>

                
            </div>
            <hr/>

            

    <div class="reprint" id="reprint-statement">
        
            <div class="reprint__author">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-user">
                        Author:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="/about" rel="external nofollow noreferrer">Chaoqun Yin</a>
                </span>
            </div>
            <div class="reprint__type">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-link">
                        Link:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="https://yintelligence.tech/2021/02/25/2021-02-25-MSA,-profiles-and-motif-detection/">https://yintelligence.tech/2021/02/25/2021-02-25-MSA,-profiles-and-motif-detection/</a>
                </span>
            </div>
            <div class="reprint__notice">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-copyright">
                        Reprint policy:
                    </i>
                </span>
                <span class="reprint-info">
                    All articles in this blog are used except for special statements
                    <a href="https://creativecommons.org/licenses/by/4.0/deed.zh" rel="external nofollow noreferrer" target="_blank">CC BY 4.0</a>
                    reprint polocy. If reproduced, please indicate source
                    <a href="/about" target="_blank">Chaoqun Yin</a>
                    !
                </span>
            </div>
        
    </div>

    <script async defer>
      document.addEventListener("copy", function (e) {
        let toastHTML = '<span>Copied successfully, please follow the reprint policy of this article</span><button class="btn-flat toast-action" onclick="navToReprintStatement()" style="font-size: smaller">more</a>';
        M.toast({html: toastHTML})
      });

      function navToReprintStatement() {
        $("html, body").animate({scrollTop: $("#reprint-statement").offset().top - 80}, 800);
      }
    </script>



            <div class="tag_share" style="display: block;">
                <div class="post-meta__tag-list" style="display: inline-block;">
                    
                        <div class="article-tag">
                            
                                <a href="/tags/Algorithms-in-Computational-biology-and-bioinformatics/">
                                    <span class="chip bg-color">Algorithms in Computational biology and bioinformatics</span>
                                </a>
                            
                        </div>
                    
                </div>
                <div class="post_share" style="zoom: 80%; width: fit-content; display: inline-block; float: right; margin: -0.15rem 0;">
                    <link rel="stylesheet" type="text/css" href="/libs/share/css/share.min.css">
<div id="article-share">

    
    <div class="social-share" data-sites="twitter,facebook,google,qq,qzone,wechat,weibo,douban,linkedin" data-wechat-qrcode-helper="<p>微信扫一扫即可分享！</p>"></div>
    <script src="/libs/share/js/social-share.min.js"></script>
    

    

</div>

                </div>
            </div>
            
        </div>
    </div>

    

    

    

    

    

    

    

    

<article id="prenext-posts" class="prev-next articles">
    <div class="row article-row">
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge left-badge text-color">
                <i class="fas fa-chevron-left"></i>&nbsp;Previous</div>
            <div class="card">
                <a href="/2021/02/25/2021-02-25-References,-Recursion,-and-Lists/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/6.jpg" class="responsive-img" alt="References, Recursion, and Lists">
                        
                        <span class="card-title">References, Recursion, and Lists</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            
                        
                    </div>
                    <div class="publish-info">
                        <span class="publish-date">
                            <i class="far fa-clock fa-fw icon-date"></i>2021-02-25
                        </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/Data-model/" class="post-category">
                                    Data model
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Data-Structures/">
                        <span class="chip bg-color">Data Structures</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge right-badge text-color">
                Next&nbsp;<i class="fas fa-chevron-right"></i>
            </div>
            <div class="card">
                <a href="/2021/02/25/2021-02-25-Exercises-for-Proofs-of-Propositional-logic/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/3.jpg" class="responsive-img" alt="Exercises for Proofs of Propositional logic">
                        
                        <span class="card-title">Exercises for Proofs of Propositional logic</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            Propositional Logic
                        
                    </div>
                    <div class="publish-info">
                            <span class="publish-date">
                                <i class="far fa-clock fa-fw icon-date"></i>2021-02-25
                            </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/Exercises/" class="post-category">
                                    Exercises
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Fundamentals-of-Computer-Science/">
                        <span class="chip bg-color">Fundamentals of Computer Science</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
    </div>
</article>

</div>


<script>
    $('#articleContent').on('copy', function (e) {
        // IE8 or earlier browser is 'undefined'
        if (typeof window.getSelection === 'undefined') return;

        var selection = window.getSelection();
        // if the selection is short let's not annoy our users.
        if (('' + selection).length < Number.parseInt('120')) {
            return;
        }

        // create a div outside of the visible area and fill it with the selected text.
        var bodyElement = document.getElementsByTagName('body')[0];
        var newdiv = document.createElement('div');
        newdiv.style.position = 'absolute';
        newdiv.style.left = '-99999px';
        bodyElement.appendChild(newdiv);
        newdiv.appendChild(selection.getRangeAt(0).cloneContents());

        // we need a <pre> tag workaround.
        // otherwise the text inside "pre" loses all the line breaks!
        if (selection.getRangeAt(0).commonAncestorContainer.nodeName === 'PRE') {
            newdiv.innerHTML = "<pre>" + newdiv.innerHTML + "</pre>";
        }

        var url = document.location.href;
        newdiv.innerHTML += '<br />'
            + 'From: Chaoqun Yin<br />'
            + 'Author: Chaoqun Yin<br />'
            + 'Link: <a href="' + url + '">' + url + '</a><br />'
            + '本文章著作权归作者所有，任何形式的转载都请注明出处。';

        selection.selectAllChildren(newdiv);
        window.setTimeout(function () {bodyElement.removeChild(newdiv);}, 200);
    });
</script>


<!-- 代码块功能依赖 -->
<script type="text/javascript" src="/libs/codeBlock/codeBlockFuction.js"></script>

<!-- 代码语言 -->

<script type="text/javascript" src="/libs/codeBlock/codeLang.js"></script>


<!-- 代码块复制 -->

<script type="text/javascript" src="/libs/codeBlock/codeCopy.js"></script>


<!-- 代码块收缩 -->

<script type="text/javascript" src="/libs/codeBlock/codeShrink.js"></script>


    </div>
    <div id="toc-aside" class="expanded col l3 hide-on-med-and-down">
        <div class="toc-widget card" style="background-color: white;">
            <div class="toc-title"><i class="far fa-list-alt"></i>&nbsp;&nbsp;TOC</div>
            <div id="toc-content"></div>
        </div>
    </div>
</div>

<!-- TOC 悬浮按钮. -->

<div id="floating-toc-btn" class="hide-on-med-and-down">
    <a class="btn-floating btn-large bg-color">
        <i class="fas fa-list-ul"></i>
    </a>
</div>


<script src="/libs/tocbot/tocbot.min.js"></script>
<script>
    $(function () {
        tocbot.init({
            tocSelector: '#toc-content',
            contentSelector: '#articleContent',
            headingsOffset: -($(window).height() * 0.4 - 45),
            collapseDepth: Number('0'),
            headingSelector: 'h1, h2, h3, h4'
        });

        // modify the toc link href to support Chinese.
        let i = 0;
        let tocHeading = 'toc-heading-';
        $('#toc-content a').each(function () {
            $(this).attr('href', '#' + tocHeading + (++i));
        });

        // modify the heading title id to support Chinese.
        i = 0;
        $('#articleContent').children('h1, h2, h3, h4').each(function () {
            $(this).attr('id', tocHeading + (++i));
        });

        // Set scroll toc fixed.
        let tocHeight = parseInt($(window).height() * 0.4 - 64);
        let $tocWidget = $('.toc-widget');
        $(window).scroll(function () {
            let scroll = $(window).scrollTop();
            /* add post toc fixed. */
            if (scroll > tocHeight) {
                $tocWidget.addClass('toc-fixed');
            } else {
                $tocWidget.removeClass('toc-fixed');
            }
        });

        
        /* 修复文章卡片 div 的宽度. */
        let fixPostCardWidth = function (srcId, targetId) {
            let srcDiv = $('#' + srcId);
            if (srcDiv.length === 0) {
                return;
            }

            let w = srcDiv.width();
            if (w >= 450) {
                w = w + 21;
            } else if (w >= 350 && w < 450) {
                w = w + 18;
            } else if (w >= 300 && w < 350) {
                w = w + 16;
            } else {
                w = w + 14;
            }
            $('#' + targetId).width(w);
        };

        // 切换TOC目录展开收缩的相关操作.
        const expandedClass = 'expanded';
        let $tocAside = $('#toc-aside');
        let $mainContent = $('#main-content');
        $('#floating-toc-btn .btn-floating').click(function () {
            if ($tocAside.hasClass(expandedClass)) {
                $tocAside.removeClass(expandedClass).hide();
                $mainContent.removeClass('l9');
            } else {
                $tocAside.addClass(expandedClass).show();
                $mainContent.addClass('l9');
            }
            fixPostCardWidth('artDetail', 'prenext-posts');
        });
        
    });
</script>

    

</main>


<script src="https://cdn.bootcss.com/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<script>
    MathJax.Hub.Config({
        tex2jax: {inlineMath: [['$', '$'], ['\(', '\)']]}
    });
</script>



    <footer class="page-footer bg-color">
    
    <div class="container row center-align" style="margin-bottom: 15px !important;">
        <div class="col s12 m8 l8 copy-right">
            Copyright&nbsp;&copy;
            
                <span id="year">2021</span>
            
            <span id="year">2021</span>
            <a href="/about" target="_blank">Chaoqun Yin</a>
            |&nbsp;Powered by&nbsp;<a href="https://hexo.io/" target="_blank">Hexo</a>
            |&nbsp;Theme&nbsp;<a href="https://github.com/blinkfox/hexo-theme-matery" target="_blank">Matery</a>
            <br>
            
            
            
            
            
            
            <span id="busuanzi_container_site_pv">
                |&nbsp;<i class="far fa-eye"></i>&nbsp;总访问量:&nbsp;<span id="busuanzi_value_site_pv"
                    class="white-color"></span>&nbsp;次
            </span>
            
            
            <span id="busuanzi_container_site_uv">
                |&nbsp;<i class="fas fa-users"></i>&nbsp;总访问人数:&nbsp;<span id="busuanzi_value_site_uv"
                    class="white-color"></span>&nbsp;人
            </span>
            
            <br>
            
            <span id="sitetime">载入运行时间...</span>
            <script>
                function siteTime() {
                    var seconds = 1000;
                    var minutes = seconds * 60;
                    var hours = minutes * 60;
                    var days = hours * 24;
                    var years = days * 365;
                    var today = new Date();
                    var startYear = "2021";
                    var startMonth = "1";
                    var startDate = "28";
                    var startHour = "2";
                    var startMinute = "1";
                    var startSecond = "0";
                    var todayYear = today.getFullYear();
                    var todayMonth = today.getMonth() + 1;
                    var todayDate = today.getDate();
                    var todayHour = today.getHours();
                    var todayMinute = today.getMinutes();
                    var todaySecond = today.getSeconds();
                    var t1 = Date.UTC(startYear, startMonth, startDate, startHour, startMinute, startSecond);
                    var t2 = Date.UTC(todayYear, todayMonth, todayDate, todayHour, todayMinute, todaySecond);
                    var diff = t2 - t1;
                    var diffYears = Math.floor(diff / years);
                    var diffDays = Math.floor((diff / days) - diffYears * 365);
                    var diffHours = Math.floor((diff - (diffYears * 365 + diffDays) * days) / hours);
                    var diffMinutes = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours) /
                        minutes);
                    var diffSeconds = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours -
                        diffMinutes * minutes) / seconds);
                    if (startYear == todayYear) {
                        document.getElementById("year").innerHTML = todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已安全运行 " + diffDays + " 天 " + diffHours +
                            " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    } else {
                        document.getElementById("year").innerHTML = startYear + " - " + todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已安全运行 " + diffYears + " 年 " + diffDays +
                            " 天 " + diffHours + " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    }
                }
                setInterval(siteTime, 1000);
            </script>
            
            <br>
            
        </div>
        <div class="col s12 m4 l4 social-link social-statis">
    <a href="https://github.com/yintelligence" class="tooltipped" target="_blank" data-tooltip="访问我的GitHub" data-position="top" data-delay="50">
        <i class="fab fa-github"></i>
    </a>



    <a href="mailto:Yintelligence@outlook.com" class="tooltipped" target="_blank" data-tooltip="邮件联系我" data-position="top" data-delay="50">
        <i class="fas fa-envelope-open"></i>
    </a>













</div>
    </div>
</footer>

<div class="progress-bar"></div>


    <!-- 搜索遮罩框 -->
<div id="searchModal" class="modal">
    <div class="modal-content">
        <div class="search-header">
            <span class="title"><i class="fas fa-search"></i>&nbsp;&nbsp;Search</span>
            <input type="search" id="searchInput" name="s" placeholder="Please enter a search keyword"
                   class="search-input">
        </div>
        <div id="searchResult"></div>
    </div>
</div>

<script type="text/javascript">
$(function () {
    var searchFunc = function (path, search_id, content_id) {
        'use strict';
        $.ajax({
            url: path,
            dataType: "xml",
            success: function (xmlResponse) {
                // get the contents from search data
                var datas = $("entry", xmlResponse).map(function () {
                    return {
                        title: $("title", this).text(),
                        content: $("content", this).text(),
                        url: $("url", this).text()
                    };
                }).get();
                var $input = document.getElementById(search_id);
                var $resultContent = document.getElementById(content_id);
                $input.addEventListener('input', function () {
                    var str = '<ul class=\"search-result-list\">';
                    var keywords = this.value.trim().toLowerCase().split(/[\s\-]+/);
                    $resultContent.innerHTML = "";
                    if (this.value.trim().length <= 0) {
                        return;
                    }
                    // perform local searching
                    datas.forEach(function (data) {
                        var isMatch = true;
                        var data_title = data.title.trim().toLowerCase();
                        var data_content = data.content.trim().replace(/<[^>]+>/g, "").toLowerCase();
                        var data_url = data.url;
                        data_url = data_url.indexOf('/') === 0 ? data.url : '/' + data_url;
                        var index_title = -1;
                        var index_content = -1;
                        var first_occur = -1;
                        // only match artiles with not empty titles and contents
                        if (data_title !== '' && data_content !== '') {
                            keywords.forEach(function (keyword, i) {
                                index_title = data_title.indexOf(keyword);
                                index_content = data_content.indexOf(keyword);
                                if (index_title < 0 && index_content < 0) {
                                    isMatch = false;
                                } else {
                                    if (index_content < 0) {
                                        index_content = 0;
                                    }
                                    if (i === 0) {
                                        first_occur = index_content;
                                    }
                                }
                            });
                        }
                        // show search results
                        if (isMatch) {
                            str += "<li><a href='" + data_url + "' class='search-result-title'>" + data_title + "</a>";
                            var content = data.content.trim().replace(/<[^>]+>/g, "");
                            if (first_occur >= 0) {
                                // cut out 100 characters
                                var start = first_occur - 20;
                                var end = first_occur + 80;
                                if (start < 0) {
                                    start = 0;
                                }
                                if (start === 0) {
                                    end = 100;
                                }
                                if (end > content.length) {
                                    end = content.length;
                                }
                                var match_content = content.substr(start, end);
                                // highlight all keywords
                                keywords.forEach(function (keyword) {
                                    var regS = new RegExp(keyword, "gi");
                                    match_content = match_content.replace(regS, "<em class=\"search-keyword\">" + keyword + "</em>");
                                });

                                str += "<p class=\"search-result\">" + match_content + "...</p>"
                            }
                            str += "</li>";
                        }
                    });
                    str += "</ul>";
                    $resultContent.innerHTML = str;
                });
            }
        });
    };

    searchFunc('/search.xml', 'searchInput', 'searchResult');
});
</script>

    <!-- 回到顶部按钮 -->
<div id="backTop" class="top-scroll">
    <a class="btn-floating btn-large waves-effect waves-light" href="#!">
        <i class="fas fa-arrow-up"></i>
    </a>
</div>


    <script src="/libs/materialize/materialize.min.js"></script>
    <script src="/libs/masonry/masonry.pkgd.min.js"></script>
    <script src="/libs/aos/aos.js"></script>
    <script src="/libs/scrollprogress/scrollProgress.min.js"></script>
    <script src="/libs/lightGallery/js/lightgallery-all.min.js"></script>
    <script src="/js/matery.js"></script>

    <!-- Baidu Analytics -->

    <!-- Baidu Push -->

<script>
    (function () {
        var bp = document.createElement('script');
        var curProtocol = window.location.protocol.split(':')[0];
        if (curProtocol === 'https') {
            bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
        } else {
            bp.src = 'http://push.zhanzhang.baidu.com/push.js';
        }
        var s = document.getElementsByTagName("script")[0];
        s.parentNode.insertBefore(bp, s);
    })();
</script>

    
    <script src="/libs/others/clicklove.js" async="async"></script>
    
    
    <script async src="/libs/others/busuanzi.pure.mini.js"></script>
    

    

    

	
    

    

    

    
    <script src="/libs/instantpage/instantpage.js" type="module"></script>
    

</body>

</html>
