<!DOCTYPE HTML>
<html lang="en">


<head>
    <meta charset="utf-8">
    <meta name="keywords" content="Quantitative imaging, Chaoqun Yin">
    <meta name="description" content="Veni, Vidi, Vici">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="renderer" content="webkit|ie-stand|ie-comp">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="format-detection" content="telephone=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <!-- Global site tag (gtag.js) - Google Analytics -->


    <title>Quantitative imaging | Chaoqun Yin</title>
    <link rel="icon" type="image/png" href="/favicon.png">

    <link rel="stylesheet" type="text/css" href="/libs/awesome/css/all.css">
    <link rel="stylesheet" type="text/css" href="/libs/materialize/materialize.min.css">
    <link rel="stylesheet" type="text/css" href="/libs/aos/aos.css">
    <link rel="stylesheet" type="text/css" href="/libs/animate/animate.min.css">
    <link rel="stylesheet" type="text/css" href="/libs/lightGallery/css/lightgallery.min.css">
    <link rel="stylesheet" type="text/css" href="/css/matery.css">
    <link rel="stylesheet" type="text/css" href="/css/my.css">

    <script src="/libs/jquery/jquery.min.js"></script>

<meta name="generator" content="Hexo 5.3.0"></head>




<body>
    <header class="navbar-fixed">
    <nav id="headNav" class="bg-color nav-transparent">
        <div id="navContainer" class="nav-wrapper container">
            <div class="brand-logo">
                <a href="/" class="waves-effect waves-light">
                    
                    <img src="/medias/logo.png" class="logo-img" alt="LOGO">
                    
                    <span class="logo-span">Chaoqun Yin</span>
                </a>
            </div>
            

<a href="#" data-target="mobile-nav" class="sidenav-trigger button-collapse"><i class="fas fa-bars"></i></a>
<ul class="right nav-menu">
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/" class="waves-effect waves-light">
      
      <i class="fas fa-home" style="zoom: 0.6;"></i>
      
      <span>Index</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/tags" class="waves-effect waves-light">
      
      <i class="fas fa-tags" style="zoom: 0.6;"></i>
      
      <span>Tags</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/categories" class="waves-effect waves-light">
      
      <i class="fas fa-bookmark" style="zoom: 0.6;"></i>
      
      <span>Categories</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/archives" class="waves-effect waves-light">
      
      <i class="fas fa-archive" style="zoom: 0.6;"></i>
      
      <span>Archives</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/about" class="waves-effect waves-light">
      
      <i class="fas fa-user-circle" style="zoom: 0.6;"></i>
      
      <span>About</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/contact" class="waves-effect waves-light">
      
      <i class="fas fa-comments" style="zoom: 0.6;"></i>
      
      <span>Contact</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/friends" class="waves-effect waves-light">
      
      <i class="fas fa-address-book" style="zoom: 0.6;"></i>
      
      <span>Friends</span>
    </a>
    
  </li>
  
  <li>
    <a href="#searchModal" class="modal-trigger waves-effect waves-light">
      <i id="searchIcon" class="fas fa-search" title="Search" style="zoom: 0.85;"></i>
    </a>
  </li>
</ul>


<div id="mobile-nav" class="side-nav sidenav">

    <div class="mobile-head bg-color">
        
        <img src="/medias/logo.png" class="logo-img circle responsive-img">
        
        <div class="logo-name">Chaoqun Yin</div>
        <div class="logo-desc">
            
            Veni, Vidi, Vici
            
        </div>
    </div>

    

    <ul class="menu-list mobile-menu-list">
        
        <li class="m-nav-item">
	  
		<a href="/" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-home"></i>
			
			Index
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/tags" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-tags"></i>
			
			Tags
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/categories" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-bookmark"></i>
			
			Categories
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/archives" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-archive"></i>
			
			Archives
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/about" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-user-circle"></i>
			
			About
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/contact" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-comments"></i>
			
			Contact
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/friends" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-address-book"></i>
			
			Friends
		</a>
          
        </li>
        
        
        <li><div class="divider"></div></li>
        <li>
            <a href="https://github.com/blinkfox/hexo-theme-matery" class="waves-effect waves-light" target="_blank">
                <i class="fab fa-github-square fa-fw"></i>Fork Me
            </a>
        </li>
        
    </ul>
</div>


        </div>

        
            <style>
    .nav-transparent .github-corner {
        display: none !important;
    }

    .github-corner {
        position: absolute;
        z-index: 10;
        top: 0;
        right: 0;
        border: 0;
        transform: scale(1.1);
    }

    .github-corner svg {
        color: #0f9d58;
        fill: #fff;
        height: 64px;
        width: 64px;
    }

    .github-corner:hover .octo-arm {
        animation: a 0.56s ease-in-out;
    }

    .github-corner .octo-arm {
        animation: none;
    }

    @keyframes a {
        0%,
        to {
            transform: rotate(0);
        }
        20%,
        60% {
            transform: rotate(-25deg);
        }
        40%,
        80% {
            transform: rotate(10deg);
        }
    }
</style>

<a href="https://github.com/blinkfox/hexo-theme-matery" class="github-corner tooltipped hide-on-med-and-down" target="_blank"
   data-tooltip="Fork Me" data-position="left" data-delay="50">
    <svg viewBox="0 0 250 250" aria-hidden="true">
        <path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path>
        <path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
              fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path>
        <path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
              fill="currentColor" class="octo-body"></path>
    </svg>
</a>
        
    </nav>

</header>

    



<div class="bg-cover pd-header post-cover" style="background-image: url('/medias/featureimages/15.jpg')">
    <div class="container" style="right: 0px;left: 0px;">
        <div class="row">
            <div class="col s12 m12 l12">
                <div class="brand">
                    <h1 class="description center-align post-title">Quantitative imaging</h1>
                </div>
            </div>
        </div>
    </div>
</div>




<main class="post-container content">

    
    <link rel="stylesheet" href="/libs/tocbot/tocbot.css">
<style>
    #articleContent h1::before,
    #articleContent h2::before,
    #articleContent h3::before,
    #articleContent h4::before,
    #articleContent h5::before,
    #articleContent h6::before {
        display: block;
        content: " ";
        height: 100px;
        margin-top: -100px;
        visibility: hidden;
    }

    #articleContent :focus {
        outline: none;
    }

    .toc-fixed {
        position: fixed;
        top: 64px;
    }

    .toc-widget {
        width: 345px;
        padding-left: 20px;
    }

    .toc-widget .toc-title {
        padding: 35px 0 15px 17px;
        font-size: 1.5rem;
        font-weight: bold;
        line-height: 1.5rem;
    }

    .toc-widget ol {
        padding: 0;
        list-style: none;
    }

    #toc-content {
        padding-bottom: 30px;
        overflow: auto;
    }

    #toc-content ol {
        padding-left: 10px;
    }

    #toc-content ol li {
        padding-left: 10px;
    }

    #toc-content .toc-link:hover {
        color: #42b983;
        font-weight: 700;
        text-decoration: underline;
    }

    #toc-content .toc-link::before {
        background-color: transparent;
        max-height: 25px;

        position: absolute;
        right: 23.5vw;
        display: block;
    }

    #toc-content .is-active-link {
        color: #42b983;
    }

    #floating-toc-btn {
        position: fixed;
        right: 15px;
        bottom: 76px;
        padding-top: 15px;
        margin-bottom: 0;
        z-index: 998;
    }

    #floating-toc-btn .btn-floating {
        width: 48px;
        height: 48px;
    }

    #floating-toc-btn .btn-floating i {
        line-height: 48px;
        font-size: 1.4rem;
    }
</style>
<div class="row">
    <div id="main-content" class="col s12 m12 l9">
        <!-- 文章内容详情 -->
<div id="artDetail">
    <div class="card">
        <div class="card-content article-info">
            <div class="row tag-cate">
                <div class="col s7">
                    
                    <div class="article-tag">
                        
                            <a href="/tags/Biomedical-Signals-and-Images/">
                                <span class="chip bg-color">Biomedical Signals and Images</span>
                            </a>
                        
                    </div>
                    
                </div>
                <div class="col s5 right-align">
                    
                    <div class="post-cate">
                        <i class="fas fa-bookmark fa-fw icon-category"></i>
                        
                            <a href="/categories/Digital-Health/" class="post-category">
                                Digital Health
                            </a>
                        
                    </div>
                    
                </div>
            </div>

            <div class="post-info">
                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-minus fa-fw"></i>Publish Date:&nbsp;&nbsp;
                    2021-04-26
                </div>
                

                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-check fa-fw"></i>Update Date:&nbsp;&nbsp;
                    2021-08-15
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-file-word fa-fw"></i>Word Count:&nbsp;&nbsp;
                    6k
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-clock fa-fw"></i>Read Times:&nbsp;&nbsp;
                    30 Min
                </div>
                

                
                    <div id="busuanzi_container_page_pv" class="info-break-policy">
                        <i class="far fa-eye fa-fw"></i>Read Count:&nbsp;&nbsp;
                        <span id="busuanzi_value_page_pv"></span>
                    </div>
				
            </div>
        </div>
        <hr class="clearfix">

        
        <!-- 是否加载使用自带的 prismjs. -->
        <link rel="stylesheet" href="/libs/prism/prism.css">
        

        

        <div class="card-content article-card-content">
            <div id="articleContent">
                <h1 id="quantitative-imaging">Quantitative imaging</h1>
<p><a href="https://yintelligence.tech/2021/01/06/2021-01-06-VUB-4019776DNR-Biomedical-Signals-and-Images/">VUB 4019776DNR Biomedical Signals and Images</a></p>
<p><a target="_blank" rel="noopener" href="https://canvas.vub.be/courses/18933/files?preview=651177">Slide</a> + <a target="_blank" rel="noopener" href="https://vub.cloud.panopto.eu/Panopto/Pages/Viewer.aspx?id=5f1c0cf4-94ed-485c-947d-ac7b00f255b6">Video</a></p>
<h2 id="quantitative-imaging-1">Quantitative imaging</h2>
<h3 id="used-definitions">Used definitions</h3>
<p><u>The extraction of quantifiable features from medical images</u> for the assessment of normal or the severity, degree of change, or status of a disease, injury, or chronic condition relative to normal</p>
<p>从医学图像中提取<strong>可量化的特征</strong>，以评估正常情况或疾病、损伤或慢性病相对于正常情况的严重程度、变化程度或状态</p>
<p>In the clinic, it may cover</p>
<ul>
<li><p>Anatomical, functional and molecular imaging</p>
<p>解剖、功能和分子成像</p></li>
<li><p>Data analyses</p>
<p>数据分析</p></li>
<li><p>Display methods</p>
<p>显示方式</p></li>
<li><p>Reporting structures</p>
<p>报告结构</p></li>
</ul>
<h4 id="biomarkers-生物标志物">Biomarkers 生物标志物</h4>
<p><strong>Definition</strong>:</p>
<ul>
<li><p>Also biological marker</p></li>
<li><p>A <strong>measurable indicator</strong> of some biological state or condition. Biomarkers are often measured and evaluated to examine normal biological processes, pathogenic processes, or pharmacologic responses to a therapeutic intervention</p>
<p>某种生物状态或状况的<strong>可测量指标</strong>。 通常测量和评估生物标志物以检查正常的生物过程、致病过程或对治疗干预的药理反应。</p></li>
</ul>
<p>An imaging biomarker: A biomarker detectable in an image</p>
<p>成像生物标志物：可在图像中检测到的生物标志物</p>
<h4 id="reproducible-research-recommendations">Reproducible Research Recommendations</h4>
<ul>
<li>Reproducibility 再现性
<ul>
<li>A study is reproducible if you can take the original data and the computer code used to analyse the data and reproduce all of the numerical findings from the study. This may initially sound like a trivial task but experience has shown that it’s not always easy to achieve this seemingly minimal standard.</li>
</ul></li>
<li>Replicability 可复制性
<ul>
<li>This is the act of repeating an entire study, independently of the original investigator without the use of original data （but generally using the same methods）.</li>
</ul></li>
</ul>
<h4 id="other-definitions">Other definitions</h4>
<p>Repeatability, Reproducibility, and Replicability: Tackling the 3R challenge in bio-interface science and engineering</p>
<p>可重复性、再现性和可复制性：应对生物界面科学与工程中的 3R 挑战</p>
<h3 id="what-does-quantitative-imply">What does quantitative imply?</h3>
<ul>
<li>Reproducible 可重现
<ul>
<li><p>Agreement （up to a tolerance） of <u>repeated measurements on the same patient</u></p>
<p>对同一患者重复测量的一致性（达到公差）</p></li>
</ul></li>
<li>Comparable 可比
<ul>
<li><p>Measures are suited to be <strong>compared between patients</strong></p>
<p>措施适合<strong>在患者之间</strong>进行<strong>比较</strong></p></li>
</ul></li>
<li>In practice also, Replicability 可复制性
<ul>
<li><p>Being able to <u>independently achieve similar measurements</u> across institutes, with slightly varying set-up, procedure, etc</p>
<p>能够在不同机构独立实现类似的测量，设置、程序等略有不同</p></li>
</ul></li>
</ul>
<h3 id="how-to-obtain-quantitative-measurements">How to obtain quantitative measurements</h3>
<ul>
<li><p>Morphology 形态学</p>
<ul>
<li><p>Thickness, length, surface, volume, shape, … of a structure</p>
<blockquote>
<p>The size of the detected object depends on it's intensity</p>
</blockquote></li>
</ul></li>
<li><p>Intensity 强度</p>
<ul>
<li><p>Mean, median, standard deviation, skew, … of a structure</p>
<blockquote>
<p>The intensity parameters depend on the region considered</p>
</blockquote></li>
</ul>
<blockquote>
<p>Main features to measure</p>
</blockquote></li>
<li><p>Texture 质地</p>
<ul>
<li><p>A systematic local variation of the image values – Scale dependent.</p>
<p>图像值的系统性局部变化——与比例相关。</p></li>
<li><p>Different types of texture are visible at different scales</p>
<p>不同的尺度上可以看到不同类型的纹理</p>
<blockquote>
<p>Is in inherently a variation in the intensity 本质上是强度的变化</p>
</blockquote></li>
</ul></li>
<li><p>Frequencies 频率</p>
<ul>
<li><p>Fourier, wavelet or other transforms</p>
<blockquote>
<p>Closely related to texture, and intensity</p>
<p>与质地和强度密切相关</p>
</blockquote></li>
</ul></li>
</ul>
<h4 id="relation-between-measures">Relation between measures</h4>
<ul>
<li><p>All measures rely on a <strong>segmentation</strong> of the <strong>region of interest （ROI）</strong></p>
<p>所有措施都依赖于感兴趣区域的分割（ROI）</p>
<ul>
<li><p>Which was performed using a region-based or edge-based technique</p>
<p>这是使用基于区域或基于边缘的技术执行的</p></li>
<li><p>All measures are therefore influenced by the quality of the intensities or the intensity differences （edge）…</p>
<p>因此，所有测量都受到强度质量或强度差异（边缘）的影响……</p></li>
</ul></li>
<li><p>There is an overlap between texture and frequency</p>
<p>纹理和频率之间存在重叠</p>
<ul>
<li>But the are not the same 但不一样</li>
</ul></li>
<li><p>Measures are <strong>related</strong>, but do carry <strong>complementary</strong> information</p>
<p>措施是<strong>相关的</strong>，但确实带有<strong>补充</strong>信息</p></li>
<li><p>All require the imaging to be performed / adhere to certain standards</p>
<p>所有都需要执行成像/遵守某些标准</p>
<ul>
<li>In order to be quantitative</li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314370.png" /></p>
<h4 id="how-to-obtain-quantitative-measures">How to obtain quantitative measures?</h4>
<ul>
<li>Identify the measures 确定措施
<ul>
<li>Normal variance 正态方差</li>
<li>Abnormal values 异常值</li>
</ul></li>
<li>Acquire the images 获取图像
<ul>
<li>Quality assurance 质量保证</li>
<li>Standardization of protocols 协议标准化</li>
</ul></li>
<li>Extract the measures 提取措施
<ul>
<li>Reduce observer variability 减少观察者的可变性</li>
<li>Algorithm reproducibility 算法重现性</li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314371.png" /></p>
<h5 id="quantitative-measures-in-ct">Quantitative measures in CT</h5>
<ul>
<li><p>Image intensity in CT is standardized CT中的图像强度是标准化的</p>
<p>Remember: <strong>CT or Hounsfield units</strong></p>
<ul>
<li><p>Linear transformation of the original linear attenuation coefficient w</p>
<p>原始线性衰减系数 w 的线性变换</p></li>
<li><p>HU value is linearly related to attenuation µ within that volume （~electron density）</p>
<p>HU 值与该体积内的衰减 μ 线性相关（~电子密度） <span class="math display">\[
CT(x, y) =1000 \frac{µ(x, y) − µ_{water}}{µ_{water}}
\]</span></p></li>
</ul></li>
<li><p>Intensity measurements are, to a certain extent, <strong>reproducible, comparable, replicable</strong></p>
<p>强度测量在某种程度上是<strong>可重复的、可比较的、可复制的</strong></p>
<ul>
<li><p>Quantification can be improved by standardizing the protocol （energy （kV）, exposure （mA）, reconstruction）</p>
<p>量化可以通过标准化协议来提高（能量（kV），曝光（mA），重建）</p></li>
</ul></li>
</ul>
<p>CT has excellent geometric accuracy</p>
<p>CT具有出色的几何精度</p>
<ul>
<li>Low geometric distortion 低几何失真
<ul>
<li><p>Distances, angles and straight lines are preserved</p>
<p>保留距离、角度和直线</p></li>
</ul></li>
<li>High resolution 高分辨率
<ul>
<li><p>Sub-millimetre resolution in all planes</p>
<p>所有平面的亚毫米分辨率</p></li>
</ul></li>
</ul>
<p>Morphological assessments are, to a certain extent, <strong>reproducible, comparable, replicable</strong></p>
<p>形态学评估在一定程度上是可<strong>重复的、可比较的、可复制的</strong></p>
<h5 id="quantitative-measures-in-mri">Quantitative measures in MRI</h5>
<p>Geometric accuracy is usually lower than CT</p>
<p>几何精度通常低于CT</p>
<ul>
<li><p>Lower resolution, depending on the site and protocol</p>
<p>较低的分辨率，取决于站点和协议</p></li>
<li><p>Some sequences exhibit distortion （DWI）</p>
<p>部分序列出现失真（DWI）</p></li>
</ul>
<p><strong>Can still be used for morphological assessments, given a suitable tolerance</strong></p>
<p><strong>仍可用于形态学评估，给定合适的耐受性</strong></p>
<p>Image intensities in MRI are relative 强度是相对的</p>
<ul>
<li>MRI is in essence a contrast map MRI本质上是一张对比图</li>
<li>No direct meaning of the image intensity 图像强度没有直接意义</li>
</ul>
<p><strong>How to obtain quantitative intensity measurements?</strong> <strong>如何获得定量强度测量值？</strong></p>
<p>Quantitative MRI is often obtained by extracting <strong>physical quantities</strong> using advanced sequences</p>
<p>定量 MRI 通常是通过使用高级序列提取<strong>物理量</strong>来获得的</p>
<ul>
<li><p>Perfusion measurements from dynamic contrast enhanced （DCE） or perfusion weighted imaging （PWI）</p>
<p>来自动态对比增强（DCE）或灌注加权成像（PWI）的灌注测量</p>
<ul>
<li>Blood flow, blood volume, permeability, …</li>
</ul></li>
<li><p>Diffusion measurements from diffusion-weighted imaging （DWI）</p>
<p>弥散加权成像（DWI）的弥散测量</p>
<ul>
<li>Apparent diffusion coefficient （ADC）</li>
</ul></li>
<li><p>Physical quantities which, if measured correctly are <strong>reproducible, comparable</strong></p>
<p>物理量，如果正确测量是可<strong>再现的，可比</strong></p>
<ul>
<li><p>Depending on how complex the protocol is, also <strong>replicable</strong></p>
<p>根据协议的复杂程度，也<strong>可复制</strong></p></li>
</ul></li>
</ul>
<h5 id="quantitative-measures-in-pet--ct">Quantitative measures in PET （-CT）</h5>
<p>Geometric accuracy is much lower than CT 几何精度远低于CT</p>
<ul>
<li><p>Significantly lower resolution, depending on the acquisition time and protocol</p>
<p>分辨率显着降低，具体取决于采集时间和协议</p></li>
<li><p>The combination with CT can offer a compromise</p>
<p>与 CT 的组合可以提供折衷方案</p></li>
<li><p><strong>Morphological assessments of PET-positive lesions, measured on PET and/or CT</strong></p>
<p><strong>PET 阳性病变的形态学评估，在 PET 和/或 CT 上测量</strong></p></li>
</ul>
<p>Image intensities in PET depend on many things PET 中的图像强度取决于许多因素</p>
<ul>
<li>The activity injected 注入的活动</li>
<li>The size of the patient 患者体型</li>
<li>The time after injection 注射后时间</li>
<li>The metabolic activity of the cells 细胞的代谢活动</li>
<li>…</li>
</ul>
<p><strong>How to obtain quantitative intensity measurements from PET?</strong></p>
<ul>
<li><p><strong>Measurements related to the metabolic activity the cells</strong></p>
<p>与<strong>细胞代谢活动</strong>相关的测量</p></li>
</ul>
<h3 id="standardized-uptake-value-suv">Standardized Uptake Value （SUV）</h3>
<p>Standardise the measurement procedure</p>
<ul>
<li><del>Always same activity injected</del></li>
<li><del>Always same time after injection</del></li>
<li><del>Always same account the size of the patient… euh?!?</del></li>
</ul>
<p>Not a very practical way to work</p>
<ul>
<li>Reproducible</li>
<li>Comparable on a patient basis, not interpatient</li>
<li>Difficult to replicate</li>
</ul>
<p><span class="math display">\[
SUV(time)=\frac{Tracer\ uptake}{Injected\ Activity\ /\ Weight}
\]</span></p>
<p>Under certain circumstances, <sup>18</sup>FDG SUV correlates with metabolic rate of glucose and/or the number of viable tumor cells</p>
<p>在某些情况下，<sup>18</sup>FDG SUV 与葡萄糖代谢率和/或存活肿瘤细胞的数量相关</p>
<ul>
<li><p>SUV is a simplified quantitative measure that can be routinely performed in clinical PET studies</p>
<p>SUV 是一种简化的定量测量，可在临床 PET 研究中常规进行</p></li>
</ul>
<p>Adjusts image intensities for 调整图像强度</p>
<ul>
<li>Differences in patient size 患者大小的差异</li>
<li>Injected activity 注入活动</li>
<li>Correct activity to decay at a reference time 在参考时间衰减的正确活动</li>
</ul>
<p>The resulting intensities correlate with metabolic activity of tumor cells AND are <strong>reproducible</strong>, <strong>comparable</strong> and <strong>replicable</strong> to a certain extent.</p>
<p>由此产生的强度与肿瘤细胞的代谢活动相关，并且在一定程度上是可<strong>重复的</strong>、<strong>可比较的</strong>和<strong>可复制的</strong>。</p>
<h4 id="suv-units">SUV Units</h4>
<ul>
<li><p>Based on the following assumptions （approximations）:</p>
<p>基于以下假设（近似值）</p>
<ul>
<li><p>Water-equivalent tissue 水当量组织</p></li>
<li><p>A body weight correction in grams 以克为单位的体重校正</p></li>
<li><p>Decay-correction to the time of injection 对注射时间的衰减校正</p></li>
<li><p>Known concentration in consistent units of mCi/ml or MBq/cc</p>
<p>已知浓度，单位为 mCi/ml 或 MBq/cc</p></li>
</ul></li>
<li><p>The SUV is a relative, unit-less quantity 一个相对的、无单位的数量</p>
<ul>
<li><p>It relates the image derived radioactivity concentration to the whole-body concentration of the injected radioactivity</p>
<p>它将图像衍生的放射性浓度与注射放射性的全身浓度相关联</p></li>
<li><p>Under standard settings （time, injection, activity, etc ）</p>
<p>在标准设置下（时间、注射、活动等）</p></li>
</ul></li>
</ul>
<h4 id="suv-计算的变化">SUV 计算的变化</h4>
<ul>
<li><p>Instead of body mass 代替体重</p>
<ul>
<li>Body surface area or lean body weight 体表面积或瘦体重</li>
<li>Sometimes denoted with subscripts （SUV<sub>BW</sub>, SUV<sub>BSA</sub>, SUV<sub>LBM</sub>）</li>
</ul></li>
<li><p>Linear correction for circulating glucose 循环葡萄糖的线性校正</p></li>
<li><p>Designated uptake period 指定吸收期</p>
<ul>
<li><p>Delayed scanning may reduce background physiologic uptake</p>
<p>延迟扫描可能会减少背景生理摄取</p></li>
</ul></li>
</ul>
<h4 id="practical-examples-from-clinical-practice">Practical examples from clinical practice</h4>
<ul>
<li><p>Role of imaging in oncology 影像学在肿瘤学中的作用</p>
<ul>
<li><p>Morphological response criteria for solid tumours</p>
<p>实体瘤的形态学反应标准</p>
<ul>
<li><strong>RECIST</strong></li>
<li>Volumetry</li>
</ul></li>
<li><p>Functional assessment of solid tumours</p>
<p>实体瘤的功能评估</p>
<ul>
<li>Mean, max and peak SUV</li>
</ul></li>
</ul></li>
<li><p>Cartilage for osteo-arthritis 用于骨关节炎的软骨</p></li>
</ul>
<p>Role of quantitative imaging in <strong>oncology</strong></p>
<p><img src="/images/HMS/2314372.png" /></p>
<h5 id="recist">RECIST</h5>
<p><strong>RECIST</strong> 1.1 – Response Evaluation Criteria in Solid Tumors 实体瘤的反应评估标准</p>
<p>Simple measurement of tumor diameter. 简单测量肿瘤直径</p>
<ul>
<li><p>Maximum of 5 biggest target lesions</p>
<p>最多 5 个最大的目标病灶</p></li>
<li><p>Baseline modules must measure &gt; 10 mm</p>
<p>基线模块的尺寸必须 &gt; 10 mm</p></li>
<li><p>Preferred imaging technique – CT （variability of MRI scan parameters）</p>
<p>首选成像技术——CT（MRI扫描参数的可变性）</p></li>
</ul>
<blockquote>
<p>WHO – Similar criteria based on two orthogonal measures</p>
</blockquote>
<p><img src="/images/HMS/2314373.png" /></p>
<h5 id="advantages-limitations">Advantages &amp; Limitations</h5>
<ul>
<li><p>Practical guidelines that can be applied on a wide range of image datasets of varying characteristics</p>
<p>可<strong>应用于各种不同特征</strong>的图像数据集的实用指南</p></li>
<li><p>Downsides</p>
<ul>
<li><p>One- or two-dimensional measurement for 3D structure</p>
<p>3D 结构的一维或二维测量</p></li>
<li><p>Only 5 top structures are taken into account</p>
<p>仅考虑 5 个顶层结构</p></li>
<li><p>Long time between treatment effect and visible diameter change</p>
<p>治疗效果和可见直径变化之间的时间长</p>
<ul>
<li><p>Killing a tumor does not necessarily mean it disappears</p>
<p>杀死肿瘤并不一定意味着它会消失</p></li>
</ul></li>
<li><p>Variability between doctors</p>
<p>医生之间的差异</p></li>
<li><p>Time consuming 耗时的</p></li>
</ul></li>
</ul>
<p><strong>Trend towards volumetric measurements</strong></p>
<h5 id="d-2d-vs-volumetric-3d">1D, 2D vs volumetric （3D）</h5>
<p><img src="/images/HMS/2314374.png" /></p>
<p>Assessment of cavitated lesions</p>
<ul>
<li><p>Tumoural volume decreased over time</p>
<p>肿瘤体积随时间减少</p></li>
<li><p>1D or 2D tumour measurements increased over time</p>
<p>1D 或 2D 肿瘤测量值随时间增加</p></li>
</ul>
<h4 id="comparing-pet-scans">Comparing PET scans</h4>
<ul>
<li><p>Need to identify and delineate the tumors （segmentation）</p>
<p>需要对肿瘤进行识别和勾画（segmentation）</p></li>
<li><p>Find the corresponding volume in the follow-up scan （registration）</p>
<p>在后续扫描中找到对应的卷（注册）</p></li>
<li><p>For each tumor, derive a single SUV value</p>
<p>对于每个肿瘤，得出一个单一的 SUV 值</p></li>
<li><p>Change compared to an empirical threshold</p>
<p>与经验阈值相比的变化</p>
<ul>
<li><p>Provided in recommendations such as EORTC, PERCIST</p>
<p>在 EORTC、PERCIST 等建议中提供</p></li>
</ul></li>
<li><p>Tedious when there are many tumor sites</p>
<p>肿瘤部位多时繁琐</p>
<ul>
<li><p>For metastatic disease, <u>total metabolic tumour volume</u> is increasingly used</p>
<p>对于转移性疾病，越来越多地使用总代谢肿瘤体积</p></li>
</ul></li>
</ul>
<h4 id="variations-in-suv-measurements">Variations in SUV measurements</h4>
<p>Several statistics are currently being used to perform the measurement of tumour progression over time:</p>
<p>目前正在使用几种统计数据来测量随时间的肿瘤进展</p>
<ul>
<li><p>Limited resolution complicates direct comparison of intensities</p>
<p>有限的分辨率使强度的直接比较变得复杂</p></li>
<li><p>Partial volume effect, localization of lesions, etc.</p>
<p>部分体积效应、病灶定位等。</p></li>
</ul>
<p>Mean of the tumour region 肿瘤区域的平均值</p>
<ul>
<li><p>Tumors are heterogenous, and activity is averaged out over a large volume</p>
<p>肿瘤是异质性的，活性在大体积上是平均的</p></li>
</ul>
<p>Max over the tumour region 肿瘤区域的最大值</p>
<ul>
<li><p>Better, but sensitive to noise</p>
<p>更好，但对噪音敏感</p></li>
<li><p>Inherently, only based on one pixel</p>
<p>本质上，仅基于一个像素</p></li>
</ul>
<p>Peak over the ROI</p>
<ul>
<li><p>Small, fixed-size region of interest ROI<sub>peak</sub></p>
<p>小的、固定大小的感兴趣区域</p></li>
<li><p>ROI cantered at high intensity region （max, variants exist）</p>
<p>ROI 在高强度区域慢跑（最大，存在变体）</p></li>
<li><p>Peak measurement is the average over that ROI</p>
<p>峰值测量值是该 ROI 的平均值</p></li>
</ul>
<p><img src="/images/HMS/2314375.png" /></p>
<h4 id="measuring-bone-cartilage-for-osteoarthritis-测量骨软骨以治疗骨关节炎">Measuring bone cartilage for osteoarthritis 测量骨软骨以治疗骨关节炎</h4>
<ul>
<li>The thickness of a thin layer of bone cartilage 一层薄薄的骨软骨的厚度
<ul>
<li>How to measure and monitor over time? 如何随时间测量和监控</li>
</ul></li>
<li>Image analysis workflow 图像分析工作流程
<ul>
<li>Segmentation of structures 结构分割</li>
<li>Unfold the cartilage surface 展开软骨表面</li>
<li>Perform measurements of 执行测量
<ul>
<li>Thickness</li>
<li>Roughness</li>
</ul></li>
</ul></li>
</ul>
<p>Monitor local thickness and bone remodelling changes over time</p>
<p><img src="/images/HMS/2314376.png" /></p>
<h4 id="metabolic-tumour-volume-pet">Metabolic tumour volume （PET）</h4>
<p><img src="/images/HMS/2314377.png" /></p>
<h3 id="need-for-standardizing-measures">Need for standardizing measures</h3>
<ul>
<li>CT
<ul>
<li>CT Angiography</li>
<li>CT Volumetry</li>
<li>Lung Density</li>
<li>Small Lung Nodule</li>
</ul></li>
<li>MR
<ul>
<li>Arterial Spin Labeling （ASL） – {EIBIR supported}</li>
<li>Dynamic Contrast-Enhanced （DCE） MRI</li>
<li>Dynamic Susceptibility Contrast （DSC） MRI</li>
<li>Diffusion Tensor Imaging （DTI） MRI</li>
<li>Diffusion Weighted Imaging （DWI） MRI</li>
<li>Functional Magnetic Resonance Imaging （fMRI）</li>
<li>Magnetic Resonance Elastography （MRE）</li>
<li>Proton Density Fat Fraction （PDFF）</li>
<li>Musculoskeletal （MSK）</li>
</ul></li>
<li>PET
<ul>
<li>FDG-PET/CT</li>
<li>Quantifying Dopamine Transporters with 123 Iodine Labelled Ioflupane in Neurodegenerative Diseases （I-123）</li>
<li>PET-Amyloid</li>
<li>Technetium-99m for body, oncology and immunology （TC99m）</li>
</ul></li>
<li>US
<ul>
<li>Contrast Enhanced Ultrasound （CEUS）</li>
<li>Pulse-Echo Quantitative Ultrasound （PEQUS）– {AIUM supported}</li>
<li>Ultrasound Shear Wave Speed （SWS）</li>
<li>Ultrasound Volume Blood Flow （VBF） – {AIUM supported}</li>
</ul></li>
</ul>
<h2 id="image-features">Image features</h2>
<h3 id="feature-extraction-general-pipeline">Feature extraction: general pipeline</h3>
<p><img src="/images/HMS/2314378.png" /></p>
<ul>
<li><p>For each region of interest （ROI） or candidate region</p>
<ul>
<li><p>Often defined in space through a segmentation</p>
<p>通常通过分割在空间中定义</p></li>
</ul></li>
<li><p>Extract the quantitative property that characterize the region</p>
<p>提取表征区域的定量属性</p>
<ul>
<li>The appearance in terms of <strong>intensity</strong> and <strong>texture</strong></li>
<li>The appearance in terms of <strong>shape</strong> and <strong>outer surface</strong></li>
<li>The <strong>size</strong> and <strong>volume</strong> of the region</li>
<li>The （relative） <strong>position</strong> of the region</li>
<li>…</li>
</ul></li>
</ul>
<h3 id="intensity-based-features-基于强度的特征">Intensity-based features 基于强度的特征</h3>
<ul>
<li><p>Examples of intensity features over region Ω in image <code>f(x)</code> with N pixels</p>
<ul>
<li><p>Mean intensity value <span class="math display">\[
\mu = \frac{1}{N}\sum_{x∈\Omega}f(x)
\]</span></p></li>
<li><p>Median, minimum and maximum values</p></li>
<li><p>Standard deviation <span class="math display">\[
\sigma=\sqrt{\frac{1}{N}\sum_{x∈\Omega}(f(x)-\mu)^2}
\]</span></p></li>
<li><p>Energy <span class="math display">\[
E=\sum_{x∈\Omega}(f(x)+c)^2
\]</span></p>
<ul>
<li>c is a constant such that all values within the brackets are positive</li>
<li>Energy is related to the content of intensities, irrespective of conventions such as HU</li>
</ul></li>
</ul></li>
<li><p>Derived from the histogram</p>
<ul>
<li><p>Skewness</p>
<ul>
<li>Measure of asymmetry <span class="math display">\[
S_{sk}=\frac{\frac{1}{N}\sum_{x∈\Omega}(f(x)-\mu)^3}{\sigma^3}
\]</span></li>
</ul></li>
<li><p>Kurtosis</p>
<ul>
<li>Measure of relative weight of the tails <span class="math display">\[
S_{ku}=\frac{\frac{1}{N}\sum_{x∈\Omega}(f(x)-\mu)^4}{\sigma^4}
\]</span></li>
</ul></li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314379.png" /></p>
<h3 id="shape">Shape</h3>
<ul>
<li><p>Examples of shape features over region Ω in image <code>f(x)</code> with N pixels</p>
<ul>
<li><p>Volume</p>
<ul>
<li><p><span class="math display">\[
v = \frac{1}{N}\sum_{x∈\Omega}v(x)
\]</span></p></li>
<li><p><span class="math display">\[
v(x)=s_xs_ys_z
\]</span></p>
<p>the volume of a voxel</p></li>
</ul></li>
<li><p>Largest diameter</p>
<ul>
<li><p>D<sub>i</sub> the axes of the enclosing ellipsoid</p></li>
<li><p><span class="math display">\[
D_M=\max[D_1,D_2,D_3]
\]</span></p></li>
</ul></li>
<li><p>Smallest diameter</p>
<ul>
<li><span class="math display">\[
d_m=\min[D_1,D_2,D_3]
\]</span></li>
</ul></li>
<li><p>Elongation</p>
<ul>
<li><span class="math display">\[
\frac{D_M}{D_m}
\]</span></li>
</ul></li>
</ul></li>
<li><p>Examples considering the area A of the surface of the structure</p>
<ul>
<li><p>Relation to the surface of the compassing spheres</p></li>
<li><p>Surface Volume Ratio</p>
<ul>
<li><span class="math display">\[
R_{SV}=\frac{S}{V}
\]</span></li>
</ul></li>
<li><p>Compactness</p>
<ul>
<li><span class="math display">\[
C=\frac{V}{\sqrt{A^3\pi}}
\]</span></li>
</ul></li>
</ul></li>
</ul>
<h3 id="texture">Texture</h3>
<ul>
<li>What is texture?</li>
<li>No consensus on definition of texture in medical imaging literature
<ul>
<li>Associated with “coarse”, “edgy”, “directional”, “repetitive”, “random”, ...</li>
</ul></li>
<li>Oxford dictionary: The feel, appearance, or consistency of a surface or a substance</li>
<li>Haidekker2011: Texture is a <strong>systematic local variation</strong> of the image values</li>
<li>Petrou2011: The most important characteristic of texture is that it is <strong>scale dependent</strong>. Different types of texture are visible at different scales</li>
</ul>
<h4 id="scale">Scale</h4>
<ul>
<li><p>The same pattern at a different scale, leads to a different texture</p>
<p>相同的图案在不同的尺度上，会导致不同的质感</p></li>
<li><p>Given a certain scale,</p></li>
</ul>
<p><img src="/images/HMS/2314380.png" /></p>
<h4 id="direction-orientation">Direction / orientation</h4>
<ul>
<li><p>Textures may be have a certain orientation</p>
<p>纹理可能有一定的方向</p></li>
<li><p>Usually not relevant to measure the direction, but measures should take this into account</p>
<p>通常与测量方向无关，但测量时应考虑到这一点</p></li>
</ul>
<p><img src="/images/HMS/2314381.png" /></p>
<h4 id="gray-level-co-occurrence-matrix-灰度共生矩阵-glcm">Gray-level co-occurrence matrix 灰度共生矩阵 （GLCM）</h4>
<ul>
<li><p>The distribution of the number of occurrences of a pair of gray values and separated by a distance vector Δ = [Δ<sub>x</sub>, Δ<sub>y</sub>]</p>
<p>由距离向量Δ = [Δ<sub>x</sub>, Δ<sub>y</sub>]分隔的一对灰度值的出现次数分布</p></li>
</ul>
<p><span class="math display">\[
C\Delta_x,\Delta_y(i,j)=\sum^n_{x=1}\sum^m_{y=1}\left\{
\begin{aligned}
1 &amp; &amp; if\ I(x,y)=i\ and\ I(x+\Delta x,y+\Delta y)=j \\
0 &amp; &amp;otherwise
\end{aligned}
\right.
\]</span></p>
<p>Co-occurrence Matrix</p>
<ul>
<li>Distance =2</li>
<li>Angle = 0°</li>
</ul>
<p><img src="/images/HMS/2314382.png" /></p>
<p>The matrix leads to 13 Haralick texture descriptors</p>
<p>该矩阵导致 13 个 Haralick 纹理描述符</p>
<ul>
<li>Energy</li>
<li><strong>Entropy</strong></li>
<li>Correlation</li>
<li><strong>Contrast</strong></li>
<li>Inverse Difference Moment</li>
<li>Variance</li>
<li>Sum Mean</li>
<li>Inertia</li>
<li>Cluster Shade</li>
<li>Cluster Prominence</li>
<li>Max Probability</li>
<li>Inverse Variance</li>
<li>Mode Probability</li>
</ul>
<h5 id="second-order-histogram-statistics-二阶直方图统计">Second-Order Histogram Statistics 二阶直方图统计</h5>
<ul>
<li><p>Entropy of the matrix H <span class="math display">\[
S_H=-\sum^{y_t}_{y_q=y_1}\sum^{y_t}_{y_r=y_1}H(y_q,y_r,d)\log10[H(y_q,y_r,d)]
\]</span></p></li>
<li><p>Contrast of matrix H <span class="math display">\[
\sum^{y_t}_{y_q=y_1}\sum^{y_t}_{y_r=y_1}\partial(y_q,y_r)H(y_q,y_r,d)=\sum^{y_t}_{y_q=y_1}\sum^{y_t}_{y_r=y_1}(y_q-y_r)^2H(y_q,y_r,d)
\]</span></p></li>
</ul>
<h5 id="co-occurrence-matrices">Co-occurrence matrices</h5>
<ul>
<li>Global
<ul>
<li>Features extracted are for the entire cube</li>
</ul></li>
<li>13 Directions
<ul>
<li>Four original 2D directions</li>
<li>Nine new 3D directions</li>
</ul></li>
<li>4 Distances
<ul>
<li>1, 2, 4, and 8 pixels</li>
</ul></li>
<li>13 features extracted per distance per direction</li>
<li>13*4*13=676 features per cube</li>
</ul>
<p><img src="/images/HMS/2314383.png" /></p>
<h4 id="example-texture-feature-extraction">Example texture feature extraction</h4>
<p><img src="/images/HMS/2314384.png" /></p>
<h4 id="examples-of-textures">Examples of textures</h4>
<ul>
<li>Different textures of the lung
<ul>
<li>Associated to different pathologies</li>
<li>By locally applying textures operators, such pathologies can be detected and localised （segmented）</li>
</ul></li>
<li>Different texture of the bone
<ul>
<li>For grading severity of osteoporosis</li>
<li>Requires high resolution （μ）CT of the trabecular bone</li>
</ul></li>
</ul>
<h2 id="computer-aided-diagnosis">Computer-aided diagnosis</h2>
<p>Beyond extracting one feature</p>
<ul>
<li><p>Quantitative imaging</p></li>
<li><p>A radiologist will in practice look at several quantities</p>
<ul>
<li>Intensity</li>
<li>Shape</li>
<li>Texture</li>
</ul></li>
<li><p>Why not extract several features, and try to automate diagnosis?</p>
<p><strong>Computer-aided diagnosis （CAD）</strong></p>
<ul>
<li>Detection of structures</li>
<li>Tumor or not?</li>
<li>Benign or malignant?</li>
<li>Good or bad prognosis</li>
</ul></li>
</ul>
<h3 id="machine-learning-cad-approaches">Machine learning CAD approaches</h3>
<p>For each region of interest （ROI） or candidate region</p>
<ul>
<li>Often defined in space through a segmentation</li>
</ul>
<p>Extract <strong>all quantitative properties</strong> that characterize the region</p>
<ul>
<li>The appearance in terms of <strong>intensity</strong> and <strong>texture</strong></li>
<li>The appearance in terms of <strong>shape</strong> and <strong>outer surface</strong></li>
<li>The size and <strong>volume</strong> of the region</li>
<li>The （relative） <strong>position</strong> of the region</li>
<li>…</li>
</ul>
<p>Use training data and a <strong>classifier</strong> to learn and detect the appearance of a pathology</p>
<h3 id="introduction">Introduction</h3>
<ul>
<li>Definition
<ul>
<li>Systems that <strong>assist doctors in the interpretation</strong> of medical signals or images</li>
<li>Systems that process digital signals and images for <u>typical appearances and highlight conspicuous regions</u>; in order to offer input to support a decision taken by the professional</li>
</ul></li>
<li>Sometimes a distinction is made
<ul>
<li>Computer-aided detection （CADe） and computer-aided diagnosis （CADx）</li>
</ul></li>
<li>CAD has many applications in fields such as
<ul>
<li>Radiology</li>
<li>Nuclear medicine</li>
<li>Radiotherapy</li>
<li>Neurology</li>
<li>（Digital pathology）</li>
</ul></li>
</ul>
<h3 id="classification-and-regression">Classification and regression</h3>
<ul>
<li>Features
<ul>
<li>An individual measurable property or characteristic of a phenomenon being observed
<ul>
<li>X[i] or x[i]</li>
</ul></li>
</ul></li>
<li>Feature vector
<ul>
<li>N-dimensional vector of numerical features associated to one object or instance
<ul>
<li>X or x</li>
</ul></li>
</ul></li>
<li>Output or target
<ul>
<li>The desired result of your system for the given instances, the training data set
<ul>
<li>F（x）</li>
</ul></li>
</ul></li>
<li>（Class） Label
<ul>
<li>The output in case of a classification problem, the name characterizing the class
<ul>
<li>Dog, Cancer, Healthy, etc</li>
</ul></li>
</ul></li>
<li>Training data
<ul>
<li>Dataset of examples used for learning, e.g. training a classifier</li>
</ul></li>
<li>Test data
<ul>
<li>Dataset of examples used for testing the performance of the method
<ul>
<li>Independent of the training dataset!</li>
<li>From the same phenomena, has the same characteristics</li>
</ul></li>
</ul></li>
</ul>
<h4 id="classification">Classification</h4>
<ul>
<li>Given a set of data points and labels
<ul>
<li>Training set including the labels of classes C<sub>1</sub> or C<sub>2</sub></li>
<li>Train the classifier to lean the boundary</li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314385.png" /></p>
<ul>
<li>Given a new data point
<ul>
<li>Apply the classifier to determine C<sub>1</sub> or C<sub>2</sub></li>
</ul></li>
</ul>
<h4 id="pipeline">Pipeline</h4>
<p><img src="/images/HMS/2314386.png" /></p>
<p>Candidate detection</p>
<ul>
<li>Some way of detecting abnormalities</li>
<li>Randomly or evenly sample your image</li>
</ul>
<h3 id="example">Example</h3>
<h4 id="cad-of-lung-nodules-using-machine-learning">CAD of lung nodules using machine learning</h4>
<p><img src="/images/HMS/2314387.png" /></p>
<h5 id="pre-processing">Pre-processing</h5>
<ul>
<li>Resampling images to standard size
<ul>
<li>Isotropic resampling</li>
<li>E.g. 1 mm<sup>3</sup> voxel size</li>
</ul></li>
<li>Lung segmentation
<ul>
<li>Thresholding to extract low-intensity regions</li>
<li>Morphological operations to include holes</li>
</ul></li>
</ul>
<h5 id="candidate-detection">Candidate detection</h5>
<ul>
<li>Extract the nodule clusters</li>
<li>Divergence of normalized gradient
<ul>
<li>Retain points above empirically-determined thresholds</li>
</ul></li>
<li>Divergence points are compared with nodule clusters
<ul>
<li>Only clusters that correspond with divergence points are retained</li>
</ul></li>
</ul>
<h5 id="feature-extraction">Feature extraction</h5>
<ul>
<li>Shape</li>
<li>Volume</li>
<li>Elongation factor</li>
<li>Compactness</li>
<li>Lung wall separation</li>
<li>Mean value</li>
<li><strong>Mean value over divergence values</strong></li>
<li><strong>Mean value over nodule &amp; vessel enhanced images</strong></li>
<li>Median value</li>
<li>Maximum and minimum values</li>
<li>standard deviation</li>
</ul>
<h5 id="data-balancing">Data balancing</h5>
<ul>
<li><p>After candidate detection over the training set</p>
<ul>
<li><p>111332 candidates detected</p></li>
<li><p>~ 500 true nodules （true positives）</p>
<p><strong>Too many negatives, a typical problem in CAD</strong></p></li>
</ul></li>
<li><p>Data balancing methods</p>
<ul>
<li>Downsample your negatives
<ul>
<li>Select representative samples, e.g. SOM</li>
</ul></li>
<li>Upsample your positives
<ul>
<li>Generate similar / plausible samples in feature space</li>
</ul></li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314388.png" /></p>
<h5 id="classification-1">Classification</h5>
<ul>
<li>Support vector machine
<ul>
<li>Using a radial basis function （RBF）</li>
<li>Using a soft margin</li>
</ul></li>
<li>Tuning the parameters
<ul>
<li>10-fold cross-validation on the training set</li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314389.png" /></p>
<h2 id="radiomics"><a target="_blank" rel="noopener" href="https://www.youtube.com/watch?v=Tq980GEVP0Y">Radiomics</a></h2>
<h3 id="introduction-1">Introduction</h3>
<ul>
<li>Definition
<ul>
<li><p>Radiomics is a process for <strong>high-throughput extraction</strong> of <strong>quantitative features</strong> that result in the conversion of images into <strong>mineable data</strong>, and the subsequent analysis of these data for <strong>decision support</strong>.</p>
<p>Radiomics为处理<strong>高通量提取</strong>的<strong>定量特征</strong>其导致图像转化成<strong>可采数据</strong>，并且这些数据的随后的分析<strong>的决策支持</strong></p></li>
</ul></li>
<li>Process
<ul>
<li><p>Extract a lot of （quantitative） features 提取大量（定量）特征</p></li>
<li><p>Try to make <strong>predictions</strong> on the future outcome of the treatment</p>
<p>尝试对治疗的未来结果做出<strong>预测</strong></p></li>
</ul></li>
</ul>
<h3 id="pipeline-1">Pipeline</h3>
<p><img src="/images/HMS/2314390.png" /></p>
<ul>
<li>Image acquisition 图像采集</li>
<li>Volume of interest identification and segmentation 兴趣量识别和分割</li>
<li>Feature extraction and qualification 特征提取和限定</li>
<li>Learning algorithms 学习算法</li>
</ul>
<p><img src="/images/HMS/2314391.png" /></p>
<ol type="1">
<li>Example CT images of lung cancer patients （left）, 3D visualizations right. Note strong phenotypic differences that can be captured with routine CT imaging, such as heterogeneity and shape.</li>
<li>Process for extracting radiomics data from images.
<ol type="1">
<li>Manual/automated segmentation</li>
<li>Features are extracted from within the defined tumour contours, quantifying tumour intensity, shape, texture and wavelet texture.</li>
<li>Analysis of radiomics features through comparison with clinical data, gene-expression data….</li>
</ol></li>
</ol>
<h3 id="what-type-of-predictions">What type of predictions?</h3>
<p><strong>Beyond diagnosis and staging of the pathology</strong></p>
<ul>
<li>Based on the tumor
<ul>
<li>Intensity</li>
<li>Shape</li>
<li>Texture</li>
<li>Frequency content</li>
</ul></li>
<li>Predict the patient survival
<ul>
<li>Will the patient be alive or not in X years?</li>
</ul></li>
<li>Predict the response to a certain therapy
<ul>
<li>Will the patient respond （PR or CR） or not?</li>
</ul></li>
</ul>
<h4 id="example-1">Example</h4>
<h5 id="prediction-based-on-radiomics">Prediction based on radiomics</h5>
<p>For one feature （energy）</p>
<ul>
<li>Take the median for dataset Lung1</li>
<li>Plot the survival curves for values &gt;median and &lt; median for all datasets</li>
</ul>
<p><img src="/images/HMS/2314392.png" /></p>
<p>The curves are clearly different, demonstrating the <strong>prognostic power of the features for the population</strong></p>
<p>Deriving reliable and useful decision support for the individual remains the challenge</p>
<h4 id="a-recent-headline">A recent headline</h4>
<p>Google’s new AI algorithm predicts heart disease by looking at your eyes</p>
<h3 id="the-future">The future?</h3>
<p>Towards Automated analysis of images</p>
<ul>
<li>Yes, without a doubt</li>
<li>But how will we integrate it in the workflow?
<ul>
<li>Only AI reading the images?</li>
<li>Second reading of the computer?
<ul>
<li>Helping the doctors</li>
<li>Checking the doctors</li>
</ul></li>
<li>Running additional readings in the background</li>
</ul></li>
</ul>
<h1 id="summary">Summary</h1>
<h2 id="quantitative-imaging-2">Quantitative Imaging</h2>
<ul>
<li><p>The extraction of quantifiable features from medical images for the assessment of normal or the severity, degree of change, or status of a disease, injury, or chronic condition relative to normal</p>
<p>从医学图像中提取可量化的特征，以评估正常情况或疾病、损伤或慢性病相对于正常情况的严重程度、变化程度或状态</p></li>
<li><p>In this context, quantitative implies reproducible, comparable, replicable</p>
<p>在这种情况下，定量意味着可重复、可比较、可复制</p></li>
<li><p>Obtaining quantitative measures puts requirements on the image acquisition, and the image analysis / processing</p>
<p>获得定量测量对图像采集和图像分析/处理提出了要求</p></li>
<li><p>In CT, quantitative measures can be obtained on intensity and morphology</p>
<p>在 CT 中，可以获得强度和形态的定量测量</p></li>
<li><p>In MRI, quantitative measures can be obtained on morphology. Intensities can not be directly compared. Specific MRI sequences or analyses, lead to physical quantities, which can be compared</p>
<p>在 MRI 中，可以获得对形态学的定量测量。强度不能直接比较。特定的 MRI 序列或分析，导致可以比较的物理量</p></li>
<li><p>In PET, quantitative measurements of intensity can be obtained using standardized uptake values （SUV）</p>
<p>在 PET 中，可以使用标准化摄取值（SUV）获得强度的定量测量</p></li>
<li><p>To reliably extract quantitative measures from images, there is a need for standardization and clear criteria</p>
<p>为了从图像中可靠地提取定量测量，需要标准化和明确的标准</p>
<ul>
<li><p>QIBA alliance for standard measurement of imaging biomarkers</p>
<p>QIBA 成像生物标志物标准测量联盟</p></li>
<li><p>RECIST or WHO response criteria in oncology</p>
<p>RECIST 或 WHO 肿瘤学反应标准</p></li>
</ul></li>
</ul>
<h2 id="feature-extraction-1">Feature extraction</h2>
<ul>
<li>The general pipeline involves
<ul>
<li>the acquisition of images</li>
<li>Pre-processing and segmentation of the region of interest</li>
<li>Extraction of the quantitative measure</li>
<li>Diagnostic assessment based on the measure</li>
</ul></li>
<li>Intensity features cover
<ul>
<li>The mean and standard deviation</li>
<li>The median</li>
<li>The energy</li>
<li>The skewness and kurtosis</li>
<li>…</li>
</ul></li>
<li>Shape features cover
<ul>
<li>The volume</li>
<li>The elongation</li>
<li>The compactness</li>
<li>…</li>
</ul></li>
<li>Another group of features is related to texture （not covered in depth here）</li>
</ul>
<h2 id="computer-aided-diagnosis-1">Computer-aided diagnosis</h2>
<ul>
<li>The general pipeline involves
<ul>
<li>The acquisition of images</li>
<li>Pre-processing</li>
<li>Detection of abnormalities and segmentation</li>
<li>Extraction of many quantitative image features</li>
<li>Classification of the abnormality</li>
</ul></li>
<li>Classification is a supervised machine learning technique
<ul>
<li>Based on a training dataset, learn an optimal decision boundary between classes</li>
<li>Given a new data point, decide to which class the point belongs</li>
</ul></li>
<li>Example of a full pipeline for a CAD problem for lung nodules
<ul>
<li>Pre-processing: images are resampled, lungs are segmented</li>
<li>Candidate detection: detect suspicious structures</li>
<li>Reduce the amount of false positives using feature extraction and classification
<ul>
<li>First train a classifier based on the features of a training dataset</li>
<li>Then apply the classifier based on the feature of the input structure</li>
</ul></li>
</ul></li>
</ul>
<h2 id="radiomics-1">Radiomics</h2>
<ul>
<li>The general pipeline involves
<ul>
<li>The acquisition of images</li>
<li>Segmentation of tumours</li>
<li>Extraction of many quantitative image features</li>
<li>Classification based on clinical endpoints （survival, gene expression, etc.）</li>
</ul></li>
<li>Radiomic signatures are extracted from the tumor
<ul>
<li>Including intensity statistics</li>
<li>Shape features</li>
<li>Texture features</li>
<li>Wavelet features</li>
</ul></li>
<li>Signatures were shown to correlated with the outcome of the patients （survival）</li>
</ul>
<h2 id="question">Question</h2>
<p>Describe how to obtain quantitative measurements from the following modalities. Discuss how each can be used to obtain morphological and intensity measurements</p>
<ol type="1">
<li><p>Computed tomography</p>
<ul>
<li><p><strong>Morphology</strong>:</p>
<p>CT has excellent geometric accuracy 具有出色的几何精度</p>
<ul>
<li><p>Low geometric distortion 低几何失真</p>
<ul>
<li><p>Distances, angles and straight lines are well preserved</p>
<p>距离、角度和直线都保存完好</p></li>
</ul></li>
<li><p>High resolution 高分辨率</p>
<ul>
<li><p>Sub-millimetre resolution in all planes is possible</p>
<p>所有平面的亚毫米分辨率是可能的</p></li>
</ul></li>
<li><p>Morphological assessments are, to a certain extent, reproducible, comparable, replicable</p>
<p>在某种程度上，形态学评估是可重复的、可比较的、可复制的</p></li>
</ul></li>
<li><p><strong>Intensity</strong>:</p>
<p>Image intensity in CT is standardized because of the use of CT or Hounsfield unit</p>
<p>由于使用 CT 或 Hounsfield 单位，CT 中的图像强度是标准化的</p>
<ul>
<li><p>HU value is linearly related to attenuation 𝜇 within that volume （~electron density）</p>
<p>HU 值与该体积内的衰减𝜇 线性相关（~电子密度） <span class="math display">\[
CT(x,y)=1000\frac{\mu(x,y)-\mu_{water}}{\mu_{water}}
\]</span></p></li>
<li><p>Intensity measurements in CT are therefore are, to a certain extent, reproducible, comparable, replicable</p>
<p>因此，CT 中的强度测量在某种程度上是可重复的、可比较的、可复制的</p></li>
<li><p>Quantification of intensities can be further improved by standardizing the protocol （energy （kV）, exposure （mA）, reconstruction）</p>
<p>通过标准化协议（能量（kV）、暴露（mA）、重建）可以进一步提高强度的量化</p></li>
</ul></li>
</ul></li>
<li><p>Magnetic resonance</p>
<ul>
<li><p><strong>Morphology</strong>:</p>
<p>Geometric accuracy is usually lower than CT, depending on protocol</p>
<p>几何精度通常低于 CT，具体取决于协议</p>
<ul>
<li><p>Lower resolution, depending on the site and protocol</p>
<p>较低的分辨率，取决于站点和协议</p></li>
<li><p>Some sequences exhibit distortion （DWI）</p>
<p>一些序列表现出失真（DWI）</p></li>
</ul>
<p>Can still be used for morphological assessments, given a suitable tolerance</p>
<p>仍可用于形态学评估，给定合适的耐受性</p></li>
<li><p><strong>Intensities</strong>:</p>
<p>Image intensities in MRI are relative</p>
<p>MRI 中的图像强度是相对的</p>
<ul>
<li><p>MRI is in essence a contrast map MRI本质上是一张对比图</p></li>
<li><p>No direct physical meaning of the image intensity in standard MRI</p>
<p>标准 MRI 中图像强度没有直接的物理意义</p></li>
</ul></li>
<li><p>Quantitative intensity measurements can be derived by extracting physical quantities using advanced MRI sequences.</p>
<p>可以通过使用高级 MRI 序列提取物理量来导出定量强度测量值。</p>
<p>Examples:</p>
<ul>
<li><p>Perfusion measurements from dynamic contrast-enhanced （DCE） or perfusion weighted imaging (PWI)</p>
<p>来自动态对比增强（DCE）或灌注加权成像（PWI）的灌注测量</p>
<ul>
<li>Blood flow, blood volume, permeability, ...</li>
</ul></li>
<li><p>Diffusion measurements from diffusion-weighted imaging （DWI）</p>
<p>弥散加权成像（DWI）的弥散测量</p></li>
<li><p>Apparent diffusion coefficient （ADC）</p>
<p>表观扩散系数（ADC）</p></li>
</ul></li>
</ul></li>
</ol>

                
            </div>
            <hr/>

            

    <div class="reprint" id="reprint-statement">
        
            <div class="reprint__author">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-user">
                        Author:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="/about" rel="external nofollow noreferrer">Chaoqun Yin</a>
                </span>
            </div>
            <div class="reprint__type">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-link">
                        Link:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="https://yintelligence.tech/2021/04/26/2021-04-26-Quantitative-imaging/">https://yintelligence.tech/2021/04/26/2021-04-26-Quantitative-imaging/</a>
                </span>
            </div>
            <div class="reprint__notice">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-copyright">
                        Reprint policy:
                    </i>
                </span>
                <span class="reprint-info">
                    All articles in this blog are used except for special statements
                    <a href="https://creativecommons.org/licenses/by/4.0/deed.zh" rel="external nofollow noreferrer" target="_blank">CC BY 4.0</a>
                    reprint polocy. If reproduced, please indicate source
                    <a href="/about" target="_blank">Chaoqun Yin</a>
                    !
                </span>
            </div>
        
    </div>

    <script async defer>
      document.addEventListener("copy", function (e) {
        let toastHTML = '<span>Copied successfully, please follow the reprint policy of this article</span><button class="btn-flat toast-action" onclick="navToReprintStatement()" style="font-size: smaller">more</a>';
        M.toast({html: toastHTML})
      });

      function navToReprintStatement() {
        $("html, body").animate({scrollTop: $("#reprint-statement").offset().top - 80}, 800);
      }
    </script>



            <div class="tag_share" style="display: block;">
                <div class="post-meta__tag-list" style="display: inline-block;">
                    
                        <div class="article-tag">
                            
                                <a href="/tags/Biomedical-Signals-and-Images/">
                                    <span class="chip bg-color">Biomedical Signals and Images</span>
                                </a>
                            
                        </div>
                    
                </div>
                <div class="post_share" style="zoom: 80%; width: fit-content; display: inline-block; float: right; margin: -0.15rem 0;">
                    <link rel="stylesheet" type="text/css" href="/libs/share/css/share.min.css">
<div id="article-share">

    
    <div class="social-share" data-sites="twitter,facebook,google,qq,qzone,wechat,weibo,douban,linkedin" data-wechat-qrcode-helper="<p>微信扫一扫即可分享！</p>"></div>
    <script src="/libs/share/js/social-share.min.js"></script>
    

    

</div>

                </div>
            </div>
            
        </div>
    </div>

    

    

    

    

    

    

    

    

<article id="prenext-posts" class="prev-next articles">
    <div class="row article-row">
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge left-badge text-color">
                <i class="fas fa-chevron-left"></i>&nbsp;Previous</div>
            <div class="card">
                <a href="/2021/04/27/2021-04-27-HTML/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/3.jpg" class="responsive-img" alt="HTML">
                        
                        <span class="card-title">HTML</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            Exercise 3
                        
                    </div>
                    <div class="publish-info">
                        <span class="publish-date">
                            <i class="far fa-clock fa-fw icon-date"></i>2021-04-27
                        </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/Network/" class="post-category">
                                    Network
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Web-Technologies/">
                        <span class="chip bg-color">Web Technologies</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge right-badge text-color">
                Next&nbsp;<i class="fas fa-chevron-right"></i>
            </div>
            <div class="card">
                <a href="/2021/04/26/2021-04-26-HTML5-and-the-Open-Web-Platform/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/9.jpg" class="responsive-img" alt="HTML5 and the Open Web Platform">
                        
                        <span class="card-title">HTML5 and the Open Web Platform</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            
                        
                    </div>
                    <div class="publish-info">
                            <span class="publish-date">
                                <i class="far fa-clock fa-fw icon-date"></i>2021-04-26
                            </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/Network/" class="post-category">
                                    Network
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Web-Technologies/">
                        <span class="chip bg-color">Web Technologies</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
    </div>
</article>

</div>


<script>
    $('#articleContent').on('copy', function (e) {
        // IE8 or earlier browser is 'undefined'
        if (typeof window.getSelection === 'undefined') return;

        var selection = window.getSelection();
        // if the selection is short let's not annoy our users.
        if (('' + selection).length < Number.parseInt('120')) {
            return;
        }

        // create a div outside of the visible area and fill it with the selected text.
        var bodyElement = document.getElementsByTagName('body')[0];
        var newdiv = document.createElement('div');
        newdiv.style.position = 'absolute';
        newdiv.style.left = '-99999px';
        bodyElement.appendChild(newdiv);
        newdiv.appendChild(selection.getRangeAt(0).cloneContents());

        // we need a <pre> tag workaround.
        // otherwise the text inside "pre" loses all the line breaks!
        if (selection.getRangeAt(0).commonAncestorContainer.nodeName === 'PRE') {
            newdiv.innerHTML = "<pre>" + newdiv.innerHTML + "</pre>";
        }

        var url = document.location.href;
        newdiv.innerHTML += '<br />'
            + 'From: Chaoqun Yin<br />'
            + 'Author: Chaoqun Yin<br />'
            + 'Link: <a href="' + url + '">' + url + '</a><br />'
            + '本文章著作权归作者所有，任何形式的转载都请注明出处。';

        selection.selectAllChildren(newdiv);
        window.setTimeout(function () {bodyElement.removeChild(newdiv);}, 200);
    });
</script>


<!-- 代码块功能依赖 -->
<script type="text/javascript" src="/libs/codeBlock/codeBlockFuction.js"></script>

<!-- 代码语言 -->

<script type="text/javascript" src="/libs/codeBlock/codeLang.js"></script>


<!-- 代码块复制 -->

<script type="text/javascript" src="/libs/codeBlock/codeCopy.js"></script>


<!-- 代码块收缩 -->

<script type="text/javascript" src="/libs/codeBlock/codeShrink.js"></script>


    </div>
    <div id="toc-aside" class="expanded col l3 hide-on-med-and-down">
        <div class="toc-widget card" style="background-color: white;">
            <div class="toc-title"><i class="far fa-list-alt"></i>&nbsp;&nbsp;TOC</div>
            <div id="toc-content"></div>
        </div>
    </div>
</div>

<!-- TOC 悬浮按钮. -->

<div id="floating-toc-btn" class="hide-on-med-and-down">
    <a class="btn-floating btn-large bg-color">
        <i class="fas fa-list-ul"></i>
    </a>
</div>


<script src="/libs/tocbot/tocbot.min.js"></script>
<script>
    $(function () {
        tocbot.init({
            tocSelector: '#toc-content',
            contentSelector: '#articleContent',
            headingsOffset: -($(window).height() * 0.4 - 45),
            collapseDepth: Number('0'),
            headingSelector: 'h1, h2, h3, h4'
        });

        // modify the toc link href to support Chinese.
        let i = 0;
        let tocHeading = 'toc-heading-';
        $('#toc-content a').each(function () {
            $(this).attr('href', '#' + tocHeading + (++i));
        });

        // modify the heading title id to support Chinese.
        i = 0;
        $('#articleContent').children('h1, h2, h3, h4').each(function () {
            $(this).attr('id', tocHeading + (++i));
        });

        // Set scroll toc fixed.
        let tocHeight = parseInt($(window).height() * 0.4 - 64);
        let $tocWidget = $('.toc-widget');
        $(window).scroll(function () {
            let scroll = $(window).scrollTop();
            /* add post toc fixed. */
            if (scroll > tocHeight) {
                $tocWidget.addClass('toc-fixed');
            } else {
                $tocWidget.removeClass('toc-fixed');
            }
        });

        
        /* 修复文章卡片 div 的宽度. */
        let fixPostCardWidth = function (srcId, targetId) {
            let srcDiv = $('#' + srcId);
            if (srcDiv.length === 0) {
                return;
            }

            let w = srcDiv.width();
            if (w >= 450) {
                w = w + 21;
            } else if (w >= 350 && w < 450) {
                w = w + 18;
            } else if (w >= 300 && w < 350) {
                w = w + 16;
            } else {
                w = w + 14;
            }
            $('#' + targetId).width(w);
        };

        // 切换TOC目录展开收缩的相关操作.
        const expandedClass = 'expanded';
        let $tocAside = $('#toc-aside');
        let $mainContent = $('#main-content');
        $('#floating-toc-btn .btn-floating').click(function () {
            if ($tocAside.hasClass(expandedClass)) {
                $tocAside.removeClass(expandedClass).hide();
                $mainContent.removeClass('l9');
            } else {
                $tocAside.addClass(expandedClass).show();
                $mainContent.addClass('l9');
            }
            fixPostCardWidth('artDetail', 'prenext-posts');
        });
        
    });
</script>

    

</main>


<script src="https://cdn.bootcss.com/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<script>
    MathJax.Hub.Config({
        tex2jax: {inlineMath: [['$', '$'], ['\(', '\)']]}
    });
</script>



    <footer class="page-footer bg-color">
    
    <div class="container row center-align" style="margin-bottom: 15px !important;">
        <div class="col s12 m8 l8 copy-right">
            Copyright&nbsp;&copy;
            
                <span id="year">2021</span>
            
            <span id="year">2021</span>
            <a href="/about" target="_blank">Chaoqun Yin</a>
            |&nbsp;Powered by&nbsp;<a href="https://hexo.io/" target="_blank">Hexo</a>
            |&nbsp;Theme&nbsp;<a href="https://github.com/blinkfox/hexo-theme-matery" target="_blank">Matery</a>
            <br>
            
            
            
            
            
            
            <span id="busuanzi_container_site_pv">
                |&nbsp;<i class="far fa-eye"></i>&nbsp;总访问量:&nbsp;<span id="busuanzi_value_site_pv"
                    class="white-color"></span>&nbsp;次
            </span>
            
            
            <span id="busuanzi_container_site_uv">
                |&nbsp;<i class="fas fa-users"></i>&nbsp;总访问人数:&nbsp;<span id="busuanzi_value_site_uv"
                    class="white-color"></span>&nbsp;人
            </span>
            
            <br>
            
            <span id="sitetime">载入运行时间...</span>
            <script>
                function siteTime() {
                    var seconds = 1000;
                    var minutes = seconds * 60;
                    var hours = minutes * 60;
                    var days = hours * 24;
                    var years = days * 365;
                    var today = new Date();
                    var startYear = "2021";
                    var startMonth = "1";
                    var startDate = "28";
                    var startHour = "2";
                    var startMinute = "1";
                    var startSecond = "0";
                    var todayYear = today.getFullYear();
                    var todayMonth = today.getMonth() + 1;
                    var todayDate = today.getDate();
                    var todayHour = today.getHours();
                    var todayMinute = today.getMinutes();
                    var todaySecond = today.getSeconds();
                    var t1 = Date.UTC(startYear, startMonth, startDate, startHour, startMinute, startSecond);
                    var t2 = Date.UTC(todayYear, todayMonth, todayDate, todayHour, todayMinute, todaySecond);
                    var diff = t2 - t1;
                    var diffYears = Math.floor(diff / years);
                    var diffDays = Math.floor((diff / days) - diffYears * 365);
                    var diffHours = Math.floor((diff - (diffYears * 365 + diffDays) * days) / hours);
                    var diffMinutes = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours) /
                        minutes);
                    var diffSeconds = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours -
                        diffMinutes * minutes) / seconds);
                    if (startYear == todayYear) {
                        document.getElementById("year").innerHTML = todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已安全运行 " + diffDays + " 天 " + diffHours +
                            " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    } else {
                        document.getElementById("year").innerHTML = startYear + " - " + todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已安全运行 " + diffYears + " 年 " + diffDays +
                            " 天 " + diffHours + " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    }
                }
                setInterval(siteTime, 1000);
            </script>
            
            <br>
            
        </div>
        <div class="col s12 m4 l4 social-link social-statis">
    <a href="https://github.com/yintelligence" class="tooltipped" target="_blank" data-tooltip="访问我的GitHub" data-position="top" data-delay="50">
        <i class="fab fa-github"></i>
    </a>



    <a href="mailto:Yintelligence@outlook.com" class="tooltipped" target="_blank" data-tooltip="邮件联系我" data-position="top" data-delay="50">
        <i class="fas fa-envelope-open"></i>
    </a>













</div>
    </div>
</footer>

<div class="progress-bar"></div>


    <!-- 搜索遮罩框 -->
<div id="searchModal" class="modal">
    <div class="modal-content">
        <div class="search-header">
            <span class="title"><i class="fas fa-search"></i>&nbsp;&nbsp;Search</span>
            <input type="search" id="searchInput" name="s" placeholder="Please enter a search keyword"
                   class="search-input">
        </div>
        <div id="searchResult"></div>
    </div>
</div>

<script type="text/javascript">
$(function () {
    var searchFunc = function (path, search_id, content_id) {
        'use strict';
        $.ajax({
            url: path,
            dataType: "xml",
            success: function (xmlResponse) {
                // get the contents from search data
                var datas = $("entry", xmlResponse).map(function () {
                    return {
                        title: $("title", this).text(),
                        content: $("content", this).text(),
                        url: $("url", this).text()
                    };
                }).get();
                var $input = document.getElementById(search_id);
                var $resultContent = document.getElementById(content_id);
                $input.addEventListener('input', function () {
                    var str = '<ul class=\"search-result-list\">';
                    var keywords = this.value.trim().toLowerCase().split(/[\s\-]+/);
                    $resultContent.innerHTML = "";
                    if (this.value.trim().length <= 0) {
                        return;
                    }
                    // perform local searching
                    datas.forEach(function (data) {
                        var isMatch = true;
                        var data_title = data.title.trim().toLowerCase();
                        var data_content = data.content.trim().replace(/<[^>]+>/g, "").toLowerCase();
                        var data_url = data.url;
                        data_url = data_url.indexOf('/') === 0 ? data.url : '/' + data_url;
                        var index_title = -1;
                        var index_content = -1;
                        var first_occur = -1;
                        // only match artiles with not empty titles and contents
                        if (data_title !== '' && data_content !== '') {
                            keywords.forEach(function (keyword, i) {
                                index_title = data_title.indexOf(keyword);
                                index_content = data_content.indexOf(keyword);
                                if (index_title < 0 && index_content < 0) {
                                    isMatch = false;
                                } else {
                                    if (index_content < 0) {
                                        index_content = 0;
                                    }
                                    if (i === 0) {
                                        first_occur = index_content;
                                    }
                                }
                            });
                        }
                        // show search results
                        if (isMatch) {
                            str += "<li><a href='" + data_url + "' class='search-result-title'>" + data_title + "</a>";
                            var content = data.content.trim().replace(/<[^>]+>/g, "");
                            if (first_occur >= 0) {
                                // cut out 100 characters
                                var start = first_occur - 20;
                                var end = first_occur + 80;
                                if (start < 0) {
                                    start = 0;
                                }
                                if (start === 0) {
                                    end = 100;
                                }
                                if (end > content.length) {
                                    end = content.length;
                                }
                                var match_content = content.substr(start, end);
                                // highlight all keywords
                                keywords.forEach(function (keyword) {
                                    var regS = new RegExp(keyword, "gi");
                                    match_content = match_content.replace(regS, "<em class=\"search-keyword\">" + keyword + "</em>");
                                });

                                str += "<p class=\"search-result\">" + match_content + "...</p>"
                            }
                            str += "</li>";
                        }
                    });
                    str += "</ul>";
                    $resultContent.innerHTML = str;
                });
            }
        });
    };

    searchFunc('/search.xml', 'searchInput', 'searchResult');
});
</script>

    <!-- 回到顶部按钮 -->
<div id="backTop" class="top-scroll">
    <a class="btn-floating btn-large waves-effect waves-light" href="#!">
        <i class="fas fa-arrow-up"></i>
    </a>
</div>


    <script src="/libs/materialize/materialize.min.js"></script>
    <script src="/libs/masonry/masonry.pkgd.min.js"></script>
    <script src="/libs/aos/aos.js"></script>
    <script src="/libs/scrollprogress/scrollProgress.min.js"></script>
    <script src="/libs/lightGallery/js/lightgallery-all.min.js"></script>
    <script src="/js/matery.js"></script>

    <!-- Baidu Analytics -->

    <!-- Baidu Push -->

<script>
    (function () {
        var bp = document.createElement('script');
        var curProtocol = window.location.protocol.split(':')[0];
        if (curProtocol === 'https') {
            bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
        } else {
            bp.src = 'http://push.zhanzhang.baidu.com/push.js';
        }
        var s = document.getElementsByTagName("script")[0];
        s.parentNode.insertBefore(bp, s);
    })();
</script>

    
    <script src="/libs/others/clicklove.js" async="async"></script>
    
    
    <script async src="/libs/others/busuanzi.pure.mini.js"></script>
    

    

    

	
    

    

    

    
    <script src="/libs/instantpage/instantpage.js" type="module"></script>
    

</body>

</html>
