<!DOCTYPE HTML>
<html lang="en">


<head>
    <meta charset="utf-8">
    <meta name="keywords" content="Quantitative imaging, Chaoqun Yin">
    <meta name="description" content="Veni, Vidi, Vici">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no">
    <meta name="renderer" content="webkit|ie-stand|ie-comp">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="format-detection" content="telephone=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    <!-- Global site tag (gtag.js) - Google Analytics -->


    <title>Quantitative imaging | Chaoqun Yin</title>
    <link rel="icon" type="image/png" href="/favicon.png">

    <link rel="stylesheet" type="text/css" href="/libs/awesome/css/all.css">
    <link rel="stylesheet" type="text/css" href="/libs/materialize/materialize.min.css">
    <link rel="stylesheet" type="text/css" href="/libs/aos/aos.css">
    <link rel="stylesheet" type="text/css" href="/libs/animate/animate.min.css">
    <link rel="stylesheet" type="text/css" href="/libs/lightGallery/css/lightgallery.min.css">
    <link rel="stylesheet" type="text/css" href="/css/matery.css">
    <link rel="stylesheet" type="text/css" href="/css/my.css">

    <script src="/libs/jquery/jquery.min.js"></script>

<meta name="generator" content="Hexo 5.3.0"></head>




<body>
    <header class="navbar-fixed">
    <nav id="headNav" class="bg-color nav-transparent">
        <div id="navContainer" class="nav-wrapper container">
            <div class="brand-logo">
                <a href="/" class="waves-effect waves-light">
                    
                    <img src="/medias/logo.png" class="logo-img" alt="LOGO">
                    
                    <span class="logo-span">Chaoqun Yin</span>
                </a>
            </div>
            

<a href="#" data-target="mobile-nav" class="sidenav-trigger button-collapse"><i class="fas fa-bars"></i></a>
<ul class="right nav-menu">
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/" class="waves-effect waves-light">
      
      <i class="fas fa-home" style="zoom: 0.6;"></i>
      
      <span>Index</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/tags" class="waves-effect waves-light">
      
      <i class="fas fa-tags" style="zoom: 0.6;"></i>
      
      <span>Tags</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/categories" class="waves-effect waves-light">
      
      <i class="fas fa-bookmark" style="zoom: 0.6;"></i>
      
      <span>Categories</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/archives" class="waves-effect waves-light">
      
      <i class="fas fa-archive" style="zoom: 0.6;"></i>
      
      <span>Archives</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/about" class="waves-effect waves-light">
      
      <i class="fas fa-user-circle" style="zoom: 0.6;"></i>
      
      <span>About</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/contact" class="waves-effect waves-light">
      
      <i class="fas fa-comments" style="zoom: 0.6;"></i>
      
      <span>Contact</span>
    </a>
    
  </li>
  
  <li class="hide-on-med-and-down nav-item">
    
    <a href="/friends" class="waves-effect waves-light">
      
      <i class="fas fa-address-book" style="zoom: 0.6;"></i>
      
      <span>Friends</span>
    </a>
    
  </li>
  
  <li>
    <a href="#searchModal" class="modal-trigger waves-effect waves-light">
      <i id="searchIcon" class="fas fa-search" title="Search" style="zoom: 0.85;"></i>
    </a>
  </li>
</ul>


<div id="mobile-nav" class="side-nav sidenav">

    <div class="mobile-head bg-color">
        
        <img src="/medias/logo.png" class="logo-img circle responsive-img">
        
        <div class="logo-name">Chaoqun Yin</div>
        <div class="logo-desc">
            
            Veni, Vidi, Vici
            
        </div>
    </div>

    

    <ul class="menu-list mobile-menu-list">
        
        <li class="m-nav-item">
	  
		<a href="/" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-home"></i>
			
			Index
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/tags" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-tags"></i>
			
			Tags
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/categories" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-bookmark"></i>
			
			Categories
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/archives" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-archive"></i>
			
			Archives
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/about" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-user-circle"></i>
			
			About
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/contact" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-comments"></i>
			
			Contact
		</a>
          
        </li>
        
        <li class="m-nav-item">
	  
		<a href="/friends" class="waves-effect waves-light">
			
			    <i class="fa-fw fas fa-address-book"></i>
			
			Friends
		</a>
          
        </li>
        
        
        <li><div class="divider"></div></li>
        <li>
            <a href="https://github.com/blinkfox/hexo-theme-matery" class="waves-effect waves-light" target="_blank">
                <i class="fab fa-github-square fa-fw"></i>Fork Me
            </a>
        </li>
        
    </ul>
</div>


        </div>

        
            <style>
    .nav-transparent .github-corner {
        display: none !important;
    }

    .github-corner {
        position: absolute;
        z-index: 10;
        top: 0;
        right: 0;
        border: 0;
        transform: scale(1.1);
    }

    .github-corner svg {
        color: #0f9d58;
        fill: #fff;
        height: 64px;
        width: 64px;
    }

    .github-corner:hover .octo-arm {
        animation: a 0.56s ease-in-out;
    }

    .github-corner .octo-arm {
        animation: none;
    }

    @keyframes a {
        0%,
        to {
            transform: rotate(0);
        }
        20%,
        60% {
            transform: rotate(-25deg);
        }
        40%,
        80% {
            transform: rotate(10deg);
        }
    }
</style>

<a href="https://github.com/blinkfox/hexo-theme-matery" class="github-corner tooltipped hide-on-med-and-down" target="_blank"
   data-tooltip="Fork Me" data-position="left" data-delay="50">
    <svg viewBox="0 0 250 250" aria-hidden="true">
        <path d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"></path>
        <path d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5,78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5,87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
              fill="currentColor" style="transform-origin: 130px 106px;" class="octo-arm"></path>
        <path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
              fill="currentColor" class="octo-body"></path>
    </svg>
</a>
        
    </nav>

</header>

    



<div class="bg-cover pd-header post-cover" style="background-image: url('/medias/featureimages/15.jpg')">
    <div class="container" style="right: 0px;left: 0px;">
        <div class="row">
            <div class="col s12 m12 l12">
                <div class="brand">
                    <h1 class="description center-align post-title">Quantitative imaging</h1>
                </div>
            </div>
        </div>
    </div>
</div>




<main class="post-container content">

    
    <link rel="stylesheet" href="/libs/tocbot/tocbot.css">
<style>
    #articleContent h1::before,
    #articleContent h2::before,
    #articleContent h3::before,
    #articleContent h4::before,
    #articleContent h5::before,
    #articleContent h6::before {
        display: block;
        content: " ";
        height: 100px;
        margin-top: -100px;
        visibility: hidden;
    }

    #articleContent :focus {
        outline: none;
    }

    .toc-fixed {
        position: fixed;
        top: 64px;
    }

    .toc-widget {
        width: 345px;
        padding-left: 20px;
    }

    .toc-widget .toc-title {
        padding: 35px 0 15px 17px;
        font-size: 1.5rem;
        font-weight: bold;
        line-height: 1.5rem;
    }

    .toc-widget ol {
        padding: 0;
        list-style: none;
    }

    #toc-content {
        padding-bottom: 30px;
        overflow: auto;
    }

    #toc-content ol {
        padding-left: 10px;
    }

    #toc-content ol li {
        padding-left: 10px;
    }

    #toc-content .toc-link:hover {
        color: #42b983;
        font-weight: 700;
        text-decoration: underline;
    }

    #toc-content .toc-link::before {
        background-color: transparent;
        max-height: 25px;

        position: absolute;
        right: 23.5vw;
        display: block;
    }

    #toc-content .is-active-link {
        color: #42b983;
    }

    #floating-toc-btn {
        position: fixed;
        right: 15px;
        bottom: 76px;
        padding-top: 15px;
        margin-bottom: 0;
        z-index: 998;
    }

    #floating-toc-btn .btn-floating {
        width: 48px;
        height: 48px;
    }

    #floating-toc-btn .btn-floating i {
        line-height: 48px;
        font-size: 1.4rem;
    }
</style>
<div class="row">
    <div id="main-content" class="col s12 m12 l9">
        <!-- 文章内容详情 -->
<div id="artDetail">
    <div class="card">
        <div class="card-content article-info">
            <div class="row tag-cate">
                <div class="col s7">
                    
                    <div class="article-tag">
                        
                            <a href="/tags/Biomedical-Signals-and-Images/">
                                <span class="chip bg-color">Biomedical Signals and Images</span>
                            </a>
                        
                    </div>
                    
                </div>
                <div class="col s5 right-align">
                    
                    <div class="post-cate">
                        <i class="fas fa-bookmark fa-fw icon-category"></i>
                        
                            <a href="/categories/Digital-Health/" class="post-category">
                                Digital Health
                            </a>
                        
                    </div>
                    
                </div>
            </div>

            <div class="post-info">
                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-minus fa-fw"></i>Publish Date:&nbsp;&nbsp;
                    2021-04-26
                </div>
                

                
                <div class="post-date info-break-policy">
                    <i class="far fa-calendar-check fa-fw"></i>Update Date:&nbsp;&nbsp;
                    2021-08-01
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-file-word fa-fw"></i>Word Count:&nbsp;&nbsp;
                    3.3k
                </div>
                

                
                <div class="info-break-policy">
                    <i class="far fa-clock fa-fw"></i>Read Times:&nbsp;&nbsp;
                    20 Min
                </div>
                

                
                    <div id="busuanzi_container_page_pv" class="info-break-policy">
                        <i class="far fa-eye fa-fw"></i>Read Count:&nbsp;&nbsp;
                        <span id="busuanzi_value_page_pv"></span>
                    </div>
				
            </div>
        </div>
        <hr class="clearfix">

        
        <!-- 是否加载使用自带的 prismjs. -->
        <link rel="stylesheet" href="/libs/prism/prism.css">
        

        

        <div class="card-content article-card-content">
            <div id="articleContent">
                <h1 id="quantitative-imaging">Quantitative imaging</h1>
<p><a target="_blank" rel="noopener" href="https://canvas.vub.be/courses/18933/files?preview=651177">Slide</a> + <a target="_blank" rel="noopener" href="https://vub.cloud.panopto.eu/Panopto/Pages/Viewer.aspx?id=5f1c0cf4-94ed-485c-947d-ac7b00f255b6">Video</a></p>
<h2 id="quantitative-imaging-1">Quantitative imaging</h2>
<h3 id="used-definitions">Used definitions</h3>
<p><u>The extraction of quantifiable features from medical images</u> for the assessment of normal or the severity, degree of change, or status of a disease, injury, or chronic condition relative to normal</p>
<p>In the clinic, it may cover</p>
<ul>
<li>Anatomical, functional and molecular imaging</li>
<li>Data analyses</li>
<li>Display methods</li>
<li>Reporting structures</li>
</ul>
<h4 id="biomarkers">Biomarkers</h4>
<p><strong>Definition</strong>:</p>
<ul>
<li>Also biological marker</li>
<li>A <strong>measurable indicator</strong> of some biological state or condition. Biomarkers are often measured and evaluated to examine normal biological processes, pathogenic processes, or pharmacologic responses to a therapeutic intervention</li>
</ul>
<p>An imaging biomarker: A biomarker detectable in an image</p>
<h4 id="reproducible-research-recommendations">Reproducible Research Recommendations</h4>
<ul>
<li>Reproducibility
<ul>
<li>A study is reproducible if you can take the original data and the computer code used to analyse the data and reproduce all of the numerical findings from the study. This may initially sound like a trivial task but experience has shown that it’s not always easy to achieve this seemingly minimal standard.</li>
</ul></li>
<li>Replicability
<ul>
<li>This is the act of repeating an entire study, independently of the original investigator without the use of original data （but generally using the same methods）.</li>
</ul></li>
</ul>
<h4 id="other-definitions">Other definitions</h4>
<p>Repeatability, Reproducibility, and Replicability: Tackling the 3R challenge in bio-interface science and engineering</p>
<h3 id="what-does-quantitative-imply">What does quantitative imply?</h3>
<ul>
<li>Reproducible
<ul>
<li>Agreement （up to a tolerance） of <u>repeated measurements on the same patient</u></li>
</ul></li>
<li>Comparable
<ul>
<li>Measures are suited to be <strong>compared between patients</strong></li>
</ul></li>
<li>In practice also, Replicability
<ul>
<li>Being able to <u>independently achieve similar measurements</u> across institutes, with slightly varying set-up, procedure, etc</li>
</ul></li>
</ul>
<h3 id="how-to-obtain-quantitative-measurements">How to obtain quantitative measurements</h3>
<ul>
<li><p>Morphology</p>
<ul>
<li><p>Thickness, length, surface, volume, shape, … of a structure</p>
<blockquote>
<p>The size of the detected object depends on it's intensity</p>
</blockquote></li>
</ul></li>
<li><p>Intensity</p>
<ul>
<li><p>Mean, median, standard deviation, skew, … of a structure</p>
<blockquote>
<p>The intensity parameters depend on the region considered</p>
</blockquote></li>
</ul>
<blockquote>
<p>Main features to measure</p>
</blockquote></li>
<li><p>Texture</p>
<ul>
<li><p>A systematic local variation of the image values – Scale dependent. Different types of texture are visible at different scales</p>
<blockquote>
<p>Is in inherently a variation in the intensity</p>
</blockquote></li>
</ul></li>
<li><p>Frequencies</p>
<ul>
<li><p>Fourier, wavelet or other transforms</p>
<blockquote>
<p>Closely related to texture, and intensity</p>
</blockquote></li>
</ul></li>
</ul>
<h4 id="relation-between-measures">Relation between measures</h4>
<ul>
<li>All measures rely on a segmentation of the region of interest （ROI）
<ul>
<li>Which was performed using a region-based or edge-based technique</li>
<li>All measures are therefore influenced by the quality of the intensities or the intensity differences （edge）…</li>
</ul></li>
<li>There is an overlap between texture and frequency
<ul>
<li>But the are not the same</li>
</ul></li>
<li>Measures are <strong>related</strong>, but do carry <strong>complementary</strong> information</li>
<li>All require the imaging to be performed / adhere to certain standards
<ul>
<li>In order to be quantitative</li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314370.png" /></p>
<h4 id="how-to-obtain-quantitative-measures">How to obtain quantitative measures?</h4>
<ul>
<li>Identify the measures
<ul>
<li>Normal variance</li>
<li>Abnormal values</li>
</ul></li>
<li>Acquire the images
<ul>
<li>Quality assurance</li>
<li>Standardization of protocols</li>
</ul></li>
<li>Extract the measures
<ul>
<li>Reduce observer variability</li>
<li>Algorithm reproducibility</li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314371.png" /></p>
<h3 id="quantitative-measures-in-ct">Quantitative measures in CT</h3>
<ul>
<li><p>Image intensity in CT is standardized</p>
<p>Remember: CT or Hounsfield units</p>
<ul>
<li><p>Linear transformation of the original linear attenuation coefficient w</p></li>
<li><p>HU value is linearly related to attenuation µ within that volume （~electron density） <span class="math display">\[
CT(x, y) =1000 \frac{µ(x, y) − µ_{water}}{µ_{water}}
\]</span></p></li>
</ul></li>
<li><p>Intensity measurements are, to a certain extent,</p></li>
<li><p><strong>reproducible, comparable, replicable</strong></p>
<ul>
<li>Quantification can be improved by standardizing the protocol （energy （kV）, exposure （mA）, reconstruction）</li>
</ul></li>
</ul>
<p>CT has excellent geometric accuracy</p>
<ul>
<li>Low geometric distortion
<ul>
<li>Distances, angles and straight lines are preserved</li>
</ul></li>
<li>High resolution
<ul>
<li>Sub-millimetre resolution in all planes</li>
</ul></li>
</ul>
<p>Morphological assessments are, to a certain extent, <strong>reproducible, comparable, replicable</strong></p>
<h3 id="quantitative-measures-in-mri">Quantitative measures in MRI</h3>
<p>Geometric accuracy is usually lower than CT</p>
<ul>
<li>Lower resolution, depending on the site and protocol</li>
<li>Some sequences exhibit distortion （DWI）</li>
</ul>
<p><strong>Can still be used for morphological assessments, given a suitable tolerance</strong></p>
<p>Image intensities in MRI are relative</p>
<ul>
<li>MRI is in essence a contrast map,</li>
<li>No direct meaning of the image intensity</li>
</ul>
<p><strong>How to obtain quantitative intensity measurements?</strong></p>
<p>Quantitative MRI is often obtained by extracting <strong>physical quantities</strong> using advanced sequences</p>
<ul>
<li>Perfusion measurements from dynamic contrast enhanced （DCE） or perfusion weighted imaging （PWI）
<ul>
<li>Blood flow, blood volume, permeability, …</li>
</ul></li>
<li>Diffusion measurements from diffusion-weighted imaging （DWI）
<ul>
<li>Apparent diffusion coefficient （ADC）</li>
</ul></li>
<li>Physical quantities which, if measured correctly are <strong>reproducible, comparable</strong>
<ul>
<li>Depending on how complex the protocol is, also <strong>replicable</strong></li>
</ul></li>
</ul>
<h3 id="quantitative-measures-in-pet--ct">Quantitative measures in PET （-CT）</h3>
<p>Geometric accuracy is much lower than CT</p>
<ul>
<li>Significantly lower resolution, depending on the acquisition time and protocol</li>
<li>The combination with CT can offer a compromise</li>
<li><strong>Morphological assessments of PET-positive lesions, measured on PET and/or CT</strong></li>
</ul>
<p>Image intensities in PET depend on many things</p>
<ul>
<li>The activity injected</li>
<li>The size of the patient</li>
<li>The time after injection</li>
<li>The metabolic activity of the cells –</li>
<li>…</li>
</ul>
<p><strong>How to obtain quantitative intensity measurements from PET?</strong></p>
<ul>
<li><strong>Measurements related to the metabolic activity the cells</strong></li>
</ul>
<h4 id="standardise-the-measurement-procedure">Standardise the measurement procedure</h4>
<ul>
<li>Always same activity injected</li>
<li>Always same time after injection</li>
<li>Always same account the size of the patient… euh?!?</li>
</ul>
<p>Not a very practical way to work</p>
<ul>
<li>Reproducible</li>
<li>Comparable on a patient basis, not interpatient</li>
<li>Difficult to replicate</li>
</ul>
<h4 id="standardized-uptake-value">Standardized Uptake Value</h4>
<p><span class="math display">\[
SUV(time)=\frac{Tracer\ uptake}{Injected\ Activity\ /\ Weight}
\]</span></p>
<p>Under certain circumstances, <sup>18</sup>FDG SUV correlates with metabolic rate of glucose and/or the number of viable tumor cells</p>
<ul>
<li>SUV is a simplified quantitative measure that can be routinely performed in clinical PET studies</li>
</ul>
<p>Adjusts image intensities for</p>
<ul>
<li>Differences in patient size</li>
<li>Injected activity,</li>
<li>Correct activity to decay at a reference time</li>
</ul>
<p>The resulting intensities correlate with metabolic activity of tumor cells AND are <strong>reproducible</strong>, <strong>comparable</strong> and <strong>replicable</strong> to a certain extent.</p>
<h5 id="suv-units">SUV Units</h5>
<ul>
<li>Based on the following assumptions （approximations）:
<ul>
<li>Water-equivalent tissue</li>
<li>A body weight correction in grams</li>
<li>Decay-correction to the time of injection</li>
<li>Known concentration in consistent units of mCi/ml or MBq/cc</li>
</ul></li>
<li>The SUV is a relative, unit-less quantity
<ul>
<li>It relates the image derived radioactivity concentration to the whole-body concentration of the injected radioactivity</li>
<li>Under standard settings （time, injection, activity, etc ）</li>
</ul></li>
</ul>
<h5 id="variations-in-suv-computations">Variations in SUV computations</h5>
<ul>
<li>Instead of body mass
<ul>
<li>Body surface area or lean body weight</li>
<li>Sometimes denoted with subscripts （SUV<sub>BW</sub>, SUV<sub>BSA</sub>, SUV<sub>LBM</sub>）</li>
</ul></li>
<li>Linear correction for circulating glucose</li>
<li>Designated uptake period
<ul>
<li>Delayed scanning may reduce background physiologic uptake</li>
</ul></li>
</ul>
<h3 id="practical-examples-from-clinical-practice">Practical examples from clinical practice</h3>
<ul>
<li>Role of imaging in oncology
<ul>
<li>Morphological response criteria for solid tumours
<ul>
<li>RECIST</li>
<li>Volumetry</li>
</ul></li>
<li>Functional assessment of solid tumours
<ul>
<li>Mean, max and peak SUV</li>
</ul></li>
</ul></li>
<li>Cartilage for osteo-arthritis</li>
</ul>
<h4 id="role-of-quantitative-imaging-in-oncology">Role of quantitative imaging in oncology</h4>
<p><img src="/images/HMS/2314372.png" /></p>
<h4 id="evaluating-cancer-treatment-response">Evaluating cancer treatment response</h4>
<p>RECIST 1.1 – Response Evaluation Criteria in Solid Tumors</p>
<p>Simple measurement of tumor diameter.</p>
<ul>
<li>Maximum of 5 biggest target lesions</li>
<li>Baseline modules must measure &gt; 10 mm</li>
<li>Preferred imaging technique – CT （variability of MRI scan parameters）</li>
</ul>
<blockquote>
<p>WHO – Similar criteria based on two orthogonal measures</p>
</blockquote>
<p><img src="/images/HMS/2314373.png" /></p>
<h5 id="advantages-limitations">Advantages &amp; Limitations</h5>
<ul>
<li><p>Practical guidelines that can be applied on a wide range of image datasets of varying characteristics</p></li>
<li><p>Downsides</p>
<ul>
<li>One- or two-dimensional measurement for 3D structure</li>
<li>Only 5 top structures are taken into account</li>
<li>Long time between treatment effect and visible diameter change
<ul>
<li>Killing a tumor does not necessarily mean it disappears</li>
</ul></li>
<li>Variability between doctors</li>
<li>Time consuming</li>
</ul></li>
</ul>
<p><strong>Trend towards volumetric measurements</strong></p>
<h4 id="d-2d-vs-volumetric-3d">1D, 2D vs volumetric （3D）</h4>
<p><img src="/images/HMS/2314374.png" /></p>
<p>Assessment of cavitated lesions</p>
<ul>
<li>Tumoural volume decreased over time</li>
<li>1D or 2D tumour measurements increased over time</li>
</ul>
<h4 id="comparing-pet-scans">Comparing PET scans</h4>
<ul>
<li>Need to identify and delineate the tumors （segmentation）</li>
<li>Find the corresponding volume in the follow-up scan （registration）</li>
<li>For each tumor, derive a single SUV value</li>
<li>Change compared to an empirical threshold
<ul>
<li>Provided in recommendations such as EORTC, PERCIST</li>
</ul></li>
<li>Tedious when there are many tumor sites
<ul>
<li>For metastatic disease, <u>total metabolic tumour volume</u> is increasingly used</li>
</ul></li>
</ul>
<h4 id="variations-in-suv-measurements">Variations in SUV measurements</h4>
<p>Several statistics are currently being used to perform the measurement of tumour progression over time:</p>
<ul>
<li>Limited resolution complicates direct comparison of intensities</li>
<li>Partial volume effect, localization of lesions, etc.</li>
</ul>
<p>Mean of the tumour region</p>
<ul>
<li>Tumors are heterogenous, and activity is averaged out over a large volume</li>
</ul>
<p>Max over the tumour region</p>
<ul>
<li>Better, but sensitive to noise</li>
<li>Inherently, only based on one pixel</li>
</ul>
<p>Peak over the ROI</p>
<ul>
<li>Small, fixed-size region of interest ROI<sub>peak</sub></li>
<li>ROI cantered at high intensity region （max, variants exist）</li>
<li>Peak measurement is the average over that ROI</li>
</ul>
<p><img src="/images/HMS/2314375.png" /></p>
<h4 id="measuring-bone-cartilage-for-osteoarthritis">Measuring bone cartilage for osteoarthritis</h4>
<ul>
<li>The thickness of a thin layer of bone cartilage
<ul>
<li>How to measure and monitor over time?</li>
</ul></li>
<li>Image analysis workflow
<ul>
<li>Segmentation of structures</li>
<li>Unfold the cartilage surface</li>
<li>Perform measurements of
<ul>
<li>Thickness</li>
<li>Roughness</li>
</ul></li>
</ul></li>
</ul>
<p>Monitor local thickness and bone remodelling changes over time</p>
<p><img src="/images/HMS/2314376.png" /></p>
<h4 id="metabolic-tumour-volume-pet">Metabolic tumour volume （PET）</h4>
<p><img src="/images/HMS/2314377.png" /></p>
<h3 id="need-for-standardizing-measures">Need for standardizing measures</h3>
<ul>
<li>CT
<ul>
<li>CT Angiography</li>
<li>CT Volumetry</li>
<li>Lung Density</li>
<li>Small Lung Nodule</li>
</ul></li>
<li>MR
<ul>
<li>Arterial Spin Labeling （ASL） – {EIBIR supported}</li>
<li>Dynamic Contrast-Enhanced （DCE） MRI</li>
<li>Dynamic Susceptibility Contrast （DSC） MRI</li>
<li>Diffusion Tensor Imaging （DTI） MRI</li>
<li>Diffusion Weighted Imaging （DWI） MRI</li>
<li>Functional Magnetic Resonance Imaging （fMRI）</li>
<li>Magnetic Resonance Elastography （MRE）</li>
<li>Proton Density Fat Fraction （PDFF）</li>
<li>Musculoskeletal （MSK）</li>
</ul></li>
<li>NM
<ul>
<li>FDG-PET/CT</li>
<li>Quantifying Dopamine Transporters with 123 Iodine Labelled Ioflupane in Neurodegenerative Diseases （I-123）</li>
<li>PET-Amyloid</li>
<li>Technetium-99m for body, oncology and immunology （TC99m）</li>
</ul></li>
<li>US
<ul>
<li>Contrast Enhanced Ultrasound （CEUS）</li>
<li>Pulse-Echo Quantitative Ultrasound （PEQUS）– {AIUM supported}</li>
<li>Ultrasound Shear Wave Speed （SWS）</li>
<li>Ultrasound Volume Blood Flow （VBF） – {AIUM supported}</li>
</ul></li>
</ul>
<h2 id="image-features">Image features</h2>
<h3 id="feature-extraction-general-pipeline">Feature extraction: general pipeline</h3>
<p><img src="/images/HMS/2314378.png" /></p>
<ul>
<li>For each region of interest （ROI） or candidate region –
<ul>
<li>Often defined in space through a segmentation</li>
</ul></li>
<li>Extract the quantitative property that characterize the region
<ul>
<li>The appearance in terms of <strong>intensity</strong> and <strong>texture</strong></li>
<li>The appearance in terms of <strong>shape</strong> and <strong>outer surface</strong></li>
<li>The <strong>size</strong> and <strong>volume</strong> of the region</li>
<li>The （relative） <strong>position</strong> of the region</li>
<li>…</li>
</ul></li>
</ul>
<h3 id="intensity-based-features">Intensity-based features</h3>
<ul>
<li><p>Examples of intensity features over region Ω in image <code>f(x)</code> with N pixels</p>
<ul>
<li><p>Mean intensity value <span class="math display">\[
\mu = \frac{1}{N}\sum_{x∈\Omega}f(x)
\]</span></p></li>
<li><p>Median, minimum and maximum values</p></li>
<li><p>Standard deviation <span class="math display">\[
\sigma=\sqrt{\frac{1}{N}\sum_{x∈\Omega}(f(x)-\mu)^2}
\]</span></p></li>
<li><p>Energy <span class="math display">\[
E=\sum_{x∈\Omega}(f(x)+c)^2
\]</span></p>
<ul>
<li>c is a constant such that all values within the brackets are positive</li>
<li>Energy is related to the content of intensities, irrespective of conventions such as HU</li>
</ul></li>
</ul></li>
<li><p>Derived from the histogram</p>
<ul>
<li><p>Skewness</p>
<ul>
<li>Measure of asymmetry <span class="math display">\[
S_{sk}=\frac{\frac{1}{N}\sum_{x∈\Omega}(f(x)-\mu)^3}{\sigma^3}
\]</span></li>
</ul></li>
<li><p>Kurtosis</p>
<ul>
<li>Measure of relative weight of the tails <span class="math display">\[
S_{ku}=\frac{\frac{1}{N}\sum_{x∈\Omega}(f(x)-\mu)^4}{\sigma^4}
\]</span></li>
</ul></li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314379.png" /></p>
<h3 id="shape">Shape</h3>
<ul>
<li><p>Examples of shape features over region Ω in image <code>f(x)</code> with N pixels</p>
<ul>
<li><p>Volume</p>
<ul>
<li><p><span class="math display">\[
v = \frac{1}{N}\sum_{x∈\Omega}v(x)
\]</span></p></li>
<li><p><span class="math display">\[
v(x)=s_xs_ys_z
\]</span></p>
<p>the volume of a voxel</p></li>
</ul></li>
<li><p>Largest diameter</p>
<ul>
<li><p>D<sub>i</sub> the axes of the enclosing ellipsoid</p></li>
<li><p><span class="math display">\[
D_M=\max[D_1,D_2,D_3]
\]</span></p></li>
</ul></li>
<li><p>Smallest diameter</p>
<ul>
<li><span class="math display">\[
d_m=\min[D_1,D_2,D_3]
\]</span></li>
</ul></li>
<li><p>Elongation</p>
<ul>
<li><span class="math display">\[
\frac{D_M}{D_m}
\]</span></li>
</ul></li>
</ul></li>
<li><p>Examples considering the area A of the surface of the structure</p>
<ul>
<li><p>Relation to the surface of the compassing spheres</p></li>
<li><p>Surface Volume Ratio</p>
<ul>
<li><span class="math display">\[
R_{SV}=\frac{S}{V}
\]</span></li>
</ul></li>
<li><p>Compactness</p>
<ul>
<li><span class="math display">\[
C=\frac{V}{\sqrt{A^3\pi}}
\]</span></li>
</ul></li>
</ul></li>
</ul>
<h3 id="texture">Texture</h3>
<ul>
<li>What is texture?</li>
<li>No consensus on definition of texture in medical imaging literature
<ul>
<li>Associated with “coarse”, “edgy”, “directional”, “repetitive”, “random”, ...</li>
</ul></li>
<li>Oxford dictionary: The feel, appearance, or consistency of a surface or a substance</li>
<li></li>
<li></li>
</ul>
<h4 id="scale">Scale</h4>
<ul>
<li>The same pattern at a different scale, leads to a different texture</li>
<li>Given a certain scale,</li>
</ul>
<p><img src="/images/HMS/2314380.png" /></p>
<h4 id="direction-orientation">Direction / orientation</h4>
<ul>
<li>Textures may be have a certain orientation</li>
<li>Usually not relevant to measure the direction, but measures should take this into account</li>
</ul>
<p><img src="/images/HMS/2314381.png" /></p>
<h4 id="gray-level-co-occurrence-matrix-glcm">Gray-level co-occurrence matrix （GLCM）</h4>
<ul>
<li>The distribution of the number of occurrences of a pair of gray values and separated by a distance vector Δ = [Δ<sub>x</sub>, Δ<sub>y</sub>]</li>
</ul>
<p><span class="math display">\[
C\Delta_x,\Delta_y(i,j)=\sum^n_{x=1}\sum^m_{y=1}\left\{
\begin{aligned}
1 &amp; &amp; if\ I(x,y)=i\ and\ I(x+\Delta x,y+\Delta y)=j \\
0 &amp; &amp;otherwise
\end{aligned}
\right.
\]</span></p>
<p>Co-occurrence Matrix</p>
<ul>
<li>Distance =2</li>
<li>Angle = 0°</li>
</ul>
<p><img src="/images/HMS/2314382.png" /></p>
<p>The matrix leads to 13 Haralick texture descriptors</p>
<ul>
<li>Energy</li>
<li><strong>Entropy</strong></li>
<li>Correlation</li>
<li><strong>Contrast</strong></li>
<li>Inverse Difference Moment</li>
<li>Variance</li>
<li>Sum Mean</li>
<li>Inertia</li>
<li>Cluster Shade</li>
<li>Cluster Prominence</li>
<li>Max Probability</li>
<li>Inverse Variance</li>
<li>Mode Probability</li>
</ul>
<h5 id="second-order-histogram-statistics">Second-Order Histogram Statistics</h5>
<ul>
<li><p>Entropy of the matrix H <span class="math display">\[
S_H=-\sum^{y_t}_{y_q=y_1}\sum^{y_t}_{y_r=y_1}H(y_q,y_r,d)\log10[H(y_q,y_r,d)]
\]</span></p></li>
<li><p>Contrast of matrix H <span class="math display">\[
\sum^{y_t}_{y_q=y_1}\sum^{y_t}_{y_r=y_1}\partial(y_q,y_r)H(y_q,y_r,d)=\sum^{y_t}_{y_q=y_1}\sum^{y_t}_{y_r=y_1}(y_q-y_r)^2H(y_q,y_r,d)
\]</span></p></li>
</ul>
<h5 id="co-occurrence-matrices">Co-occurrence matrices</h5>
<ul>
<li>Global
<ul>
<li>Features extracted are for the entire cube</li>
</ul></li>
<li>13 Directions
<ul>
<li>Four original 2D directions</li>
<li>Nine new 3D directions</li>
</ul></li>
<li>4 Distances
<ul>
<li>1, 2, 4, and 8 pixels</li>
</ul></li>
<li>13 features extracted per distance per direction</li>
<li>13*4*13=676 features per cube</li>
</ul>
<p><img src="/images/HMS/2314383.png" /></p>
<h4 id="example-texture-feature-extraction">Example texture feature extraction</h4>
<p><img src="/images/HMS/2314384.png" /></p>
<h4 id="examples-of-textures">Examples of textures</h4>
<ul>
<li>Different textures of the lung
<ul>
<li>Associated to different pathologies</li>
<li>By locally applying textures operators, such pathologies can be detected and localised （segmented）</li>
</ul></li>
<li>Different texture of the bone
<ul>
<li>For grading severity of osteoporosis</li>
<li>Requires high resolution （μ）CT of the trabecular bone</li>
</ul></li>
</ul>
<h2 id="computer-aided-diagnosis">Computer-aided diagnosis</h2>
<p>Beyond extracting one feature</p>
<ul>
<li><p>Quantitative imaging</p></li>
<li><p>A radiologist will in practice look at several quantities</p>
<ul>
<li>Intensity</li>
<li>Shape</li>
<li>Texture</li>
</ul></li>
<li><p>Why not extract several features, and try to automate diagnosis?</p>
<p><strong>Computer-aided diagnosis （CAD）</strong></p>
<ul>
<li>Detection of structures</li>
<li>Tumor or not?</li>
<li>Benign or malignant?</li>
<li>Good or bad prognosis</li>
</ul></li>
</ul>
<h3 id="machine-learning-cad-approaches">Machine learning CAD approaches</h3>
<p>For each region of interest （ROI） or candidate region</p>
<ul>
<li>Often defined in space through a segmentation</li>
</ul>
<p>Extract <strong>all quantitative properties</strong> that characterize the region</p>
<ul>
<li>The appearance in terms of <strong>intensity</strong> and <strong>texture</strong></li>
<li>The appearance in terms of <strong>shape</strong> and <strong>outer surface</strong></li>
<li>The size and <strong>volume</strong> of the region</li>
<li>The （relative） <strong>position</strong> of the region</li>
<li>…</li>
</ul>
<p>Use training data and a <strong>classifier</strong> to learn and detect the appearance of a pathology</p>
<h3 id="introduction">Introduction</h3>
<ul>
<li>Definition
<ul>
<li>Systems that <strong>assist doctors in the interpretation</strong> of medical signals or images</li>
<li>Systems that process digital signals and images for <u>typical appearances and highlight conspicuous regions</u>; in order to offer input to support a decision taken by the professional</li>
</ul></li>
<li>Sometimes a distinction is made
<ul>
<li>Computer-aided detection （CADe） and computer-aided diagnosis （CADx）</li>
</ul></li>
<li>CAD has many applications in fields such as
<ul>
<li>Radiology</li>
<li>Nuclear medicine</li>
<li>Radiotherapy</li>
<li>Neurology</li>
<li>（Digital pathology）</li>
</ul></li>
</ul>
<h3 id="classification-and-regression">Classification and regression</h3>
<ul>
<li>Features
<ul>
<li>An individual measurable property or characteristic of a phenomenon being observed
<ul>
<li>X[i] or x[i]</li>
</ul></li>
</ul></li>
<li>Feature vector
<ul>
<li>N-dimensional vector of numerical features associated to one object or instance
<ul>
<li>X or x</li>
</ul></li>
</ul></li>
<li>Output or target
<ul>
<li>The desired result of your system for the given instances, the training data set
<ul>
<li>F（x）</li>
</ul></li>
</ul></li>
<li>（Class） Label
<ul>
<li>The output in case of a classification problem, the name characterizing the class
<ul>
<li>Dog, Cancer, Healthy, etc</li>
</ul></li>
</ul></li>
<li>Training data
<ul>
<li>Dataset of examples used for learning, e.g. training a classifier</li>
</ul></li>
<li>Test data
<ul>
<li>Dataset of examples used for testing the performance of the method
<ul>
<li>Independent of the training dataset!</li>
<li>From the same phenomena, has the same characteristics</li>
</ul></li>
</ul></li>
</ul>
<h3 id="classification">Classification</h3>
<ul>
<li>Given a set of data points and labels
<ul>
<li>Training set including the labels of classes C<sub>1</sub> or C<sub>2</sub></li>
<li>Train the classifier to lean the boundary</li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314385.png" /></p>
<ul>
<li>Given a new data point
<ul>
<li>Apply the classifier to determine C<sub>1</sub> or C<sub>2</sub></li>
</ul></li>
</ul>
<h3 id="pipeline">Pipeline</h3>
<p><img src="/images/HMS/2314386.png" /></p>
<p>Candidate detection</p>
<ul>
<li>Some way of detecting abnormalities</li>
<li>Randomly or evenly sample your image</li>
</ul>
<h3 id="example">Example</h3>
<h4 id="cad-of-lung-nodules-using-machine-learning">CAD of lung nodules using machine learning</h4>
<p><img src="/images/HMS/2314387.png" /></p>
<h5 id="pre-processing">Pre-processing</h5>
<ul>
<li>Resampling images to standard size
<ul>
<li>Isotropic resampling</li>
<li>E.g. 1 mm<sup>3</sup> voxel size</li>
</ul></li>
<li>Lung segmentation
<ul>
<li>Thresholding to extract low-intensity regions</li>
<li>Morphological operations to include holes</li>
</ul></li>
</ul>
<h5 id="candidate-detection">Candidate detection</h5>
<ul>
<li>Extract the nodule clusters</li>
<li>Divergence of normalized gradient
<ul>
<li>Retain points above empirically-determined thresholds</li>
</ul></li>
<li>Divergence points are compared with nodule clusters
<ul>
<li>Only clusters that correspond with divergence points are retained</li>
</ul></li>
</ul>
<h5 id="feature-extraction">Feature extraction</h5>
<ul>
<li>Shape</li>
<li>Volume</li>
<li>Elongation factor</li>
<li>Compactness</li>
<li>Lung wall separation</li>
<li>Mean value</li>
<li><strong>Mean value over divergence values</strong></li>
<li><strong>Mean value over nodule &amp; vessel enhanced images</strong></li>
<li>Median value</li>
<li>Maximum and minimum values</li>
<li>standard deviation</li>
</ul>
<h5 id="data-balancing">Data balancing</h5>
<ul>
<li><p>After candidate detection over the training set</p>
<ul>
<li><p>111332 candidates detected</p></li>
<li><p>~ 500 true nodules （true positives）</p>
<p><strong>Too many negatives, a typical problem in CAD</strong></p></li>
</ul></li>
<li><p>Data balancing methods</p>
<ul>
<li>Downsample your negatives
<ul>
<li>Select representative samples, e.g. SOM</li>
</ul></li>
<li>Upsample your positives
<ul>
<li>Generate similar / plausible samples in feature space</li>
</ul></li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314388.png" /></p>
<h5 id="classification-1">Classification</h5>
<ul>
<li>Support vector machine
<ul>
<li>Using a radial basis function （RBF）</li>
<li>Using a soft margin</li>
</ul></li>
<li>Tuning the parameters
<ul>
<li>10-fold cross-validation on the training set</li>
</ul></li>
</ul>
<p><img src="/images/HMS/2314389.png" /></p>
<h2 id="radiomics"><a target="_blank" rel="noopener" href="https://www.youtube.com/watch?v=Tq980GEVP0Y">Radiomics</a></h2>
<h3 id="introduction-1">Introduction</h3>
<ul>
<li>Definition
<ul>
<li>Radiomics is a process for <strong>high-throughput extraction</strong> of <strong>quantitative features</strong> that result in the conversion of images into <strong>mineable data</strong>, and the subsequent analysis of these data for <strong>decision support</strong>.</li>
</ul></li>
<li>Process
<ul>
<li>Extract a lot of （quantitative） features</li>
<li>Try to make <strong>predictions</strong> on the future outcome of the treatment</li>
</ul></li>
</ul>
<h3 id="pipeline-1">Pipeline</h3>
<p><img src="/images/HMS/2314390.png" /></p>
<ul>
<li>Image acquisition</li>
<li>Volume of interest identification and segmentation</li>
<li>Feature extraction and qualification</li>
<li>Learning algorithms</li>
</ul>
<p><img src="/images/HMS/2314391.png" /></p>
<ol type="1">
<li>Example CT images of lung cancer patients （left）, 3D visualizations right. Note strong phenotypic differences that can be captured with routine CT imaging, such as heterogeneity and shape.</li>
<li>Process for extracting radiomics data from images.
<ol type="1">
<li>Manual/automated segmentation</li>
<li>Features are extracted from within the defined tumour contours, quantifying tumour intensity, shape, texture and wavelet texture.</li>
<li>Analysis of radiomics features through comparison with clinical data, gene-expression data….</li>
</ol></li>
</ol>
<h3 id="what-type-of-predictions">What type of predictions?</h3>
<p><strong>Beyond diagnosis and staging of the pathology</strong></p>
<ul>
<li>Based on the tumor
<ul>
<li>Intensity</li>
<li>Shape</li>
<li>Texture</li>
<li>Frequency content</li>
</ul></li>
<li>Predict the patient survival
<ul>
<li>Will the patient be alive or not in X years?</li>
</ul></li>
<li>Predict the response to a certain therapy
<ul>
<li>Will the patient respond （PR or CR） or not?</li>
</ul></li>
</ul>
<h3 id="example-1">Example</h3>
<h4 id="prediction-based-on-radiomics">Prediction based on radiomics</h4>
<p>For one feature （energy）</p>
<ul>
<li>Take the median for dataset Lung1</li>
<li>Plot the survival curves for values &gt;median and &lt; median for all datasets</li>
</ul>
<p><img src="/images/HMS/2314392.png" /></p>
<p>The curves are clearly different, demonstrating the <strong>prognostic power of the features for the population</strong></p>
<p>Deriving reliable and useful decision support for the individual remains the challenge</p>
<h4 id="a-recent-headline">A recent headline</h4>
<p>Google’s new AI algorithm predicts heart disease by looking at your eyes</p>
<h3 id="the-future">The future?</h3>
<p>Towards Automated analysis of images</p>
<ul>
<li>Yes, without a doubt</li>
<li>But how will we integrate it in the workflow?
<ul>
<li>Only AI reading the images?</li>
<li>Second reading of the computer?
<ul>
<li>Helping the doctors</li>
<li>Checking the doctors</li>
</ul></li>
<li>Running additional readings in the background</li>
</ul></li>
</ul>
<h1 id="summary">Summary</h1>
<h2 id="quantitative-imaging-2">Quantitative Imaging</h2>
<ul>
<li>The extraction of quantifiable features from medical images for the assessment of normal or the severity, degree of change, or status of a disease, injury, or chronic condition relative to normal</li>
<li>In this context, quantitative implies reproducible, comparable, replicable</li>
<li>Obtaining quantitative measures puts requirements on the image acquisition, and the image analysis / processing</li>
<li>In CT, quantitative measures can be obtained on intensity and morphology</li>
<li>In MRI, quantitative measures can be obtained on morphology. Intensities can not be directly compared. Specific MRI sequences or analyses, lead to physical quantities, which can be compared</li>
<li>In PET, quantitative measurements of intensity can be obtained using standardized uptake values （SUV）</li>
<li>To reliably extract quantitative measures from images, there is a need for standardization and clear criteria
<ul>
<li>QIBA alliance for standard measurement of imaging biomarkers</li>
<li>RECIST or WHO response criteria in oncology</li>
</ul></li>
</ul>
<h2 id="feature-extraction-1">Feature extraction</h2>
<ul>
<li>The general pipeline involves
<ul>
<li>the acquisition of images</li>
<li>Pre-processing and segmentation of the region of interest</li>
<li>Extraction of the quantitative measure</li>
<li>Diagnostic assessment based on the measure</li>
</ul></li>
<li>Intensity features cover
<ul>
<li>The mean and standard deviation</li>
<li>The median</li>
<li>The energy</li>
<li>The skewness and kurtosis</li>
<li>…</li>
</ul></li>
<li>Shape features cover
<ul>
<li>The volume</li>
<li>The elongation</li>
<li>The compactness</li>
<li>…</li>
</ul></li>
<li>Another group of features is related to texture （not covered in depth here）</li>
</ul>
<h2 id="computer-aided-diagnosis-1">Computer-aided diagnosis</h2>
<ul>
<li>The general pipeline involves
<ul>
<li>The acquisition of images</li>
<li>Pre-processing</li>
<li>Detection of abnormalities and segmentation</li>
<li>Extraction of many quantitative image features</li>
<li>Classification of the abnormality</li>
</ul></li>
<li>Classification is a supervised machine learning technique
<ul>
<li>Based on a training dataset, learn an optimal decision boundary between classes</li>
<li>Given a new data point, decide to which class the point belongs</li>
</ul></li>
<li>Example of a full pipeline for a CAD problem for lung nodules
<ul>
<li>Pre-processing: images are resampled, lungs are segmented</li>
<li>Candidate detection: detect suspicious structures</li>
<li>Reduce the amount of false positives using feature extraction and classification
<ul>
<li>First train a classifier based on the features of a training dataset</li>
<li>Then apply the classifier based on the feature of the input structure</li>
</ul></li>
</ul></li>
</ul>
<h2 id="radiomics-1">Radiomics</h2>
<ul>
<li>The general pipeline involves
<ul>
<li>The acquisition of images</li>
<li>Segmentation of tumours</li>
<li>Extraction of many quantitative image features</li>
<li>Classification based on clinical endpoints （survival, gene expression, etc.）</li>
</ul></li>
<li>Radiomic signatures are extracted from the tumor
<ul>
<li>Including intensity statistics</li>
<li>Shape features</li>
<li>Texture features</li>
<li>Wavelet features</li>
</ul></li>
<li>Signatures were shown to correlated with the outcome of the patients （survival）</li>
</ul>

                
            </div>
            <hr/>

            

    <div class="reprint" id="reprint-statement">
        
            <div class="reprint__author">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-user">
                        Author:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="/about" rel="external nofollow noreferrer">Chaoqun Yin</a>
                </span>
            </div>
            <div class="reprint__type">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-link">
                        Link:
                    </i>
                </span>
                <span class="reprint-info">
                    <a href="https://yintelligence.tech/2021/04/26/2021-04-26-Quantitative-imaging/">https://yintelligence.tech/2021/04/26/2021-04-26-Quantitative-imaging/</a>
                </span>
            </div>
            <div class="reprint__notice">
                <span class="reprint-meta" style="font-weight: bold;">
                    <i class="fas fa-copyright">
                        Reprint policy:
                    </i>
                </span>
                <span class="reprint-info">
                    All articles in this blog are used except for special statements
                    <a href="https://creativecommons.org/licenses/by/4.0/deed.zh" rel="external nofollow noreferrer" target="_blank">CC BY 4.0</a>
                    reprint polocy. If reproduced, please indicate source
                    <a href="/about" target="_blank">Chaoqun Yin</a>
                    !
                </span>
            </div>
        
    </div>

    <script async defer>
      document.addEventListener("copy", function (e) {
        let toastHTML = '<span>Copied successfully, please follow the reprint policy of this article</span><button class="btn-flat toast-action" onclick="navToReprintStatement()" style="font-size: smaller">more</a>';
        M.toast({html: toastHTML})
      });

      function navToReprintStatement() {
        $("html, body").animate({scrollTop: $("#reprint-statement").offset().top - 80}, 800);
      }
    </script>



            <div class="tag_share" style="display: block;">
                <div class="post-meta__tag-list" style="display: inline-block;">
                    
                        <div class="article-tag">
                            
                                <a href="/tags/Biomedical-Signals-and-Images/">
                                    <span class="chip bg-color">Biomedical Signals and Images</span>
                                </a>
                            
                        </div>
                    
                </div>
                <div class="post_share" style="zoom: 80%; width: fit-content; display: inline-block; float: right; margin: -0.15rem 0;">
                    <link rel="stylesheet" type="text/css" href="/libs/share/css/share.min.css">
<div id="article-share">

    
    <div class="social-share" data-sites="twitter,facebook,google,qq,qzone,wechat,weibo,douban,linkedin" data-wechat-qrcode-helper="<p>微信扫一扫即可分享！</p>"></div>
    <script src="/libs/share/js/social-share.min.js"></script>
    

    

</div>

                </div>
            </div>
            
        </div>
    </div>

    

    

    

    

    

    

    

    

<article id="prenext-posts" class="prev-next articles">
    <div class="row article-row">
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge left-badge text-color">
                <i class="fas fa-chevron-left"></i>&nbsp;Previous</div>
            <div class="card">
                <a href="/2021/04/27/2021-04-27-HTML/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/3.jpg" class="responsive-img" alt="HTML">
                        
                        <span class="card-title">HTML</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            Exercise 3
                        
                    </div>
                    <div class="publish-info">
                        <span class="publish-date">
                            <i class="far fa-clock fa-fw icon-date"></i>2021-04-27
                        </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/Network/" class="post-category">
                                    Network
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Web-Technologies/">
                        <span class="chip bg-color">Web Technologies</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
        
        <div class="article col s12 m6" data-aos="fade-up">
            <div class="article-badge right-badge text-color">
                Next&nbsp;<i class="fas fa-chevron-right"></i>
            </div>
            <div class="card">
                <a href="/2021/04/25/2021-04-25-HTTP-Protocol/">
                    <div class="card-image">
                        
                        
                        <img src="/medias/featureimages/16.jpg" class="responsive-img" alt="HTTP Protocol">
                        
                        <span class="card-title">HTTP Protocol</span>
                    </div>
                </a>
                <div class="card-content article-content">
                    <div class="summary block-with-text">
                        
                            Exercise 2 - Hands-on experience with the HTTP protocol
                        
                    </div>
                    <div class="publish-info">
                            <span class="publish-date">
                                <i class="far fa-clock fa-fw icon-date"></i>2021-04-25
                            </span>
                        <span class="publish-author">
                            
                            <i class="fas fa-bookmark fa-fw icon-category"></i>
                            
                            <a href="/categories/Network/" class="post-category">
                                    Network
                                </a>
                            
                            
                        </span>
                    </div>
                </div>
                
                <div class="card-action article-tags">
                    
                    <a href="/tags/Web-Technologies/">
                        <span class="chip bg-color">Web Technologies</span>
                    </a>
                    
                </div>
                
            </div>
        </div>
        
    </div>
</article>

</div>


<script>
    $('#articleContent').on('copy', function (e) {
        // IE8 or earlier browser is 'undefined'
        if (typeof window.getSelection === 'undefined') return;

        var selection = window.getSelection();
        // if the selection is short let's not annoy our users.
        if (('' + selection).length < Number.parseInt('120')) {
            return;
        }

        // create a div outside of the visible area and fill it with the selected text.
        var bodyElement = document.getElementsByTagName('body')[0];
        var newdiv = document.createElement('div');
        newdiv.style.position = 'absolute';
        newdiv.style.left = '-99999px';
        bodyElement.appendChild(newdiv);
        newdiv.appendChild(selection.getRangeAt(0).cloneContents());

        // we need a <pre> tag workaround.
        // otherwise the text inside "pre" loses all the line breaks!
        if (selection.getRangeAt(0).commonAncestorContainer.nodeName === 'PRE') {
            newdiv.innerHTML = "<pre>" + newdiv.innerHTML + "</pre>";
        }

        var url = document.location.href;
        newdiv.innerHTML += '<br />'
            + 'From: Chaoqun Yin<br />'
            + 'Author: Chaoqun Yin<br />'
            + 'Link: <a href="' + url + '">' + url + '</a><br />'
            + '本文章著作权归作者所有，任何形式的转载都请注明出处。';

        selection.selectAllChildren(newdiv);
        window.setTimeout(function () {bodyElement.removeChild(newdiv);}, 200);
    });
</script>


<!-- 代码块功能依赖 -->
<script type="text/javascript" src="/libs/codeBlock/codeBlockFuction.js"></script>

<!-- 代码语言 -->

<script type="text/javascript" src="/libs/codeBlock/codeLang.js"></script>


<!-- 代码块复制 -->

<script type="text/javascript" src="/libs/codeBlock/codeCopy.js"></script>


<!-- 代码块收缩 -->

<script type="text/javascript" src="/libs/codeBlock/codeShrink.js"></script>


    </div>
    <div id="toc-aside" class="expanded col l3 hide-on-med-and-down">
        <div class="toc-widget card" style="background-color: white;">
            <div class="toc-title"><i class="far fa-list-alt"></i>&nbsp;&nbsp;TOC</div>
            <div id="toc-content"></div>
        </div>
    </div>
</div>

<!-- TOC 悬浮按钮. -->

<div id="floating-toc-btn" class="hide-on-med-and-down">
    <a class="btn-floating btn-large bg-color">
        <i class="fas fa-list-ul"></i>
    </a>
</div>


<script src="/libs/tocbot/tocbot.min.js"></script>
<script>
    $(function () {
        tocbot.init({
            tocSelector: '#toc-content',
            contentSelector: '#articleContent',
            headingsOffset: -($(window).height() * 0.4 - 45),
            collapseDepth: Number('0'),
            headingSelector: 'h1, h2, h3, h4'
        });

        // modify the toc link href to support Chinese.
        let i = 0;
        let tocHeading = 'toc-heading-';
        $('#toc-content a').each(function () {
            $(this).attr('href', '#' + tocHeading + (++i));
        });

        // modify the heading title id to support Chinese.
        i = 0;
        $('#articleContent').children('h1, h2, h3, h4').each(function () {
            $(this).attr('id', tocHeading + (++i));
        });

        // Set scroll toc fixed.
        let tocHeight = parseInt($(window).height() * 0.4 - 64);
        let $tocWidget = $('.toc-widget');
        $(window).scroll(function () {
            let scroll = $(window).scrollTop();
            /* add post toc fixed. */
            if (scroll > tocHeight) {
                $tocWidget.addClass('toc-fixed');
            } else {
                $tocWidget.removeClass('toc-fixed');
            }
        });

        
        /* 修复文章卡片 div 的宽度. */
        let fixPostCardWidth = function (srcId, targetId) {
            let srcDiv = $('#' + srcId);
            if (srcDiv.length === 0) {
                return;
            }

            let w = srcDiv.width();
            if (w >= 450) {
                w = w + 21;
            } else if (w >= 350 && w < 450) {
                w = w + 18;
            } else if (w >= 300 && w < 350) {
                w = w + 16;
            } else {
                w = w + 14;
            }
            $('#' + targetId).width(w);
        };

        // 切换TOC目录展开收缩的相关操作.
        const expandedClass = 'expanded';
        let $tocAside = $('#toc-aside');
        let $mainContent = $('#main-content');
        $('#floating-toc-btn .btn-floating').click(function () {
            if ($tocAside.hasClass(expandedClass)) {
                $tocAside.removeClass(expandedClass).hide();
                $mainContent.removeClass('l9');
            } else {
                $tocAside.addClass(expandedClass).show();
                $mainContent.addClass('l9');
            }
            fixPostCardWidth('artDetail', 'prenext-posts');
        });
        
    });
</script>

    

</main>


<script src="https://cdn.bootcss.com/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
<script>
    MathJax.Hub.Config({
        tex2jax: {inlineMath: [['$', '$'], ['\(', '\)']]}
    });
</script>



    <footer class="page-footer bg-color">
    
    <div class="container row center-align" style="margin-bottom: 15px !important;">
        <div class="col s12 m8 l8 copy-right">
            Copyright&nbsp;&copy;
            
                <span id="year">2021</span>
            
            <span id="year">2021</span>
            <a href="/about" target="_blank">Chaoqun Yin</a>
            |&nbsp;Powered by&nbsp;<a href="https://hexo.io/" target="_blank">Hexo</a>
            |&nbsp;Theme&nbsp;<a href="https://github.com/blinkfox/hexo-theme-matery" target="_blank">Matery</a>
            <br>
            
            
            
            
            
            
            <span id="busuanzi_container_site_pv">
                |&nbsp;<i class="far fa-eye"></i>&nbsp;总访问量:&nbsp;<span id="busuanzi_value_site_pv"
                    class="white-color"></span>&nbsp;次
            </span>
            
            
            <span id="busuanzi_container_site_uv">
                |&nbsp;<i class="fas fa-users"></i>&nbsp;总访问人数:&nbsp;<span id="busuanzi_value_site_uv"
                    class="white-color"></span>&nbsp;人
            </span>
            
            <br>
            
            <span id="sitetime">载入运行时间...</span>
            <script>
                function siteTime() {
                    var seconds = 1000;
                    var minutes = seconds * 60;
                    var hours = minutes * 60;
                    var days = hours * 24;
                    var years = days * 365;
                    var today = new Date();
                    var startYear = "2021";
                    var startMonth = "1";
                    var startDate = "28";
                    var startHour = "2";
                    var startMinute = "1";
                    var startSecond = "0";
                    var todayYear = today.getFullYear();
                    var todayMonth = today.getMonth() + 1;
                    var todayDate = today.getDate();
                    var todayHour = today.getHours();
                    var todayMinute = today.getMinutes();
                    var todaySecond = today.getSeconds();
                    var t1 = Date.UTC(startYear, startMonth, startDate, startHour, startMinute, startSecond);
                    var t2 = Date.UTC(todayYear, todayMonth, todayDate, todayHour, todayMinute, todaySecond);
                    var diff = t2 - t1;
                    var diffYears = Math.floor(diff / years);
                    var diffDays = Math.floor((diff / days) - diffYears * 365);
                    var diffHours = Math.floor((diff - (diffYears * 365 + diffDays) * days) / hours);
                    var diffMinutes = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours) /
                        minutes);
                    var diffSeconds = Math.floor((diff - (diffYears * 365 + diffDays) * days - diffHours * hours -
                        diffMinutes * minutes) / seconds);
                    if (startYear == todayYear) {
                        document.getElementById("year").innerHTML = todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已安全运行 " + diffDays + " 天 " + diffHours +
                            " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    } else {
                        document.getElementById("year").innerHTML = startYear + " - " + todayYear;
                        document.getElementById("sitetime").innerHTML = "本站已安全运行 " + diffYears + " 年 " + diffDays +
                            " 天 " + diffHours + " 小时 " + diffMinutes + " 分钟 " + diffSeconds + " 秒";
                    }
                }
                setInterval(siteTime, 1000);
            </script>
            
            <br>
            
        </div>
        <div class="col s12 m4 l4 social-link social-statis">
    <a href="https://github.com/yintelligence" class="tooltipped" target="_blank" data-tooltip="访问我的GitHub" data-position="top" data-delay="50">
        <i class="fab fa-github"></i>
    </a>



    <a href="mailto:Yintelligence@outlook.com" class="tooltipped" target="_blank" data-tooltip="邮件联系我" data-position="top" data-delay="50">
        <i class="fas fa-envelope-open"></i>
    </a>













</div>
    </div>
</footer>

<div class="progress-bar"></div>


    <!-- 搜索遮罩框 -->
<div id="searchModal" class="modal">
    <div class="modal-content">
        <div class="search-header">
            <span class="title"><i class="fas fa-search"></i>&nbsp;&nbsp;Search</span>
            <input type="search" id="searchInput" name="s" placeholder="Please enter a search keyword"
                   class="search-input">
        </div>
        <div id="searchResult"></div>
    </div>
</div>

<script type="text/javascript">
$(function () {
    var searchFunc = function (path, search_id, content_id) {
        'use strict';
        $.ajax({
            url: path,
            dataType: "xml",
            success: function (xmlResponse) {
                // get the contents from search data
                var datas = $("entry", xmlResponse).map(function () {
                    return {
                        title: $("title", this).text(),
                        content: $("content", this).text(),
                        url: $("url", this).text()
                    };
                }).get();
                var $input = document.getElementById(search_id);
                var $resultContent = document.getElementById(content_id);
                $input.addEventListener('input', function () {
                    var str = '<ul class=\"search-result-list\">';
                    var keywords = this.value.trim().toLowerCase().split(/[\s\-]+/);
                    $resultContent.innerHTML = "";
                    if (this.value.trim().length <= 0) {
                        return;
                    }
                    // perform local searching
                    datas.forEach(function (data) {
                        var isMatch = true;
                        var data_title = data.title.trim().toLowerCase();
                        var data_content = data.content.trim().replace(/<[^>]+>/g, "").toLowerCase();
                        var data_url = data.url;
                        data_url = data_url.indexOf('/') === 0 ? data.url : '/' + data_url;
                        var index_title = -1;
                        var index_content = -1;
                        var first_occur = -1;
                        // only match artiles with not empty titles and contents
                        if (data_title !== '' && data_content !== '') {
                            keywords.forEach(function (keyword, i) {
                                index_title = data_title.indexOf(keyword);
                                index_content = data_content.indexOf(keyword);
                                if (index_title < 0 && index_content < 0) {
                                    isMatch = false;
                                } else {
                                    if (index_content < 0) {
                                        index_content = 0;
                                    }
                                    if (i === 0) {
                                        first_occur = index_content;
                                    }
                                }
                            });
                        }
                        // show search results
                        if (isMatch) {
                            str += "<li><a href='" + data_url + "' class='search-result-title'>" + data_title + "</a>";
                            var content = data.content.trim().replace(/<[^>]+>/g, "");
                            if (first_occur >= 0) {
                                // cut out 100 characters
                                var start = first_occur - 20;
                                var end = first_occur + 80;
                                if (start < 0) {
                                    start = 0;
                                }
                                if (start === 0) {
                                    end = 100;
                                }
                                if (end > content.length) {
                                    end = content.length;
                                }
                                var match_content = content.substr(start, end);
                                // highlight all keywords
                                keywords.forEach(function (keyword) {
                                    var regS = new RegExp(keyword, "gi");
                                    match_content = match_content.replace(regS, "<em class=\"search-keyword\">" + keyword + "</em>");
                                });

                                str += "<p class=\"search-result\">" + match_content + "...</p>"
                            }
                            str += "</li>";
                        }
                    });
                    str += "</ul>";
                    $resultContent.innerHTML = str;
                });
            }
        });
    };

    searchFunc('/search.xml', 'searchInput', 'searchResult');
});
</script>

    <!-- 回到顶部按钮 -->
<div id="backTop" class="top-scroll">
    <a class="btn-floating btn-large waves-effect waves-light" href="#!">
        <i class="fas fa-arrow-up"></i>
    </a>
</div>


    <script src="/libs/materialize/materialize.min.js"></script>
    <script src="/libs/masonry/masonry.pkgd.min.js"></script>
    <script src="/libs/aos/aos.js"></script>
    <script src="/libs/scrollprogress/scrollProgress.min.js"></script>
    <script src="/libs/lightGallery/js/lightgallery-all.min.js"></script>
    <script src="/js/matery.js"></script>

    <!-- Baidu Analytics -->

    <!-- Baidu Push -->

<script>
    (function () {
        var bp = document.createElement('script');
        var curProtocol = window.location.protocol.split(':')[0];
        if (curProtocol === 'https') {
            bp.src = 'https://zz.bdstatic.com/linksubmit/push.js';
        } else {
            bp.src = 'http://push.zhanzhang.baidu.com/push.js';
        }
        var s = document.getElementsByTagName("script")[0];
        s.parentNode.insertBefore(bp, s);
    })();
</script>

    
    <script src="/libs/others/clicklove.js" async="async"></script>
    
    
    <script async src="/libs/others/busuanzi.pure.mini.js"></script>
    

    

    

	
    

    

    

    
    <script src="/libs/instantpage/instantpage.js" type="module"></script>
    

</body>

</html>
